// ids-enterprise-ng/esm2022/lib/input-validate/soho-input-validate.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoInputValidateDirective } from './soho-input-validate.directive';
import * as i0 from "@angular/core";
export class SohoInputValidateModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoInputValidateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoInputValidateModule, declarations: [SohoInputValidateDirective], imports: [CommonModule], exports: [SohoInputValidateDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoInputValidateModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoInputValidateModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoInputValidateDirective
                    ],
                    exports: [
                        SohoInputValidateDirective
                    ]
                }]
        }] });
