// ids-enterprise-ng/esm2022/lib/breadcrumb/soho-breadcrumb.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoBreadcrumbComponent, SohoBreadcrumbListComponent } from './soho-breadcrumb.component';
import * as i0 from "@angular/core";
export class SohoBreadcrumbModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBreadcrumbModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoBreadcrumbModule, declarations: [SohoBreadcrumbComponent,
            SohoBreadcrumbListComponent], imports: [CommonModule], exports: [SohoBreadcrumbComponent,
            SohoBreadcrumbListComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBreadcrumbModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBreadcrumbModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoBreadcrumbComponent,
                        SohoBreadcrumbListComponent
                    ],
                    exports: [
                        SohoBreadcrumbComponent,
                        SohoBreadcrumbListComponent
                    ]
                }]
        }] });
