// ids-enterprise-ng/esm2022/lib/contextual-action-panel/soho-contextual-action-panel.service.mjs
import { Injectable } from '@angular/core';
import { ArgumentHelper } from '../utils/argument.helper';
import { SohoContextualActionPanelRef } from './soho-contextual-action-panel.ref';
import * as i0 from "@angular/core";
/**
 * This service is used to create a panel panel, based on the content
 * of an Angular Component on the screen.
 */
export class SohoContextualActionPanelService {
    /**
     * Constructor.
     *
     * @param appRef - application reference; must not be null.
     * @param componentFactoryResolver - used to create component factories for components dynamically.
     * @param injector - the current in scope injector, use as a delegate.
     * @param ngZone - the angular zone; must not be null.
     */
    constructor(appRef, componentFactoryResolver, injector, ngZone) {
        this.appRef = appRef;
        this.componentFactoryResolver = componentFactoryResolver;
        this.injector = injector;
        this.ngZone = ngZone;
    }
    /**
     * Creates a panel defined by the given component, returning a reference to the
     * panel that can be interacted with.
     *
     * The component has the potential to include the panel reference when it is
     * instantiated, as a provider.
     *
     * The panel won't necessarily be open yet, see the panel ref Api for further methods.
     *
     * @param component - the type of the component to instantiate; must not be null.
     * @param parent - the parent container; must not be null.
     * @param options - the default options for the panel; optional.
     *
     * @return the panel reference.
     */
    contextualactionpanel(component, parent, options) {
        ArgumentHelper.checkNotNull('component', component);
        ArgumentHelper.checkNotNull('parent', parent);
        return new SohoContextualActionPanelRef(this.appRef, this.componentFactoryResolver, this.injector, this.ngZone, options, component);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextualActionPanelService, deps: [{ token: i0.ApplicationRef }, { token: i0.ComponentFactoryResolver }, { token: i0.Injector }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextualActionPanelService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextualActionPanelService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.ApplicationRef }, { type: i0.ComponentFactoryResolver }, { type: i0.Injector }, { type: i0.NgZone }]; } });
