// ids-enterprise-ng/esm2022/lib/progress/soho-progress.component.mjs
import { Component, ChangeDetectionStrategy, Output, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoProgressComponent {
    /*
     *  @param value
     */
    set progressValue(value) {
        this.value = value;
        if (this.progress) {
            this.progress.update(this.value);
        }
    }
    get progressValue() {
        return this.value;
    }
    get dispType() {
        return 'block';
    }
    constructor(element) {
        this.element = element;
        // Set the progress-bar class.
        this.progressClass = true;
        this.options = {};
    }
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement.children[0]);
        this.jQueryElement.progress(this.options);
        /**
         * Bind to jQueryElement's events
         */
        this.jQueryElement.on('change', (event) => this.change?.emit(event));
        this.progress = this.jQueryElement.data('progress');
    }
    ngOnDestroy() {
        if (this.progress) {
            this.progress.destroy();
            this.progress = null;
        }
    }
    update(value) {
        if (this.progress) {
            this.progress.update(value);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoProgressComponent, selector: "soho-progress", inputs: { progressValue: "progressValue" }, outputs: { change: "change" }, host: { properties: { "class.progress": "this.progressClass", "style.display": "this.dispType" } }, ngImport: i0, template: `<div [class.progress-bar]="true" [attr.data-value]="value"></div>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-progress',
                    template: `<div [class.progress-bar]="true" [attr.data-value]="value"></div>`,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { progressValue: [{
                type: Input
            }], change: [{
                type: Output
            }], progressClass: [{
                type: HostBinding,
                args: ['class.progress']
            }], dispType: [{
                type: HostBinding,
                args: ['style.display']
            }] } });
