// ids-enterprise-ng/esm2022/lib/progress/soho-progress.module.mjs
import { NgModule } from '@angular/core';
import { SohoProgressComponent } from './soho-progress.component';
import * as i0 from "@angular/core";
export class SohoProgressModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressModule, declarations: [SohoProgressComponent], exports: [SohoProgressComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoProgressModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [
                        SohoProgressComponent
                    ],
                    exports: [
                        SohoProgressComponent
                    ]
                }]
        }] });
