// ids-enterprise-ng/esm2022/lib/swaplist/soho-swaplist.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoButtonModule } from '../button/soho-button.module';
import { SohoSwapListCardComponent, SohoSwapListComponent } from './soho-swaplist.component';
import * as i0 from "@angular/core";
export class SohoSwapListModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwapListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoSwapListModule, declarations: [SohoSwapListCardComponent,
            SohoSwapListComponent], imports: [CommonModule,
            SohoButtonModule], exports: [SohoSwapListCardComponent,
            SohoSwapListComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwapListModule, imports: [CommonModule,
            SohoButtonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwapListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoButtonModule
                    ],
                    declarations: [
                        SohoSwapListCardComponent,
                        SohoSwapListComponent
                    ],
                    exports: [
                        SohoSwapListCardComponent,
                        SohoSwapListComponent
                    ]
                }]
        }] });
