// ids-enterprise-ng/esm2022/lib/autocomplete/soho-autocomplete.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoAutoCompleteComponent } from './soho-autocomplete.component';
import * as i0 from "@angular/core";
export class SohoAutoCompleteModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAutoCompleteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoAutoCompleteModule, declarations: [SohoAutoCompleteComponent], imports: [CommonModule], exports: [SohoAutoCompleteComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAutoCompleteModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAutoCompleteModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoAutoCompleteComponent
                    ],
                    exports: [
                        SohoAutoCompleteComponent
                    ],
                }]
        }] });
