// ids-enterprise-ng/esm2022/lib/error/soho-error.directive.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Angular Wrapper for the SoHo Error Directive.
 *
 * @deprecated - use SohoAlert instead
 */
export class SohoErrorDirective {
    constructor(el) {
        this.el = el;
        this._options = {}; // eslint:disable-line
    }
    /**
     *  message for the error.
     *
     * @deprecated - use SohoAlert instead
     */
    set message(message) {
        this._options.message = message;
        if (this.jQueryElement) {
            if (message) {
                this.jQueryElement.addError(this._options);
            }
            else {
                this.jQueryElement.removeError(this._options);
            }
        }
    }
    /**
     *  set error in tooltip.
     *
     * @deprecated - use SohoAlert instead
     */
    set showTooltip(showTooltip) {
        this._options.showTooltip = showTooltip;
    }
    /**
     * set error to be inline.
     *
     * @deprecated - use SohoAlert instead
     */
    set inline(inline) {
        this._options.inline = inline;
    }
    /**
     * Returns the data object data-errormessage
     *
     * @deprecated - use SohoAlert instead
     */
    get errorMessage() {
        return this.jQueryElement?.getErrorMessage(this._options);
    }
    /**
     * add inline error
     *
     * @deprecated - use SohoAlert instead
     */
    addInlineError(message) {
        this.showTooltip = false; // eslint-disable-line
        this.inline = true; // eslint-disable-line
        this.message = message; // eslint-disable-line
    }
    /**
     * add tooltip error
     *
     * @deprecated - use SohoAlert instead
     */
    addTooltipError(message) {
        this.showTooltip = true; // eslint-disable-line
        this.inline = false; // eslint-disable-line
        this.message = message; // eslint-disable-line
    }
    /**
     * remove inline/tooltip error
     *
     * @deprecated - use SohoAlert instead
     */
    removeError() {
        this.message = ''; // eslint-disable-line
    }
    /**
     * Scrolls the element into the visible area of the browser window
     *
     * @deprecated - use SohoAlert instead
     * @param alignToTop (boolean) optional - true (default) element will be aligned to the top of the visible area of the scrollable ancestor
     */
    scrollIntoView(alignToTop) {
        this.jQueryElement?.scrollIntoView(alignToTop, this._options);
    }
    /**
     * After the control has been initialised and the view is ready,
     * get the SoHoXi controls to activate any validations.
     */
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.el.nativeElement);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoErrorDirective, selector: "[soho-error]", inputs: { message: "message", showTooltip: "showTooltip", inline: "inline" }, exportAs: ["soho-error"], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-error]',
                    exportAs: 'soho-error'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { message: [{
                type: Input
            }], showTooltip: [{
                type: Input
            }], inline: [{
                type: Input
            }] } });
