// ids-enterprise-ng/esm2022/lib/error/soho-error.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoErrorDirective } from './soho-error.directive';
import * as i0 from "@angular/core";
export class SohoErrorModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorModule, declarations: [SohoErrorDirective // eslint-disable-line
        ], imports: [CommonModule], exports: [SohoErrorDirective // eslint-disable-line
        ] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoErrorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoErrorDirective // eslint-disable-line
                    ],
                    exports: [
                        SohoErrorDirective // eslint-disable-line
                    ]
                }]
        }] });
