// ids-enterprise-ng/esm2022/lib/utils/base-control-value-accessor.mjs
import { forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export const NOOP = () => { };
/**
 * Base class for new components that need to support NgForms.
 */
export class BaseControlValueAccessor {
    constructor() {
        /** ControlValueAccessor method called when the attached control has changed, */
        this._onChangeCallback = NOOP;
        /** ControlValueAccessor method called when the attached control has touched. */
        this._onTouchedCallback = NOOP;
    }
    /** Gets the value for the control. */
    get internalValue() {
        return this._value;
    }
    /** Sets the value for the control. */
    set internalValue(newValue) {
        if (newValue !== this._value) {
            this._value = newValue;
            this._onChangeCallback(newValue);
        }
    }
    /**
     * When touched.
     */
    touched() {
        this._onTouchedCallback();
    }
    /**
     * -------------------------------------------------
     * ControlValueAccessor implementation
     * -------------------------------------------------
     */
    /**
     * Write a new value to the element.
     */
    writeValue(value) {
        this._value = value;
    }
    /**
     * Set the function to be called when the control receives a change event.
     */
    registerOnChange(fn) {
        this._onChangeCallback = () => {
            fn(this.internalValue);
        };
    }
    /**
     * Set the function to be called when the control receives a touched event.
     */
    registerOnTouched(fn) {
        this._onTouchedCallback = fn;
    }
    /**
     * This function is called when the control status changes to or from "DISABLED".
     * Depending on the value, it will enable or disable the appropriate DOM element.
     */
    setDisabledState(_isDisabled) {
    }
}
/**
 * Provider - adds the contol value accessor.
 */
export function provideControlValueAccessor(type) {
    // creates an ngModel accessor to be used in components providers
    return {
        provide: NG_VALUE_ACCESSOR,
        useExisting: forwardRef(() => type),
        multi: true
    };
}
