// ids-enterprise-ng/esm2022/lib/modal-dialog/soho-modal-dialog.service.mjs
import { Injectable, Optional } from '@angular/core';
import { ArgumentHelper } from '../utils/argument.helper';
import { SohoModalDialogRef } from './soho-modal-dialog.ref';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
/**
 * This service is used to create a modal dialog, based on the content
 * of an Angular Componnent on the screen.
 */
export class SohoModalDialogService {
    /**
     * Constructor.
     *
     * @param appRef - application reference; must not be null.
     * @param componentFactoryResolver - used to create component factories for components dynamically.
     * @param injector - the current in scope injector, use as a delegate.
     * @param ngZone - the angular zone; must not be null.
     * @router
     */
    constructor(appRef, componentFactoryResolver, injector, ngZone, router) {
        this.appRef = appRef;
        this.componentFactoryResolver = componentFactoryResolver;
        this.injector = injector;
        this.ngZone = ngZone;
        this.router = router;
    }
    /**
     * Creates a modal dialog defined by the given component, returning a reference to the
     * dialog that can be interacted with.
     *
     * The component has the potential to include the dialog reference when it is
     * instantiated, as a provider.
     *
     * The dialog won't necessarily be open yet, see the dialog ref api for further methods.
     *
     * @param component - The type of the component to instantiate; must not be null.
     * @param parent - The parent container; obsolete but parameter kept for backwards API compatibility. Has no effect.
     * @param options - The control options to use.
     *
     * @return the modal dialog reference.
     */
    modal(component, _parent, options) {
        ArgumentHelper.checkNotNull('component', component);
        options = options || {};
        return new SohoModalDialogRef(this.router, this.appRef, this.componentFactoryResolver, this.injector, this.ngZone, options, component);
    }
    /**
     * Creates a modal dialog (using the content defined in the options), returning a reference to the
     * dialog that can be interacted with.
     *
     * The dialog won't be open.
     *
     * @return the modal dialog reference.
     */
    message(content) {
        const settings = { content };
        // Without a component, there is no destroy callback to ensure
        // the dialog's markup is removed.
        return new SohoModalDialogRef(this.router, this.appRef, this.componentFactoryResolver, this.injector, this.ngZone, settings);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModalDialogService, deps: [{ token: i0.ApplicationRef }, { token: i0.ComponentFactoryResolver }, { token: i0.Injector }, { token: i0.NgZone }, { token: i1.Router, optional: true }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModalDialogService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModalDialogService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.ApplicationRef }, { type: i0.ComponentFactoryResolver }, { type: i0.Injector }, { type: i0.NgZone }, { type: i1.Router, decorators: [{
                    type: Optional
                }] }]; } });
