// ids-enterprise-ng/esm2022/lib/arrange/soho-arrange.directive.mjs
import { Directive, Output, EventEmitter, Input, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoArrangeDirective {
    set arrangeOptions(arrangeOptions) {
        this._arrangeOptions = arrangeOptions;
    }
    constructor(elementRef, ngZone) {
        this.elementRef = elementRef;
        this.ngZone = ngZone;
        this.beforeArrange = new EventEmitter();
        this.arrangeUpdate = new EventEmitter();
        this.dragEnd = new EventEmitter();
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.elementRef.nativeElement);
            this.jQueryElement.arrange(this._arrangeOptions);
            this.arrange = this.jQueryElement.data('arrange');
            // Bind to jQuery events and emit as angular events
            this.jQueryElement.on('beforearrange', (event, status) => this.ngZone.run(() => this.beforeArrange.emit({ event, status })));
            this.jQueryElement.on('arrangeupdate', (event, status) => this.ngZone.run(() => this.arrangeUpdate.emit({ event, status })));
            this.jQueryElement.on('dragend', (event) => this.ngZone.run(() => this.dragEnd.emit({ event })));
        });
    }
    ngOnDestroy() {
        this.arrange?.destroy();
    }
    updated(settings) {
        if (settings) {
            this.arrange?.updated(settings);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoArrangeDirective, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoArrangeDirective, selector: "[soho-arrange]", inputs: { arrangeOptions: "arrangeOptions" }, outputs: { beforeArrange: "beforeArrange", arrangeUpdate: "arrangeUpdate", dragEnd: "dragEnd" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoArrangeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-arrange]',
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { beforeArrange: [{
                type: Output
            }], arrangeUpdate: [{
                type: Output
            }], dragEnd: [{
                type: Output
            }], arrangeOptions: [{
                type: Input
            }] } });
