// ids-enterprise-ng/esm2022/lib/module-nav/soho-module-nav.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoModuleNavComponent } from './soho-module-nav.component';
import * as i0 from "@angular/core";
// @TODO do we need to import Soho Accordion/Search/Dropdown/Popupmenu here?
// import { SohoAccordionHeaderComponent } from './soho-accordion-header.component';
// import { SohoAccordionPaneComponent } from './soho-accordion-pane.component';
export class SohoModuleNavModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavModule, declarations: [SohoModuleNavComponent], imports: [CommonModule], exports: [SohoModuleNavComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoModuleNavComponent
                    ],
                    exports: [
                        SohoModuleNavComponent
                    ]
                }]
        }] });
