// ids-enterprise-ng/esm2022/lib/field-options/soho-field-options.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoFieldOptionsDirective } from './soho-field-options.directive';
import * as i0 from "@angular/core";
export class SohoFieldOptionsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsModule, declarations: [SohoFieldOptionsDirective], imports: [CommonModule], exports: [SohoFieldOptionsDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoFieldOptionsDirective
                    ],
                    exports: [
                        SohoFieldOptionsDirective
                    ]
                }]
        }] });
