// ids-enterprise-ng/esm2022/lib/field-options/soho-field-options.directive.mjs
import { Directive, HostBinding, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoFieldOptionsDirective {
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        this.allFieldOptions = true;
        this._options = {};
    }
    ngAfterViewInit() {
        // -----------------------------------------------------------------------
        // Must run outside angular so that timeouts in the soho code won't cause
        // a ViewCheck in angular. Not running outside angular results in
        // unnecessary cycles causing the app to lock up (when lots of field-option
        // objects are used) until all the timeouts have resolved.
        // -----------------------------------------------------------------------
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.fieldoptions(this._options);
            this.fieldOptions = this.jQueryElement.data('field-options');
        });
    }
    /** Destructor. */
    ngOnDestroy() {
        if (this.fieldOptions) {
            this.ngZone.runOutsideAngular(() => {
                this.fieldOptions?.destroy();
                this.fieldOptions = null;
            });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsDirective, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoFieldOptionsDirective, selector: "[soho-field-options]", host: { properties: { "class.field-options": "this.allFieldOptions" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldOptionsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-field-options]' // eslint-disable-line
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { allFieldOptions: [{
                type: HostBinding,
                args: ['class.field-options']
            }] } });
