// ids-enterprise-ng/esm2022/lib/busyindicator/soho-busyindicator.module.mjs
/**
 * Created by ppatton on 8/30/2016.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoBusyIndicatorDirective } from './soho-busyindicator.directive';
import * as i0 from "@angular/core";
export class SohoBusyIndicatorModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBusyIndicatorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoBusyIndicatorModule, declarations: [SohoBusyIndicatorDirective], imports: [CommonModule], exports: [SohoBusyIndicatorDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBusyIndicatorModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBusyIndicatorModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoBusyIndicatorDirective
                    ],
                    exports: [
                        SohoBusyIndicatorDirective
                    ]
                }]
        }] });
