// ids-enterprise-ng/esm2022/lib/rating/soho-rating.component.mjs
import { Component, HostBinding, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoRatingComponent {
    get isRating() {
        return true;
    }
    constructor(element) {
        this.element = element;
        /** Options. */
        this.settings = {};
    }
    /** Setup */
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement);
        this.jQueryElement.rating(this.settings);
        this.rating = this.jQueryElement.data('rating');
    }
    /** Tear Down */
    ngOnDestroy() {
        if (this.rating) {
            this.rating.destroy();
            this.jQueryElement?.off();
            this.rating = null;
        }
    }
    /** Make control read only **/
    readonly() {
        if (this.rating) {
            this.rating.readonly();
        }
    }
    /** Enable the control **/
    enable() {
        if (this.rating) {
            this.rating.enable();
        }
    }
    /** Gets the current value of the control. If the control doesn't exist, will return -1 */
    get currentValue() {
        if (this.rating) {
            return this.rating.val();
        }
        return -1;
    }
    /** Sets the value for the control */
    set currentValue(newValue) {
        if (this.rating) {
            this.rating.val(newValue);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRatingComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoRatingComponent, selector: "[soho-rating]", host: { properties: { "class.rating": "this.isRating" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRatingComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-rating]',
                    template: '<ng-content></ng-content>'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isRating: [{
                type: HostBinding,
                args: ['class.rating']
            }] } });
