// ids-enterprise-ng/esm2022/lib/column/soho-column.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoColumnComponent {
    get isColumn() {
        return true;
    }
    /** Defines the data to use, must be specified for this component. */
    set dataset(dataset) {
        this.options.dataset = dataset;
        if (this.column) {
            this.column.settings.dataset = dataset;
            this.updateRequired = true;
        }
    }
    /** Chart Type */
    set type(value) {
        this.options.type = value;
        if (this.column) {
            this.column.settings.type = value;
            this.updateRequired = true;
        }
    }
    /** Default is a single or stacked chart. */
    set isStacked(value) {
        this.options.isStacked = value;
        if (this.column) {
            this.column.settings.isStacked = value;
            this.updateRequired = true;
        }
    }
    /** If false the legend will not be shown. */
    set showLegend(value) {
        this.options.showLegend = value;
        if (this.column) {
            this.column.settings.showLegend = value;
            this.updateRequired = true;
        }
    }
    /** true|false - will do or not do the animation, 'initial' will do only first time the animation. */
    set animate(value) {
        this.options.animate = value;
        if (this.column) {
            this.column.settings.animate = value;
            this.updateRequired = true;
        }
    }
    /** If true, the component will not resize when resizing the page. */
    set redrawOnResize(value) {
        this.options.redrawOnResize = value;
        if (this.column) {
            this.column.settings.redrawOnResize = value;
            this.updateRequired = true;
        }
    }
    /** The d3 axis format. */
    set format(value) {
        this.options.format = value;
        if (this.column) {
            this.column.settings.format = value;
            this.updateRequired = true;
        }
    }
    set formatterString(value) {
        this.options.formatterString = value;
        if (this.column) {
            this.column.settings.formatterString = value;
            this.updateRequired = true;
        }
    }
    /** Settings for the chart ticks. Can set ticks: {format: d3Format, number: n} */
    set ticks(value) {
        this.options.ticks = value;
        if (this.column) {
            this.column.settings.ticks = value;
            this.updateRequired = true;
        }
    }
    /** An empty message will be displayed when there is no chart data. */
    set emptyMessage(value) {
        this.options.emptyMessage = value;
        if (this.column) {
            this.column.settings.emptyMessage = value;
            this.updateRequired = true;
        }
    }
    /** Settings for the chart xAxis. */
    set xAxis(value) {
        this.options.xAxis = value;
        if (this.column) {
            this.column.settings.xAxis = value;
            this.updateRequired = true;
        }
    }
    /** Settings for the chart yAxis. */
    set yAxis(value) {
        this.options.yAxis = value;
        if (this.column) {
            this.column.settings.yAxis = value;
            this.updateRequired = true;
        }
    }
    /** Settings for the chart tooltip. */
    set tooltip(tooltip) {
        this.options.tooltip = tooltip;
        if (this.column) {
            this.column.settings.tooltip = tooltip;
            this.updateRequired = true;
        }
    }
    /**
     * Add extra attributes like id's to the component
     */
    set attributes(attributes) {
        this.options.attributes = attributes;
        if (this.column) {
            this.column.settings.attributes = attributes;
            this.updateRequired = true;
        }
    }
    /**
     * Add extra options to control axisLabels
     */
    set axisLabels(value) {
        this.options.axisLabels = value;
        if (this.column) {
            this.column.settings.axisLabels = value;
            this.updateRequired = true;
        }
    }
    /**
     * Adds the ability to use line chart if set to true. This will need a target value to the dataset.
     */
    set hideDots(value) {
        this.options.hideDots = value;
        if (this.column) {
            this.column.settings.hideDots = value;
            this.updateRequired = true;
        }
    }
    set useLine(value) {
        this.options.useLine = value;
        if (this.column) {
            this.column.settings.useLine = value;
            this.updateRequired = true;
        }
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        /** Options. */
        this.options = {};
        /** Events */
        this.selected = new EventEmitter();
        this.unselected = new EventEmitter();
        this.rendered = new EventEmitter();
        // eslint-disable-next-line @angular-eslint/no-output-native, @angular-eslint/no-output-rename
        this.dblclick = new EventEmitter();
        /**
         * @todo replace override of native attribute
         */
        // eslint-disable-next-line @angular-eslint/no-output-native, @angular-eslint/no-output-rename
        this.contextmenu = new EventEmitter();
        this.updateRequired = false;
    }
    /** Setup */
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.chart(this.options);
            this.column = this.jQueryElement.data('column');
            // Setup the events
            this.jQueryElement.on('selected', (_e, args) => this.ngZone.run(() => this.selected.emit(args)));
            this.jQueryElement.on('unselected', (_e, args) => this.ngZone.run(() => this.unselected.emit(args)));
            this.jQueryElement.on('rendered', (...args) => this.ngZone.run(() => this.rendered.emit(args)));
            this.jQueryElement.on('contextmenu', (...args) => this.ngZone.run(() => this.contextmenu?.emit(args)));
            this.jQueryElement.on('dblclick', (_e, args) => this.ngZone.run(() => this.dblclick.emit(args)));
        });
    }
    ngAfterViewChecked() {
        if (this.column && this.updateRequired) {
            this.ngZone.runOutsideAngular(() => this.column?.updated(this.column.settings));
            this.updateRequired = false;
        }
    }
    updated() {
        this.ngZone.runOutsideAngular(() => {
            this.options.type = 'column';
            this.jQueryElement?.chart(this.options);
            this.column = this.jQueryElement?.data('column');
        });
    }
    /** Tear Down */
    ngOnDestroy() {
        // call outside the angular zone so change detection isn't triggered by the soho component.
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.column) {
                this.column.destroy();
                this.column = null;
            }
        });
    }
    setSelected(selected) {
        this.ngZone.runOutsideAngular(() => this.column?.setSelected(selected));
    }
    toggleSelected(selected) {
        this.ngZone.runOutsideAngular(() => this.column?.toggleSelected(selected));
    }
    getSelected() {
        return this.ngZone.runOutsideAngular(() => this.column?.getSelected());
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoColumnComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoColumnComponent, selector: "[soho-column]", inputs: { dataset: "dataset", type: "type", isStacked: "isStacked", showLegend: "showLegend", animate: "animate", redrawOnResize: "redrawOnResize", format: "format", formatterString: "formatterString", ticks: "ticks", emptyMessage: "emptyMessage", xAxis: "xAxis", yAxis: "yAxis", tooltip: "tooltip", attributes: "attributes", axisLabels: "axisLabels", hideDots: "hideDots", useLine: "useLine" }, outputs: { selected: "selected", unselected: "unselected", rendered: "rendered", dblclick: "dblclick", contextmenu: "contextmenu" }, host: { properties: { "class.chart-container": "this.isColumn" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoColumnComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-column]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isColumn: [{
                type: HostBinding,
                args: ['class.chart-container']
            }], dataset: [{
                type: Input
            }], type: [{
                type: Input
            }], isStacked: [{
                type: Input
            }], showLegend: [{
                type: Input
            }], animate: [{
                type: Input
            }], redrawOnResize: [{
                type: Input
            }], format: [{
                type: Input
            }], formatterString: [{
                type: Input
            }], ticks: [{
                type: Input
            }], emptyMessage: [{
                type: Input
            }], xAxis: [{
                type: Input
            }], yAxis: [{
                type: Input
            }], tooltip: [{
                type: Input
            }], attributes: [{
                type: Input
            }], axisLabels: [{
                type: Input
            }], hideDots: [{
                type: Input
            }], useLine: [{
                type: Input
            }], selected: [{
                type: Output
            }], unselected: [{
                type: Output
            }], rendered: [{
                type: Output
            }], dblclick: [{
                type: Output
            }], contextmenu: [{
                type: Output
            }] } });
