// ids-enterprise-ng/esm2022/lib/tabs/soho-tabs.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoTabsComponent, SohoTabListContainerComponent, SohoTabListComponent, SohoTabComponent, SohoTabTitleComponent, SohoTabCountComponent, SohoTabSeparatorComponent, SohoTabPanelContainerComponent, SohoTabPanelComponent, } from './soho-tabs.component';
import * as i0 from "@angular/core";
export class SohoTabsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTabsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoTabsModule, declarations: [SohoTabsComponent,
            SohoTabListContainerComponent,
            SohoTabListComponent,
            SohoTabComponent,
            SohoTabTitleComponent,
            SohoTabCountComponent,
            SohoTabSeparatorComponent,
            SohoTabPanelContainerComponent,
            SohoTabPanelComponent], imports: [CommonModule], exports: [SohoTabsComponent,
            SohoTabListContainerComponent,
            SohoTabListComponent,
            SohoTabComponent,
            SohoTabTitleComponent,
            SohoTabCountComponent,
            SohoTabSeparatorComponent,
            SohoTabPanelContainerComponent,
            SohoTabPanelComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTabsModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTabsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoTabsComponent,
                        SohoTabListContainerComponent,
                        SohoTabListComponent,
                        SohoTabComponent,
                        SohoTabTitleComponent,
                        SohoTabCountComponent,
                        SohoTabSeparatorComponent,
                        SohoTabPanelContainerComponent,
                        SohoTabPanelComponent,
                    ],
                    exports: [
                        SohoTabsComponent,
                        SohoTabListContainerComponent,
                        SohoTabListComponent,
                        SohoTabComponent,
                        SohoTabTitleComponent,
                        SohoTabCountComponent,
                        SohoTabSeparatorComponent,
                        SohoTabPanelContainerComponent,
                        SohoTabPanelComponent,
                    ]
                }]
        }] });
