// ids-enterprise-ng/esm2022/lib/tag/soho-tag.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoTagComponent, SohoTagListComponent } from './soho-tag.component';
import * as i0 from "@angular/core";
export class SohoTagModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTagModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoTagModule, declarations: [SohoTagComponent,
            SohoTagListComponent], imports: [CommonModule], exports: [SohoTagComponent,
            SohoTagListComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTagModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTagModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoTagComponent,
                        SohoTagListComponent
                    ],
                    exports: [
                        SohoTagComponent,
                        SohoTagListComponent
                    ],
                }]
        }] });
