// ids-enterprise-ng/esm2022/lib/module-nav-container/soho-module-nav-container.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoModuleNavContainerComponent } from './soho-module-nav-container.component';
import * as i0 from "@angular/core";
export class SohoModuleNavContainerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavContainerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavContainerModule, declarations: [SohoModuleNavContainerComponent], imports: [CommonModule], exports: [SohoModuleNavContainerComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavContainerModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavContainerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoModuleNavContainerComponent,
                    ],
                    exports: [
                        SohoModuleNavContainerComponent,
                    ]
                }]
        }] });
