// ids-enterprise-ng/esm2022/lib/message/soho-message.service.mjs
import { Injectable } from '@angular/core';
import { SohoMessageRef } from './soho-message.ref';
import * as i0 from "@angular/core";
/**
 * This service is used to create message dialogs.
 */
export class SohoMessageService {
    /**
     * Constructor.
     */
    constructor() {
    }
    /**
     * Creates a dialog defined by the given message, returning a reference to the
     * associated dialog that can be interacted with.
     *
     * The dialog won't be open until open is called on the returned instance,
     * see the message ref api for further methods.
     *
     * @parent options - a compete set of options; can be null.
     *
     * @return the message reference.
     */
    message(options) {
        return new SohoMessageRef().options(options);
    }
    /**
     * Creates an error message, defined by the given options.
     *
     * The dialog won't be open until open is called on the returned instance,
     * see the message ref api for further methods.
     *
     * @return the message reference.
     */
    error(options) {
        return new SohoMessageRef().options(options).status('error');
    }
    /**
     * Creates an alert message, defined by the given options.
     *
     * The dialog won't be open until open is called on the returned instance,
     * see the message ref api for further methods.
     *
     * @return the message reference.
     */
    alert(options) {
        return new SohoMessageRef().options(options).status('alert');
    }
    /**
     * Creates an confirm (positive) message, defined by the given options.
     *
     * The dialog won't be open until open is called on the returned instance,
     * see the message ref api for further methods.
     *
     * @return the message reference.
     */
    confirm(options) {
        return new SohoMessageRef().options(options).status('confirm');
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMessageService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMessageService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMessageService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return []; } });
