// ids-enterprise-ng/esm2022/lib/trackdirty/soho-trackdirty.directive.mjs
import { Directive, EventEmitter, HostBinding, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoTrackDirtyDirective {
    get trackDirtyAttr() {
        return true;
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        /**
         * Called when element value is different from original value
         */
        this.dirty = new EventEmitter();
        /**
         * Called when element value is same as original value
         */
        this.pristine = new EventEmitter();
        /**
         * Called when the element has its original value reset to the current value
         */
        this.afterResetDirty = new EventEmitter();
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            // no options available for control
            this.jQueryElement.trackdirty();
            /**
             * Bind to jQueryElement's events
             */
            this.jQueryElement.on('dirty', (event) => this.ngZone.run(() => this.dirty.emit(event)));
            this.jQueryElement.on('pristine', (event) => this.ngZone.run(() => this.pristine.emit(event)));
            this.jQueryElement.on('afterresetdirty', (event) => this.ngZone.run(() => this.afterResetDirty.emit(event)));
            // returns a boolean, not an object
            this.trackDirty = this.jQueryElement.data('trackdirty');
        });
    }
    ngOnDestroy() {
        if (this.trackDirty) {
            this.ngZone.runOutsideAngular(() => this.trackDirty?.destroy());
            this.trackDirty = null;
        }
    }
    changeDirty() {
        if (this.trackDirty) {
            this.ngZone.runOutsideAngular(() => this.jQueryElement?.trigger('change.dirty'));
        }
    }
    resetDirty() {
        if (this.trackDirty) {
            this.ngZone.runOutsideAngular(() => this.jQueryElement?.trigger('resetdirty.dirty'));
        }
    }
    updated() {
        if (this.trackDirty) {
            this.ngZone.runOutsideAngular(() => this.trackDirty?.updated());
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyDirective, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoTrackDirtyDirective, selector: "[soho-trackdirty]", outputs: { dirty: "dirty", pristine: "pristine", afterResetDirty: "afterResetDirty" }, host: { properties: { "attr.data-trackdirty": "this.trackDirtyAttr" } }, exportAs: ["soho-trackdirty"], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyDirective, decorators: [{
            type: Directive,
            args: [{
                    // eslint-disable-next-line @angular-eslint/directive-selector
                    selector: '[soho-trackdirty]',
                    exportAs: 'soho-trackdirty'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { trackDirtyAttr: [{
                type: HostBinding,
                args: ['attr.data-trackdirty']
            }], dirty: [{
                type: Output
            }], pristine: [{
                type: Output
            }], afterResetDirty: [{
                type: Output
            }] } });
