// ids-enterprise-ng/esm2022/lib/trackdirty/soho-trackdirty.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoTrackDirtyDirective } from './soho-trackdirty.directive';
import * as i0 from "@angular/core";
export class SohoTrackDirtyModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyModule, declarations: [SohoTrackDirtyDirective], imports: [CommonModule], exports: [SohoTrackDirtyDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTrackDirtyModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoTrackDirtyDirective
                    ],
                    exports: [
                        SohoTrackDirtyDirective
                    ]
                }]
        }] });
