// ids-enterprise-ng/esm2022/lib/button/soho-button.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoButtonComponent } from './soho-button.component';
import { SohoIconModule } from '../icon/soho-icon.module';
import * as i0 from "@angular/core";
export class SohoButtonModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonModule, declarations: [SohoButtonComponent], imports: [CommonModule,
            SohoIconModule], exports: [SohoButtonComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonModule, imports: [CommonModule,
            SohoIconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoIconModule
                    ],
                    declarations: [
                        SohoButtonComponent
                    ],
                    exports: [
                        SohoButtonComponent
                    ]
                }]
        }] });
