// ids-enterprise-ng/esm2022/lib/stepchart/soho-stepchart.module.mjs
import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";
import { SohoStepChartComponent } from "./soho-stepchart.component";
import * as i0 from "@angular/core";
export class SohoStepChartModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepChartModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoStepChartModule, declarations: [SohoStepChartComponent], imports: [CommonModule], exports: [SohoStepChartComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepChartModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepChartModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoStepChartComponent
                    ],
                    exports: [
                        SohoStepChartComponent
                    ],
                }]
        }] });
