// ids-enterprise-ng/esm2022/lib/tree/soho-tree.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoTreeComponent } from './soho-tree.component';
import * as i0 from "@angular/core";
export class SohoTreeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTreeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoTreeModule, declarations: [SohoTreeComponent], imports: [CommonModule], exports: [SohoTreeComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTreeModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoTreeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoTreeComponent
                    ],
                    exports: [
                        SohoTreeComponent
                    ]
                }]
        }] });
