// ids-enterprise-ng/esm2022/lib/blockgrid/soho-blockgrid.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoBlockGridComponent } from './soho-blockgrid.component';
import * as i0 from "@angular/core";
export class SohoBlockGridModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBlockGridModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoBlockGridModule, declarations: [SohoBlockGridComponent], imports: [CommonModule], exports: [SohoBlockGridComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBlockGridModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoBlockGridModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoBlockGridComponent
                    ],
                    exports: [
                        SohoBlockGridComponent
                    ],
                }]
        }] });
