// ids-enterprise-ng/esm2022/lib/application-menu/soho-application-menu.module.mjs
/**
 * Created by ppatton on 8/30/2016.
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoApplicationMenuComponent } from './soho-application-menu.component';
import * as i0 from "@angular/core";
export class SohoApplicationMenuModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoApplicationMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoApplicationMenuModule, declarations: [SohoApplicationMenuComponent], imports: [CommonModule], exports: [SohoApplicationMenuComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoApplicationMenuModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoApplicationMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoApplicationMenuComponent
                    ],
                    exports: [
                        SohoApplicationMenuComponent
                    ]
                }]
        }] });
