// ids-enterprise-ng/esm2022/lib/radiobutton/soho-radiobutton.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * This component does not extend the ControlValueAccessor
 * as Angular already implements RadioControlValueAccessor
 * which hooks the controls up to the Forms model, see
 * https://angular.io/api/forms/RadioControlValueAccessor.
 *
 * The requirement is that the control matches the following
 * selector:
 *
 * input[type=radio][formControlName]
 * input[type=radio][formControl]
 * input[type=radio][ngModel]
 *
 * Make sure you add `type="radio"` to your markup.
 */
export class SohoRadioButtonComponent {
    /** Bind attributes to the host input element. */
    get isRadioType() {
        return 'radio';
    }
    /** Sets the class attribute for the radio button. */
    get isRadioButton() {
        return true;
    }
    /**
     * Due to conflicts with the ReactiveForms API, this Input
     * and HostBinding has been removed, in favour of using
     * [attr.disabled]="isDisabled".
     */
    set disabled(_value) {
        console.warn('The property `disabled` has been removed, please use [attr.disabled]="disable"');
    }
    /**
     * Constructor.
     */
    constructor(element) {
        this.element = element;
        /**
         * Called when the radiobutton value changes.
         *
         * @todo remove override of native element
         */
        // eslint-disable-next-line @angular-eslint/no-output-native
        this.change = new EventEmitter();
    }
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement);
        // There is no control initializer for radiobutton, since its css based
        if (this.value) {
            this.jQueryElement.val(this.value);
        }
        this.jQueryElement
            .on('change', (event) => this.onChange(event));
    }
    onChange(event) {
        const newValue = this.jQueryElement?.val();
        // Update the model
        this.value = newValue;
        // Update the data
        event.data = newValue;
        // Finally emit the changed value
        this.change.emit(event);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRadioButtonComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoRadioButtonComponent, selector: "input[soho-radiobutton]", inputs: { value: "value", disabled: "disabled", checked: "checked" }, outputs: { change: "change" }, host: { properties: { "attr.type": "this.isRadioType", "class.radio": "this.isRadioButton", "attr.checked": "this.checked" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRadioButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'input[soho-radiobutton]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { value: [{
                type: Input
            }], change: [{
                type: Output
            }], isRadioType: [{
                type: HostBinding,
                args: ['attr.type']
            }], isRadioButton: [{
                type: HostBinding,
                args: ['class.radio']
            }], disabled: [{
                type: Input
            }], checked: [{
                type: HostBinding,
                args: ['attr.checked']
            }, {
                type: Input
            }] } });
