// ids-enterprise-ng/esm2022/lib/popdown/soho-popdown-contents.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoPopDownContentsComponent {
    get isPopDownContents() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownContentsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoPopDownContentsComponent, selector: "soho-popdown-contents", host: { properties: { "class.popdown": "this.isPopDownContents" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownContentsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-popdown-contents',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isPopDownContents: [{
                type: HostBinding,
                args: ['class.popdown']
            }] } });
