// ids-enterprise-ng/esm2022/lib/popdown/soho-popdown.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoPopDownDirective {
    get isPopdown() {
        return true;
    }
    set keepOpen(keepOpen) {
        this.popdownOptions.keepOpen = keepOpen;
    }
    /*
    * If defined, provides a way to place the popdown against an alternate element.
    */
    set trigger(trigger) {
        this.popdownOptions.trigger = trigger;
    }
    /*
    * If true, when the popdown is opened, the first available input/button in its content area will be focused.
    */
    set autoFocus(autoFocus) {
        this.popdownOptions.autoFocus = autoFocus;
    }
    /*
    * If true, popdown will be toggle soon focused on the popdown trigger.
    */
    set toggleOnFocus(toggleOnFocus) {
        this.popdownOptions.toggleOnFocus = toggleOnFocus;
    }
    /**
     * Hook to work with tabbing in and out of the popdown.
     */
    set firstLastTab(firstLastTab) {
        this.popdownOptions.firstLastTab = firstLastTab;
    }
    constructor(element) {
        this.element = element;
        this.popdownOptions = {};
    }
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement);
        this.jQueryElement.popdown(this.popdownOptions);
        this.popdown = this.jQueryElement.data('popdown');
    }
    ngOnDestroy() {
        if (this.popdown) {
            this.popdown.destroy();
        }
    }
    open() {
        this.popdown?.open();
    }
    close() {
        this.popdown?.close();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoPopDownDirective, selector: "[soho-popdown]", inputs: { keepOpen: "keepOpen", trigger: "trigger", autoFocus: "autoFocus", toggleOnFocus: "toggleOnFocus", firstLastTab: "firstLastTab" }, host: { properties: { "attr.popdown": "this.isPopdown" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-popdown]', // eslint-disable-line
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isPopdown: [{
                type: HostBinding,
                args: ['attr.popdown']
            }], keepOpen: [{
                type: Input
            }], trigger: [{
                type: Input
            }], autoFocus: [{
                type: Input
            }], toggleOnFocus: [{
                type: Input
            }], firstLastTab: [{
                type: Input
            }] } });
