// ids-enterprise-ng/esm2022/lib/popdown/soho-popdown.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoPopDownContentsComponent } from './soho-popdown-contents.component';
import { SohoPopDownDirective } from './soho-popdown.directive';
import * as i0 from "@angular/core";
export class SohoPopDownModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownModule, declarations: [SohoPopDownContentsComponent,
            SohoPopDownDirective], imports: [CommonModule], exports: [SohoPopDownContentsComponent,
            SohoPopDownDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopDownModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoPopDownContentsComponent,
                        SohoPopDownDirective
                    ],
                    exports: [
                        SohoPopDownContentsComponent,
                        SohoPopDownDirective
                    ]
                }]
        }] });
