// ids-enterprise-ng/esm2022/lib/expandablearea/soho-expandablearea.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ExpandableAreaComponent, ExpandableFooterComponent, ExpandableHeaderComponent, ExpandablePaneComponent } from './soho-expandablearea.component';
import * as i0 from "@angular/core";
export class SohoExpandableAreaModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoExpandableAreaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoExpandableAreaModule, declarations: [ExpandableAreaComponent,
            ExpandableFooterComponent,
            ExpandableHeaderComponent,
            ExpandablePaneComponent], imports: [CommonModule], exports: [ExpandableAreaComponent,
            ExpandableFooterComponent,
            ExpandableHeaderComponent,
            ExpandablePaneComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoExpandableAreaModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoExpandableAreaModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        ExpandableAreaComponent,
                        ExpandableFooterComponent,
                        ExpandableHeaderComponent,
                        ExpandablePaneComponent,
                    ],
                    exports: [
                        ExpandableAreaComponent,
                        ExpandableFooterComponent,
                        ExpandableHeaderComponent,
                        ExpandablePaneComponent,
                    ]
                }]
        }] });
