// ids-enterprise-ng/esm2022/lib/wizard/soho-wizard-buttonbar.component.mjs
import { Component, ChangeDetectionStrategy, HostBinding, Input, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./soho-wizard.component";
import * as i2 from "@angular/common";
import * as i3 from "../button/soho-button.component";
/**
 * Angular wrapper for the soho wizard buttonbar.
 *
 * Looks for a `div` annotated with `soho-wizard-buttonbar`, this
 * simply adds a component used to handle the button selection.
 */
export class SohoWizardButtonbarComponent {
    constructor(wizard, changeDetectorRef) {
        this.wizard = wizard;
        this.changeDetectorRef = changeDetectorRef;
        // @todo need types for these button, these seem to be different from SohoButtonOptions
        this.buttons = [
            {
                id: 'previous',
                text: Soho.Locale.translate('Previous'),
                click: () => this.wizard.previous(),
                disabled: () => !this.wizard.hasPrevious(),
                position: 'middle'
            },
            {
                id: 'next',
                text: Soho.Locale.translate('Next'),
                click: () => {
                    this.wizard.next();
                },
                disabled: () => !this.wizard.hasNext(),
                isDefault: true,
                position: 'middle'
            },
            {
                id: 'finish',
                text: 'Finish',
                click: () => {
                    this.wizard.finish();
                },
                disabled: () => this.wizard.hasFinished(),
                position: 'right'
            }
        ];
        this.isButtonBar = false;
        // Ensures all the button are redrawn, keeping track of their enabled state.
        this.wizard.afteractivated.subscribe(() => {
            this.changeDetectorRef.markForCheck();
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardButtonbarComponent, deps: [{ token: i1.SohoWizardComponent }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWizardButtonbarComponent, selector: "div[soho-wizard-buttonbar]", inputs: { buttons: "buttons" }, host: { properties: { "class.buttonset": "this.isButtonBar" } }, ngImport: i0, template: `
    <hr class="fieldset-hr">
    <div class="buttonbar">
      <div class="left">
        <ng-container *ngFor="let button of buttons" >
          <button *ngIf="button.position==='left'"
            [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
            (click)='button?.click()'>{{button?.text}}</button>
        </ng-container>
      </div>
      <div class="middle">
      <ng-container *ngFor="let button of buttons" >
        <button *ngIf="button.position==='middle'"
          [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
          (click)='button?.click()'>{{button?.text}}</button>
      </ng-container>
      </div>
      <div class="right">
        <ng-container *ngFor="let button of buttons" >
          <button *ngIf="button.position==='right'"
            [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
            (click)='button?.click()'>{{button?.text}}</button>
        </ng-container>
        <ng-content></ng-content>
      </div>
    </div>
  `, isInline: true, styles: [":host{flex:0}\n", ".buttonbar{display:flex}\n", ".left{flex:1;display:flex;justify-content:flex-start;align-items:center}\n", ".middle{flex:1;display:flex;justify-content:center;align-items:center}\n", ".right{flex:1;display:flex;justify-content:flex-end;align-items:center}\n"], dependencies: [{ kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i3.SohoButtonComponent, selector: "button[soho-button]", inputs: ["soho-button", "buttonOptions", "toggleOnIcon", "toggleOffIcon", "replaceText", "hideMenuArrow", "hitbox", "notificationBadge", "notificationBadgeOptions", "extraIconClass", "isToggle", "isTogglePressed", "icon", "iconAlign", "isSubmit", "ripple", "state", "toggle", "expandableExpander", "moveToLeft", "moveToRight", "moveToSelected"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardButtonbarComponent, decorators: [{
            type: Component,
            args: [{ selector: 'div[soho-wizard-buttonbar]', template: `
    <hr class="fieldset-hr">
    <div class="buttonbar">
      <div class="left">
        <ng-container *ngFor="let button of buttons" >
          <button *ngIf="button.position==='left'"
            [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
            (click)='button?.click()'>{{button?.text}}</button>
        </ng-container>
      </div>
      <div class="middle">
      <ng-container *ngFor="let button of buttons" >
        <button *ngIf="button.position==='middle'"
          [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
          (click)='button?.click()'>{{button?.text}}</button>
      </ng-container>
      </div>
      <div class="right">
        <ng-container *ngFor="let button of buttons" >
          <button *ngIf="button.position==='right'"
            [soho-button]="button?.type" [icon]="button?.icon" [id]='button?.id' [disabled]="button?.disabled()"
            (click)='button?.click()'>{{button?.text}}</button>
        </ng-container>
        <ng-content></ng-content>
      </div>
    </div>
  `, changeDetection: ChangeDetectionStrategy.OnPush, styles: [":host{flex:0}\n", ".buttonbar{display:flex}\n", ".left{flex:1;display:flex;justify-content:flex-start;align-items:center}\n", ".middle{flex:1;display:flex;justify-content:center;align-items:center}\n", ".right{flex:1;display:flex;justify-content:flex-end;align-items:center}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.SohoWizardComponent }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { buttons: [{
                type: Input
            }], isButtonBar: [{
                type: HostBinding,
                args: ['class.buttonset']
            }] } });
