// ids-enterprise-ng/esm2022/lib/wizard/soho-wizard.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoButtonModule } from '../button/soho-button.module';
import { SohoWizardComponent } from './soho-wizard.component';
import { SohoWizardTickComponent } from './soho-wizard-tick.component';
import { SohoWizardHeaderComponent } from './soho-wizard-header.component';
import { SohoWizardPageComponent } from './soho-wizard-page.component';
import { SohoWizardPagesComponent } from './soho-wizard-pages.component';
import { SohoWizardButtonbarComponent } from './soho-wizard-buttonbar.component';
import * as i0 from "@angular/core";
export class SohoWizardModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardModule, declarations: [SohoWizardComponent,
            SohoWizardTickComponent,
            SohoWizardHeaderComponent,
            SohoWizardPagesComponent,
            SohoWizardPageComponent,
            SohoWizardButtonbarComponent], imports: [CommonModule,
            SohoButtonModule], exports: [SohoWizardComponent,
            SohoWizardTickComponent,
            SohoWizardHeaderComponent,
            SohoWizardPagesComponent,
            SohoWizardPageComponent,
            SohoWizardButtonbarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardModule, imports: [CommonModule,
            SohoButtonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoButtonModule
                    ],
                    declarations: [
                        SohoWizardComponent,
                        SohoWizardTickComponent,
                        SohoWizardHeaderComponent,
                        SohoWizardPagesComponent,
                        SohoWizardPageComponent,
                        SohoWizardButtonbarComponent
                    ],
                    exports: [
                        SohoWizardComponent,
                        SohoWizardTickComponent,
                        SohoWizardHeaderComponent,
                        SohoWizardPagesComponent,
                        SohoWizardPageComponent,
                        SohoWizardButtonbarComponent
                    ]
                }]
        }] });
