// ids-enterprise-ng/esm2022/lib/wizard/soho-wizard-tick.component.mjs
import { Component, ChangeDetectionStrategy, HostBinding, Input, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Angular wrapper for the soho wizard tick.
 *
 * This wraps anchor tags annotated with the `soho-wizard-tick`.
 */
export class SohoWizardTickComponent {
    /**
     * For the `href` use the id.
     */
    get hrefAttr() {
        return this.tickId ? '#' + this.tickId : null;
    }
    /**
     * Disable the click handler.
     */
    onClick(event) {
        event.preventDefault();
        event.stopPropagation();
    }
    isCurrentTick() {
        // A step is selected if the element has the current class.
        return this.jQueryElement?.hasClass('current') || false;
    }
    /**
     * Constructor.
     *
     * @param elementRef - the element matching this component.
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        /**
         * Ensure any classes are set correctly.
         */
        this.isWizardTick = true;
        /**
         *
         */
        this.current = false;
        /**
         * Disabled the tick.
         */
        this.disabled = false;
    }
    ngAfterViewInit() {
        // Wrap the "unordered list" element in a jQuery selector.
        this.jQueryElement = jQuery(this.elementRef.nativeElement);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardTickComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWizardTickComponent, selector: "a[soho-wizard-tick]", inputs: { shortLabel: "shortLabel", tickId: "tickId", current: "current", disabled: "disabled" }, host: { listeners: { "click": "onClick($event)" }, properties: { "class.tick": "this.isWizardTick", "attr.href": "this.hrefAttr", "attr.tickid": "this.tickId", "class.current": "this.current", "class.is-disabled": "this.disabled" } }, ngImport: i0, template: '<span class="label" attr.data-shortLabel="{{shortLabel}}"><ng-content></ng-content></span>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardTickComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'a[soho-wizard-tick]',
                    template: '<span class="label" attr.data-shortLabel="{{shortLabel}}"><ng-content></ng-content></span>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isWizardTick: [{
                type: HostBinding,
                args: ['class.tick']
            }], hrefAttr: [{
                type: HostBinding,
                args: ['attr.href']
            }], shortLabel: [{
                type: Input
            }], tickId: [{
                type: HostBinding,
                args: ['attr.tickid']
            }, {
                type: Input
            }], current: [{
                type: HostBinding,
                args: ['class.current']
            }, {
                type: Input
            }], disabled: [{
                type: HostBinding,
                args: ['class.is-disabled']
            }, {
                type: Input
            }], onClick: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
