// ids-enterprise-ng/esm2022/lib/wizard/soho-wizard.component.mjs
import { ChangeDetectionStrategy, Component, Output, EventEmitter, HostBinding, Input, ContentChild } from '@angular/core';
import { SohoWizardHeaderComponent } from './soho-wizard-header.component';
import { SohoWizardPagesComponent } from './soho-wizard-pages.component';
import * as i0 from "@angular/core";
/**
 * Angular Wrapper for the Soho Wizard Component.
 *
 * This component searches for a div with the attribute
 * 'soho-wizard' in the DOM, initialising those found with
 * the SoHo Wizard control.
 *
 * TODO:
 * =====
 *
 * - handling of ticks / tick model (based on underlying widget)
 * - model driven
 * - support Builder Panel style (with title?)
 * - support for "modal style" buttons.
 * - extract state machine
 */
export class SohoWizardComponent {
    // -------------------------------------------
    // Inputs
    // -------------------------------------------
    /**
     * Ticks for the settings - this does not really work yet (i think).
     */
    set ticks(ticks) {
        this._options.ticks = ticks;
        if (this.wizard) {
            this.wizard.settings.ticks = this._options.ticks;
            this.wizard.updated();
        }
    }
    get ticks() {
        return this._options.ticks;
    }
    /** Id of the current tick. */
    set currentTickId(tickId) {
        this.pagesContainer?.pages.forEach(p => p.hidden = (p.tickId !== tickId));
        const step = this.header?.steps?.find(s => s.tickId === tickId);
        if (this.wizard && step?.jQueryElement) {
            this.wizard.activate(null, step.jQueryElement);
        }
    }
    get currentTickId() {
        const step = this.currentStep();
        const tickId = step?.tickId;
        return tickId || '';
    }
    /**
     * Constructor.
     *
     * @param elementRef - the element matching this component.
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        /**
         * This event is fired when a tick is activated.
         */
        this.activated = new EventEmitter();
        /**
         * This event is fired after a tick is activated.
         */
        this.afteractivated = new EventEmitter();
        // -------------------------------------------
        // Hostbinding
        // -------------------------------------------
        this.isWizardClass = true;
        /** An internal options object that gets updated by using the component's Inputs(). */
        this._options = {};
        /**
         * Has the wizard finished.
         */
        this.finished = false;
    }
    // -------------------------------------------
    // Public API
    // -------------------------------------------
    /**
     * Returns the currently selected page.
     */
    get currentPage() {
        if (this.pagesContainer && this.pagesContainer?.pages) {
            return this.pagesContainer?.pages.find(p => p.tickId === this.currentTickId);
        }
        return undefined;
    }
    /**
     * Moves to the first state if possible.
     */
    first() {
        if (this.wizard && !this.finished && this.stepCount() > 0) {
            const step = this.stepAt(0);
            if (step && step.jQueryElement) {
                this.wizard.activate(null, step.jQueryElement);
            }
        }
    }
    /**
     * Attempts to move to the next step, if allowed.
     *
     *
     */
    next() {
        // This is a bit grim ... but we need to rely on ticks for the state.
        let currentIndex = this.currentIndex();
        // @todo handle disabled states.
        if (!this.finished && ++currentIndex < this.stepCount()) {
            const step = this.stepAt(currentIndex);
            if (step && step.jQueryElement) {
                this.wizard?.activate(null, step.jQueryElement);
            }
        }
    }
    /**
     * Attempts to move to the previous step, if allowed.
     *
     *
     */
    previous() {
        let currentIndex = this.currentIndex();
        if (!this.finished && --currentIndex >= 0) {
            const step = this.stepAt(currentIndex);
            if (step && step.jQueryElement) {
                this.wizard?.activate(null, step.jQueryElement);
            }
        }
    }
    /**
     * Attempts to move to the last step, if allowed.
     *
     *
     */
    last() {
        const step = this.stepAt(this.stepCount() - 1);
        if (step && step.jQueryElement) {
            this.wizard?.activate(null, step.jQueryElement);
        }
    }
    /**
     * Attempts to move to the last step, and finish the wizard.
     *
     *
     */
    finish() {
        // Mark the wizard as finished.
        this.finished = true;
        // Disabled all ticks.
        this.header?.steps?.forEach(p => p.disabled = this.finished);
        // Move to the last tick.
        this.last();
    }
    /**
     * Is there another step after the current step?
     *
     * @return  true if there is another step; otherwise false.
     *
     */
    hasNext() {
        return !this.finished && this.currentIndex() < this.stepCount() - 1;
    }
    /**
     * Is there previous step.
     */
    hasPrevious() {
        return !this.finished && this.currentIndex() > 0;
    }
    /**
     * Returns true if the process has finished,
     * otherwise false.
     */
    hasFinished() {
        return this.finished;
    }
    /**
     * Resets the state machine, moving to the first page
     * and clearing finished flag.
     */
    reset() {
        this.finished = false;
        this.first();
    }
    // ------------------------------------------
    // Lifecycle Events
    // ------------------------------------------
    ngAfterViewInit() {
        // Wrap the "unordered list" element in a jQuery selector.
        this.jQueryElement = jQuery(this.elementRef.nativeElement);
        // Initialise the Soho control.
        this.jQueryElement.wizard(this._options);
        // Once the control is initialised, extract the control
        // plug-in from the element.  The element name is
        // defined by the plug-in, but in this case it is 'wizard'.
        this.wizard = this.jQueryElement.data('wizard');
        // Initialize any event handlers.
        this.jQueryElement
            .on('beforeactivate', (_e, tick) => this.onBeforeActivate(tick))
            .on('activated', (_e, tick) => this.onActivated(tick))
            .on('afteractivated', (_e, tick) => this.afteractivated.next({ tick }));
        // Reset the cached steps if the list of steps changes.
        if (this.header) {
            this.header.steps?.changes.subscribe(() => {
                this._steps = [];
            });
        }
    }
    ngAfterContentInit() {
        // Added delay otherwise the component is not complete
        // causing the active page to not be displayed.
        setTimeout(() => {
            if (this.header) {
                const step = this.header.steps?.find(s => s.isCurrentTick());
                this.pagesContainer?.pages.forEach(p => p.hidden = (!step || step.tickId !== p.tickId));
            }
        });
    }
    /**
     * Handle component destruction by clearing down the SoHo
     * wizard component.
     *
     *
     */
    ngOnDestroy() {
        if (this.wizard) {
            this.wizard.destroy();
            this.wizard = undefined;
        }
    }
    onActivated(tick) {
        // When activated - make sure the rest of the component is
        // updated ...
        if (tick) {
            // ... find the id of the tick activated ...
            const tickId = tick.attr('tickId');
            // ... if we have one (to avoid errors) ...
            if (tickId) {
                // hide all the inactive pages and show the active page.
                this.pagesContainer?.pages.forEach((p) => {
                    if (tickId === p.tickId) {
                        // fire an event on the page.
                        p.fireActivated({ tick });
                        // ... show it.
                        p.hidden = false;
                    }
                    else {
                        // hide it.
                        p.hidden = true;
                    }
                });
                // ... publish.
                this.activated.next({ tick });
            }
        }
    }
    onBeforeActivate(tick) {
        // Check for vetoing.
        return this.beforeActivate != null ? this.beforeActivate.call(tick) : true;
    }
    stepCount() {
        return this.header?.steps?.length || 0;
    }
    steps() {
        if (!this._steps && this.header) {
            this._steps = this.header?.steps?.toArray();
        }
        return this._steps;
    }
    stepAt(index) {
        const steps = this.steps();
        return steps ? steps[index] : undefined;
    }
    currentIndex() {
        const steps = this.steps();
        const currentStep = this.currentStep();
        return steps && currentStep ? steps.indexOf(currentStep) : -1;
    }
    currentStep() {
        const steps = this.steps();
        return steps ? steps.find(s => s.isCurrentTick()) : undefined;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWizardComponent, selector: "div[soho-wizard]", inputs: { ticks: "ticks", currentTickId: "currentTickId", beforeActivate: "beforeActivate" }, outputs: { activated: "activated", afteractivated: "afteractivated" }, host: { properties: { "class.wizard": "this.isWizardClass" } }, queries: [{ propertyName: "pagesContainer", first: true, predicate: SohoWizardPagesComponent, descendants: true, static: true }, { propertyName: "header", first: true, predicate: SohoWizardHeaderComponent, descendants: true, static: true }], ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, styles: [":host{display:flex;flex:1;flex-direction:column}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardComponent, decorators: [{
            type: Component,
            args: [{ selector: 'div[soho-wizard]', template: `<ng-content></ng-content>`, changeDetection: ChangeDetectionStrategy.OnPush, styles: [":host{display:flex;flex:1;flex-direction:column}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { pagesContainer: [{
                type: ContentChild,
                args: [SohoWizardPagesComponent, { static: true }]
            }], header: [{
                type: ContentChild,
                args: [SohoWizardHeaderComponent, { static: true }]
            }], ticks: [{
                type: Input
            }], currentTickId: [{
                type: Input
            }], beforeActivate: [{
                type: Input
            }], activated: [{
                type: Output
            }], afteractivated: [{
                type: Output
            }], isWizardClass: [{
                type: HostBinding,
                args: ['class.wizard']
            }] } });
