// ids-enterprise-ng/esm2022/lib/wizard/soho-wizard-page.component.mjs
import { HostBinding, Input, Component, ChangeDetectionStrategy, Output, EventEmitter } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Angular wrapper for the soho wizard page.
 */
export class SohoWizardPageComponent {
    /**
     * Constructor.
     *
     * @param el owning DOM element
     *
     */
    constructor(el) {
        this.el = el;
        this.isWizardPage = true;
        /**
         * Is this wizard page hidden? Return true by default to avoid
         * displaying all the pages for a moment.  The wizard will
         * ensure only a single page is displayed.
         *
         *
         */
        this.hidden = true; // tslint: ignore-line
        /**
         * Event fired when the page is activated.
         */
        this.activated = new EventEmitter();
    }
    ngAfterViewInit() {
        this.jQueryElement = $(this.el.nativeElement);
    }
    /**
     * Fire the activated event, using the given SohoWizardEvent.
     *
     * I'd have rather done this with an event, but we end up
     * with a circular dependency.
     *
     * @param e - the soho wizard event.
     *
     */
    fireActivated(e) {
        this.activated.next(e);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardPageComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWizardPageComponent, selector: "div[soho-wizard-page]", inputs: { tickId: "tickId" }, outputs: { activated: "activated" }, host: { properties: { "class.wizard-page": "this.isWizardPage", "class.hidden": "this.hidden" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, styles: [":host{flex:1;display:flex;flex-direction:column;overflow-y:auto}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWizardPageComponent, decorators: [{
            type: Component,
            args: [{ selector: 'div[soho-wizard-page]', template: `<ng-content></ng-content>`, changeDetection: ChangeDetectionStrategy.OnPush, styles: [":host{flex:1;display:flex;flex-direction:column;overflow-y:auto}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isWizardPage: [{
                type: HostBinding,
                args: ['class.wizard-page']
            }], hidden: [{
                type: HostBinding,
                args: ['class.hidden']
            }], tickId: [{
                type: Input
            }], activated: [{
                type: Output
            }] } });
