// ids-enterprise-ng/esm2022/lib/renderLoop/soho-render-loop.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoRenderLoopService {
    constructor(ngZone) {
        this.ngZone = ngZone;
        this.renderLoopCount = 0;
    }
    /**
     * Start the entire render loop
     */
    start() {
        this.ngZone.runOutsideAngular(() => {
            Soho.renderLoop.register(() => {
                this.renderLoopCount++;
            }, undefined, 'angular-timer-count');
            Soho.renderLoop.start();
        });
    }
    /**
     * Stops the entire render loop
     */
    stop() {
        Soho.renderLoop.stop();
    }
    getCurrentCount() {
        return this.renderLoopCount;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRenderLoopService, deps: [{ token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRenderLoopService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoRenderLoopService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.NgZone }]; } });
