// ids-enterprise-ng/esm2022/lib/accordion/soho-accordion-header.component.mjs
import { Component, HostBinding, ChangeDetectionStrategy, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoAccordionHeaderComponent {
    /**
     * Constructor.
     *
     * @param elementRef - the element matching the component's selector.
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.isDisabled = false;
        this.isBlockDisplay = 'block';
        this.isAccordionHeader = true;
        this.isExpanded = false;
        this.moduleNavItem = false;
    }
    // ------------------------------------------
    // Lifecycle Events
    // ------------------------------------------
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.elementRef.nativeElement);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionHeaderComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoAccordionHeaderComponent, selector: "soho-accordion-header", inputs: { isExpanded: "isExpanded", moduleNavItem: "moduleNavItem" }, host: { properties: { "class.is-disabled": "this.isDisabled", "style.display": "this.isBlockDisplay", "class.accordion-header": "this.isAccordionHeader", "class.is-expanded": "this.isExpanded", "class.module-nav-item": "this.moduleNavItem" } }, ngImport: i0, template: "<a href=\"#\" aria-haspopup=\"true\" role=\"button\" [attr.aria-expanded]=\"isExpanded\">\n  <span>\n    <ng-content></ng-content>\n  </span>\n</a>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionHeaderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'soho-accordion-header', changeDetection: ChangeDetectionStrategy.OnPush, template: "<a href=\"#\" aria-haspopup=\"true\" role=\"button\" [attr.aria-expanded]=\"isExpanded\">\n  <span>\n    <ng-content></ng-content>\n  </span>\n</a>\n" }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isDisabled: [{
                type: HostBinding,
                args: ['class.is-disabled']
            }], isBlockDisplay: [{
                type: HostBinding,
                args: ['style.display']
            }], isAccordionHeader: [{
                type: HostBinding,
                args: ['class.accordion-header']
            }], isExpanded: [{
                type: HostBinding,
                args: ['class.is-expanded']
            }, {
                type: Input
            }], moduleNavItem: [{
                type: HostBinding,
                args: ['class.module-nav-item']
            }, {
                type: Input
            }] } });
