// ids-enterprise-ng/esm2022/lib/accordion/soho-accordion.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoAccordionComponent } from './soho-accordion.component';
import { SohoAccordionHeaderComponent } from './soho-accordion-header.component';
import { SohoAccordionPaneComponent } from './soho-accordion-pane.component';
import * as i0 from "@angular/core";
export class SohoAccordionModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionModule, declarations: [SohoAccordionComponent,
            SohoAccordionHeaderComponent,
            SohoAccordionPaneComponent], imports: [CommonModule], exports: [SohoAccordionComponent,
            SohoAccordionHeaderComponent,
            SohoAccordionPaneComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoAccordionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoAccordionComponent,
                        SohoAccordionHeaderComponent,
                        SohoAccordionPaneComponent,
                    ],
                    exports: [
                        SohoAccordionComponent,
                        SohoAccordionHeaderComponent,
                        SohoAccordionPaneComponent,
                    ]
                }]
        }] });
