// ids-enterprise-ng/esm2022/lib/hierarchy/soho-hierarchy.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoHierarchyComponent, SohoHierarchyLeafTemplateComponent } from './soho-hierarchy.component';
import * as i0 from "@angular/core";
export class SohoHierarchyModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHierarchyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoHierarchyModule, declarations: [SohoHierarchyComponent,
            SohoHierarchyLeafTemplateComponent], imports: [CommonModule], exports: [SohoHierarchyComponent,
            SohoHierarchyLeafTemplateComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHierarchyModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHierarchyModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoHierarchyComponent,
                        SohoHierarchyLeafTemplateComponent
                    ],
                    exports: [
                        SohoHierarchyComponent,
                        SohoHierarchyLeafTemplateComponent
                    ]
                }]
        }] });
