// ids-enterprise-ng/esm2022/lib/listview/soho-listview.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoListViewComponent, SohoListViewHeaderComponent, SohoListViewItemComponent, SohoListViewMicroComponent, SohoListViewSearchComponent, SohoListViewSubHeaderComponent, } from './soho-listview.component';
import { SohoSearchFieldModule } from '../searchfield/soho-searchfield.module';
import * as i0 from "@angular/core";
export class SohoListViewModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListViewModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoListViewModule, declarations: [SohoListViewComponent,
            SohoListViewHeaderComponent,
            SohoListViewItemComponent,
            SohoListViewMicroComponent,
            SohoListViewSearchComponent,
            SohoListViewSubHeaderComponent], imports: [CommonModule,
            SohoSearchFieldModule], exports: [SohoListViewComponent,
            SohoListViewHeaderComponent,
            SohoListViewItemComponent,
            SohoListViewMicroComponent,
            SohoListViewSearchComponent,
            SohoListViewSubHeaderComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListViewModule, imports: [CommonModule,
            SohoSearchFieldModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListViewModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SohoListViewComponent,
                        SohoListViewHeaderComponent,
                        SohoListViewItemComponent,
                        SohoListViewMicroComponent,
                        SohoListViewSearchComponent,
                        SohoListViewSubHeaderComponent,
                    ],
                    exports: [
                        SohoListViewComponent,
                        SohoListViewHeaderComponent,
                        SohoListViewItemComponent,
                        SohoListViewMicroComponent,
                        SohoListViewSearchComponent,
                        SohoListViewSubHeaderComponent,
                    ],
                    imports: [
                        CommonModule,
                        SohoSearchFieldModule
                    ], // TODO: Add toolbar module
                }]
        }] });
