// ids-enterprise-ng/esm2022/lib/pager/soho-pager.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoIconModule } from '../icon/soho-icon.module';
import { SohoStandalonePagerComponent } from './soho-standalone-pager.component';
import * as i0 from "@angular/core";
export class SohoPagerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPagerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoPagerModule, declarations: [SohoStandalonePagerComponent], imports: [CommonModule,
            SohoIconModule], exports: [SohoStandalonePagerComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPagerModule, imports: [CommonModule,
            SohoIconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPagerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoIconModule
                    ],
                    declarations: [
                        SohoStandalonePagerComponent
                    ],
                    exports: [
                        SohoStandalonePagerComponent
                    ]
                }]
        }] });
