// ids-enterprise-ng/esm2022/lib/circlepager/soho-circlepager.component.mjs
import { ChangeDetectionStrategy, Component, HostBinding, Input, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoCirclepagerComponent {
    get isCirclePager() {
        return true;
    }
    /** The number of slides to show in one view / pane. */
    set slidesToShow(value) {
        this.settings.slidesToShow = value;
        if (this.circlepager) {
            this.circlepager.settings.slidesToShow = value;
            // todo: does this setting require an updated call to circlepager.js?
        }
    }
    /** First showing slide/group, an 0-based integer */
    set startingSlide(value) {
        this.settings.startingSlide = value;
        if (this.circlepager) {
            this.circlepager.settings.startingSlide = value;
            this.updateRequired = true;
        }
    }
    /** Setting loop: true will loop back after next/previous reached to end */
    set loop(value) {
        this.settings.loop = value;
        if (this.circlepager) {
            this.circlepager.settings.loop = value;
            this.updateRequired = true;
        }
    }
    /** Add extra attributes like id's to the component **/
    set attributes(attributes) {
        this.settings.attributes = attributes;
        if (this.circlepager) {
            this.circlepager.settings.attributes = attributes;
            this.updateRequired = true;
        }
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        /** Options. */
        this.settings = {};
        this.updateRequired = false;
    }
    /** Setup */
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.circlepager(this.settings);
            this.circlepager = this.jQueryElement.data('circlepager');
        });
    }
    ngAfterViewChecked() {
        if (this.circlepager && this.updateRequired) {
            this.ngZone.runOutsideAngular(() => this.circlepager?.updated(this.circlepager.settings));
            this.updateRequired = false;
        }
    }
    /** Tear Down */
    ngOnDestroy() {
        // call outside the angular zone so change detection isn't triggered by the soho component.
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.circlepager) {
                try {
                    this.circlepager.destroy();
                }
                catch (e) {
                    console.error(e);
                }
                this.circlepager = null;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCirclepagerComponent, selector: "[soho-circlepager]", inputs: { slidesToShow: "slidesToShow", startingSlide: "startingSlide", loop: "loop", attributes: "attributes" }, host: { properties: { "class.circlepager": "this.isCirclePager" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-circlepager]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isCirclePager: [{
                type: HostBinding,
                args: ['class.circlepager']
            }], slidesToShow: [{
                type: Input
            }], startingSlide: [{
                type: Input
            }], loop: [{
                type: Input
            }], attributes: [{
                type: Input
            }] } });
