// ids-enterprise-ng/esm2022/lib/circlepager/soho-circlepager.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoCirclepagerComponent } from './soho-circlepager.component';
import * as i0 from "@angular/core";
export class SohoCirclepagerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerModule, declarations: [SohoCirclepagerComponent], imports: [CommonModule], exports: [SohoCirclepagerComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCirclepagerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoCirclepagerComponent
                    ],
                    exports: [
                        SohoCirclepagerComponent
                    ],
                }]
        }] });
