// ids-enterprise-ng/esm2022/lib/context-menu/soho-context-menu.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoContextMenuDirective, SohoContextMenuHeadingComponent, SohoContextMenuShortCutTextComponent, SohoContextMenuItemComponent, SohoContextMenuItemLabelComponent, SohoContextMenuSeparatorComponent } from './soho-context-menu.directive';
import * as i0 from "@angular/core";
export class SohoContextMenuModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoContextMenuModule, declarations: [SohoContextMenuDirective,
            SohoContextMenuHeadingComponent,
            SohoContextMenuShortCutTextComponent,
            SohoContextMenuItemComponent,
            SohoContextMenuItemLabelComponent,
            SohoContextMenuSeparatorComponent], imports: [CommonModule], exports: [SohoContextMenuDirective,
            SohoContextMenuHeadingComponent,
            SohoContextMenuShortCutTextComponent,
            SohoContextMenuItemComponent,
            SohoContextMenuItemLabelComponent,
            SohoContextMenuSeparatorComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextMenuModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoContextMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoContextMenuDirective,
                        SohoContextMenuHeadingComponent,
                        SohoContextMenuShortCutTextComponent,
                        SohoContextMenuItemComponent,
                        SohoContextMenuItemLabelComponent,
                        SohoContextMenuSeparatorComponent
                    ],
                    exports: [
                        SohoContextMenuDirective,
                        SohoContextMenuHeadingComponent,
                        SohoContextMenuShortCutTextComponent,
                        SohoContextMenuItemComponent,
                        SohoContextMenuItemLabelComponent,
                        SohoContextMenuSeparatorComponent
                    ]
                }]
        }] });
