// ids-enterprise-ng/esm2022/lib/buttonset/soho-buttonset.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoButtonsetComponent } from './soho-buttonset.component';
import * as i0 from "@angular/core";
export class SohoButtonsetModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonsetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonsetModule, declarations: [SohoButtonsetComponent], imports: [CommonModule], exports: [SohoButtonsetComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonsetModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoButtonsetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoButtonsetComponent
                    ],
                    exports: [
                        SohoButtonsetComponent
                    ]
                }]
        }] });
