// ids-enterprise-ng/esm2022/lib/module-nav-settings/soho-module-nav-settings.component.mjs
// tslint:disable-next-line:no-unused-variable
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Angular Wrapper for the Soho Module Nav Settings element.
 * This Component attaches to an element annotated with the `soho-module-nav-settings` attribute,
 */
export class SohoModuleNavSettingsComponent {
    /** Constructor. */
    constructor(elementRef, ngZone) {
        this.elementRef = elementRef;
        this.ngZone = ngZone;
        /** Stored settings */
        this._options = {
            displayMode: false,
        };
        /** Internal use flags */
        this._updateRequired = false;
    }
    // -------------------------------------------
    // Public API
    // -------------------------------------------
    accordionAPI() {
        return this.modulenavsettings?.accordionAPI;
    }
    accordionEl() {
        return this.modulenavsettings?.accordionEl;
    }
    menuAPI() {
        return this.modulenavsettings?.menuAPI;
    }
    init() {
        this.modulenavsettings?.init();
    }
    /** Triggers a UI Resync. */
    updated(val) {
        if (val) {
            this._options = jQuery.extend({}, this._options, val);
            if (this.modulenavsettings) {
                this.ngZone.runOutsideAngular(() => {
                    this.modulenavsettings?.updated(this._options);
                });
            }
        }
    }
    teardown() {
        this.modulenavsettings?.teardown();
    }
    // ------------------------------------------
    // Lifecycle Events
    // ------------------------------------------
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            // Initialize/store instance
            this.jQueryElement = jQuery(this.elementRef.nativeElement);
            this.jQueryElement.modulenav(this._options);
            this.modulenavsettings = this.jQueryElement.data('modulenavsettings');
        });
    }
    ngAfterViewChecked() {
        if (this.modulenavsettings && this._updateRequired) {
            this.ngZone.runOutsideAngular(() => this.modulenavsettings?.updated(this._options));
            this._updateRequired = false;
        }
    }
    ngOnDestroy() {
        // call outside the angular zone so change detection
        // isn't triggered by the soho component.
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.modulenavsettings) {
                this.modulenavsettings.destroy();
                this.modulenavsettings = null;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoModuleNavSettingsComponent, selector: "soho-module-nav-settings, [soho-module-nav-settings]", ngImport: i0, template: "<ng-content></ng-content>\n", styles: ["soho-module-nav-settings{display:contents}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsComponent, decorators: [{
            type: Component,
            args: [{ selector: 'soho-module-nav-settings, [soho-module-nav-settings]', changeDetection: ChangeDetectionStrategy.OnPush, template: "<ng-content></ng-content>\n", styles: ["soho-module-nav-settings{display:contents}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; } });
