// ids-enterprise-ng/esm2022/lib/module-nav-settings/soho-module-nav-settings.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoModuleNavSettingsComponent } from './soho-module-nav-settings.component';
import * as i0 from "@angular/core";
export class SohoModuleNavSettingsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsModule, declarations: [SohoModuleNavSettingsComponent], imports: [CommonModule], exports: [SohoModuleNavSettingsComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSettingsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoModuleNavSettingsComponent,
                    ],
                    exports: [
                        SohoModuleNavSettingsComponent,
                    ]
                }]
        }] });
