// ids-enterprise-ng/esm2022/lib/field-filter/soho-field-filter.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoFieldFilterDirective } from './soho-field-filter.directive';
import * as i0 from "@angular/core";
export class SohoFieldFilterModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldFilterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldFilterModule, declarations: [SohoFieldFilterDirective], imports: [CommonModule], exports: [SohoFieldFilterDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldFilterModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFieldFilterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoFieldFilterDirective
                    ],
                    exports: [
                        SohoFieldFilterDirective
                    ]
                }]
        }] });
