// ids-enterprise-ng/esm2022/lib/notification/soho-notification.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoNotificationService {
    static { this.ERROR = 'error'; }
    static { this.ALERT = 'alert'; }
    static { this.INFO = 'info'; }
    static { this.SUCCESS = 'success'; }
    /**
     * Show a notification using the specified options.
     */
    show(options) {
        jQuery('body').notification(options).data('notification').registerNotification();
    }
    /**
     * Hide a specified notification
     *
     * @param id id of notification
     */
    close(id) {
        jQuery('body').data('notification')?.close(id);
    }
    /**
     * Hide the latest notification in the list
     */
    closeLatest() {
        jQuery('body').data('notification')?.closeLatest();
    }
    /**
     * Hide all the notifications in the list
     */
    closeAll() {
        jQuery('body').data('notification')?.closeAll();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationService }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationService, decorators: [{
            type: Injectable
        }] });
