// ids-enterprise-ng/esm2022/lib/swipe-action/soho-swipe-action.component.mjs
import { Component, HostBinding, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoSwipeActionComponent {
    get isSwipeAction() {
        return true;
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            // Initialize the component
            this.jQueryElement.swipeaction();
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.swipeaction) {
                this.swipeaction.destroy();
                this.swipeaction = null;
            }
        });
    }
    updated(settings) {
        this.ngZone.runOutsideAngular(() => this.swipeaction?.updated(settings));
        return this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoSwipeActionComponent, selector: "[soho-swipe-action]", host: { properties: { "class.swipe-action": "this.isSwipeAction" } }, ngImport: i0, template: `
    <div class="swipe-container">
      <ng-content></ng-content>
    </div>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-swipe-action]',
                    template: `
    <div class="swipe-container">
      <ng-content></ng-content>
    </div>`,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isSwipeAction: [{
                type: HostBinding,
                args: ['class.swipe-action']
            }] } });
