// ids-enterprise-ng/esm2022/lib/swipe-action/soho-swipe-action.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoSwipeActionComponent } from './soho-swipe-action.component';
import * as i0 from "@angular/core";
export class SohoSwipeActionModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionModule, declarations: [SohoSwipeActionComponent], imports: [CommonModule], exports: [SohoSwipeActionComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSwipeActionModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoSwipeActionComponent,
                    ],
                    exports: [
                        SohoSwipeActionComponent,
                    ],
                }]
        }] });
