// ids-enterprise-ng/esm2022/lib/personalize/soho-personalize.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoPersonalizeDirective } from './soho-personalize.directive';
import * as i0 from "@angular/core";
export class SohoPersonalizeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPersonalizeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoPersonalizeModule, declarations: [SohoPersonalizeDirective], imports: [CommonModule], exports: [SohoPersonalizeDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPersonalizeModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPersonalizeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoPersonalizeDirective
                    ],
                    exports: [
                        SohoPersonalizeDirective
                    ]
                }]
        }] });
