// ids-enterprise-ng/esm2022/lib/header/soho-header.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoHeaderComponent {
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.isHeader = true;
        this.isPersonalizable = true;
        /**
         * This event is fired when the status of the header is changed.
         *
         */
        this.updated = new EventEmitter();
    }
    /**
     * Used to manually remove the back button when
     * Which is used in the header via the list/detail pattern
     */
    removeBackButton() {
        this.header?.removeBackButton();
    }
    ngAfterViewInit() {
        // Wrap for later.
        this.jQueryElement = jQuery(this.elementRef.nativeElement);
        // Initialise the SoHoXi control.
        this.jQueryElement.header();
        // Once the control is initialised, extract the control
        // plug-in from the element.  The element name is
        // defined by the plug-in, but in this case is 'sohoxiHeader'.
        this.header = this.jQueryElement.data('header');
        // Initialize any event handlers.
        this.jQueryElement.on('updated', (_e, args) => {
            this.updated.emit(args);
        });
    }
    ngOnDestroy() {
        if (this.jQueryElement) {
            this.jQueryElement.off();
            this.jQueryElement = undefined;
        }
        if (this.header) {
            this.header.destroy();
            this.header = undefined;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHeaderComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoHeaderComponent, selector: "soho-header", inputs: { hasToolbar: "hasToolbar", hasTabs: "hasTabs" }, outputs: { updated: "updated" }, host: { properties: { "class.header": "this.isHeader", "class.is-personalizable": "this.isPersonalizable", "class.has-toolbar": "this.hasToolbar", "class.has-tabs": "this.hasTabs" } }, ngImport: i0, template: "<ng-content></ng-content>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHeaderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'soho-header', changeDetection: ChangeDetectionStrategy.OnPush, template: "<ng-content></ng-content>\n" }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isHeader: [{
                type: HostBinding,
                args: ['class.header']
            }], isPersonalizable: [{
                type: HostBinding,
                args: ['class.is-personalizable']
            }], hasToolbar: [{
                type: HostBinding,
                args: ['class.has-toolbar']
            }, {
                type: Input
            }], hasTabs: [{
                type: HostBinding,
                args: ['class.has-tabs']
            }, {
                type: Input
            }], updated: [{
                type: Output
            }] } });
