// ids-enterprise-ng/esm2022/lib/spinbox/soho-spinbox.module.mjs
import { NgModule } from '@angular/core';
import { SohoSpinboxComponent } from './soho-spinbox.component';
import * as i0 from "@angular/core";
export class SohoSpinboxModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSpinboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoSpinboxModule, declarations: [SohoSpinboxComponent], exports: [SohoSpinboxComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSpinboxModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSpinboxModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                    declarations: [
                        SohoSpinboxComponent
                    ],
                    exports: [
                        SohoSpinboxComponent
                    ]
                }]
        }] });
