// ids-enterprise-ng/esm2022/lib/menu-button/soho-menu-button.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoIconModule } from '../icon/soho-icon.module';
import { SohoMenuButtonComponent } from './soho-menu-button.component';
import * as i0 from "@angular/core";
export class SohoMenuButtonModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMenuButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoMenuButtonModule, declarations: [SohoMenuButtonComponent], imports: [CommonModule,
            SohoIconModule], exports: [SohoMenuButtonComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMenuButtonModule, imports: [CommonModule,
            SohoIconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoMenuButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoIconModule
                    ],
                    declarations: [
                        SohoMenuButtonComponent
                    ],
                    exports: [
                        SohoMenuButtonComponent
                    ]
                }]
        }] });
