// ids-enterprise-ng/esm2022/lib/listbuilder/soho-listbuilder.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, ViewChild } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "../button/soho-button.component";
import * as i2 from "../toolbar/soho-toolbar.component";
export class SohoListBuilderComponent {
    get isListbuilder() {
        return true;
    }
    /**
     * Constructor.
     */
    constructor(ngZone) {
        this.ngZone = ngZone;
        this.options = {};
        /** Calls events after a top button action */
        this.afteradd = new EventEmitter();
        this.aftergoup = new EventEmitter();
        this.aftergodown = new EventEmitter();
        this.afterdelete = new EventEmitter();
        /** Calls events before a top button action */
        this.beforeedit = new EventEmitter();
        this.beforeadd = new EventEmitter();
        this.beforegoup = new EventEmitter();
        this.beforegodown = new EventEmitter();
        this.beforedelete = new EventEmitter();
        /** Update event on arranging items (drag and drop item) */
        this.arrangeupdate = new EventEmitter();
        /** Calls events for edit item */
        this.entereditmode = new EventEmitter();
        this.exiteditmode = new EventEmitter();
    }
    get listClass() {
        let classes = 'listbuilder';
        if (this.class) {
            classes += ` ${this.class}`;
        }
        return classes;
    }
    /**
     * Array of data
     */
    set dataset(value) {
        this.options.dataset = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.dataset = value;
            this.updateRequired = true;
        }
    }
    get dataset() {
        return this.options.dataset;
    }
    /** The CSS Class of the handle element */
    set handle(value) {
        this.options.handle = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.handle = value;
            this.updateRequired = true;
        }
    }
    /** "Add" action button (takes a string representing a "data-action" attribute */
    set btnAdd(value) {
        this.options.btnAdd = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.btnAdd = value;
            this.updateRequired = true;
        }
    }
    /** "Edit" action button (takes a string representing a "data-action" attribute */
    set btnEdit(value) {
        this.options.btnEdit = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.btnEdit = value;
            this.updateRequired = true;
        }
    }
    /** "Delete" action button (takes a string representing a "data-action" attribute */
    set btnDelete(value) {
        this.options.btnDelete = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.btnDelete = value;
            this.updateRequired = true;
        }
    }
    /** "GoUp" action button (takes a string representing a "data-action" attribute */
    set btnGoUp(value) {
        this.options.btnGoUp = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.btnGoUp = value;
            this.updateRequired = true;
        }
    }
    /** "GoDown" action button (takes a string representing a "data-action" attribute */
    set btnGoDown(value) {
        this.options.btnGoDown = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.btnGoDown = value;
            this.updateRequired = true;
        }
    }
    /** Add extra attributes like id's to the chart elements. For example `attributes: { name: 'id', value: 'my-unique-id' } */
    set attributes(attributes) {
        this.options.attributes = attributes;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.attributes = attributes;
            this.updateRequired = true;
        }
    }
    /** Html Template String of list. */
    set template(value) {
        this.options.template = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.template = value;
            this.updateRequired = true;
        }
    }
    /** Html Template String of list item. */
    set templateNewItem(value) {
        this.options.templateNewItem = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.templateNewItem = value;
            this.updateRequired = true;
        }
    }
    /** Html Template String of list item inner content. */
    set templateItemContent(value) {
        this.options.templateItemContent = value;
        if (this.jQueryElement && this.listbuilder) {
            this.listbuilder.settings.templateItemContent = value;
            this.updateRequired = true;
        }
    }
    ngAfterViewInit() {
        if (!this.listBuilderRef) {
            throw Error('Unable to find listbuilder reference...');
        }
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.listBuilderRef?.nativeElement);
            this.jQueryElement.listbuilder(this.options);
            this.listbuilder = this.jQueryElement.data('listbuilder');
            this.jQueryElement.on('afteradd', (...args) => this.ngZone.run(() => this.afteradd.emit(args)));
            this.jQueryElement.on('aftergoup', (...args) => this.ngZone.run(() => this.aftergoup.emit(args)));
            this.jQueryElement.on('aftergodown', (...args) => this.ngZone.run(() => this.aftergodown.emit(args)));
            this.jQueryElement.on('afterdelete', (...args) => this.ngZone.run(() => this.afterdelete.emit(args)));
            this.jQueryElement.on('beforeadd', (...args) => this.ngZone.run(() => this.beforeadd.emit(args)));
            this.jQueryElement.on('beforegoup', (...args) => this.ngZone.run(() => this.beforegoup.emit(args)));
            this.jQueryElement.on('beforegodown', (...args) => this.ngZone.run(() => this.beforegodown.emit(args)));
            this.jQueryElement.on('beforedelete', (...args) => this.ngZone.run(() => this.beforedelete.emit(args)));
            this.jQueryElement.on('entereditmode', (...args) => this.ngZone.run(() => this.entereditmode.emit(args)));
            this.jQueryElement.on('exiteditmode', (...args) => this.ngZone.run(() => this.exiteditmode.emit(args)));
            this.jQueryElement.on('arrangeupdate', (...args) => this.ngZone.run(() => this.arrangeupdate.emit(args)));
        });
    }
    ngAfterViewChecked() {
        if (this.updateRequired) {
            this.ngZone.runOutsideAngular(() => this.listbuilder?.updated());
            this.updateRequired = false;
        }
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.listbuilder) {
                this.listbuilder.destroy();
                this.listbuilder = null;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderComponent, deps: [{ token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoListBuilderComponent, selector: "[soho-listbuilder]", inputs: { sohoListBuilderElementId: "sohoListBuilderElementId", class: "class", dataset: "dataset", handle: "handle", btnAdd: "btnAdd", btnEdit: "btnEdit", btnDelete: "btnDelete", btnGoUp: "btnGoUp", btnGoDown: "btnGoDown", attributes: "attributes", template: "template", templateNewItem: "templateNewItem", templateItemContent: "templateItemContent" }, outputs: { afteradd: "afteradd", aftergoup: "aftergoup", aftergodown: "aftergodown", afterdelete: "afterdelete", beforeedit: "beforeedit", beforeadd: "beforeadd", beforegoup: "beforegoup", beforegodown: "beforegodown", beforedelete: "beforedelete", arrangeupdate: "arrangeupdate", entereditmode: "entereditmode", exiteditmode: "exiteditmode" }, host: { properties: { "class.listbuilder": "this.isListbuilder" } }, viewQueries: [{ propertyName: "listBuilderRef", first: true, predicate: ["listbuilder"], descendants: true, static: true }], ngImport: i0, template: "<div #listbuilder [class]=\"listClass\" [attr.id]=\"sohoListBuilderElementId\" >\n    <soho-toolbar class=\"formatter-toolbar\">\n        <soho-toolbar-button-set>\n            <button soho-button=\"secondary\" icon=\"add\" data-action=\"add\" title=\"Add New\">Add</button>\n            <div class=\"separator\"></div>\n            <button soho-button=\"secondary\" icon=\"up-arrow\" data-action=\"goup\" title=\"Go Up\">Go Up</button>\n            <button soho-button=\"secondary\" icon=\"down-arrow\" data-action=\"godown\" title=\"Go Down\">Go Down</button>\n            <div class=\"separator\"></div>\n            <button soho-button=\"secondary\" icon=\"edit\" data-action=\"edit\" title=\"Edit\">Edit</button>\n            <button soho-button=\"secondary\" icon=\"delete\" data-action=\"delete\" title=\"Delete\">Delete</button>\n        </soho-toolbar-button-set>\n    </soho-toolbar>\n    <div class=\"listbuilder-content\">\n        <div class=\"listview\">\n            <ul>\n                <ng-content select=\"li[soho-listview-item]\"></ng-content>\n            </ul>\n            <ng-content select=\"[soho-emptymessage]\"></ng-content>\n        </div>\n    </div>\n</div>", dependencies: [{ kind: "component", type: i1.SohoButtonComponent, selector: "button[soho-button]", inputs: ["soho-button", "buttonOptions", "toggleOnIcon", "toggleOffIcon", "replaceText", "hideMenuArrow", "hitbox", "notificationBadge", "notificationBadgeOptions", "extraIconClass", "isToggle", "isTogglePressed", "icon", "iconAlign", "isSubmit", "ripple", "state", "toggle", "expandableExpander", "moveToLeft", "moveToRight", "moveToSelected"] }, { kind: "component", type: i2.SohoToolbarComponent, selector: "soho-toolbar, div[soho-toolbar]", inputs: ["noActionsButton", "hasMoreButton", "maxVisibleButtons", "rightAligned", "rightAlign", "resizeContainers", "favorButtonset", "moreMenuSettings", "attributes"], outputs: ["beforeActivated", "activated", "afterActivated", "selected", "menuItemMouseOver"] }, { kind: "component", type: i2.SohoToolbarButtonSetComponent, selector: "soho-toolbar-button-set" }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderComponent, decorators: [{
            type: Component,
            args: [{ selector: '[soho-listbuilder]', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div #listbuilder [class]=\"listClass\" [attr.id]=\"sohoListBuilderElementId\" >\n    <soho-toolbar class=\"formatter-toolbar\">\n        <soho-toolbar-button-set>\n            <button soho-button=\"secondary\" icon=\"add\" data-action=\"add\" title=\"Add New\">Add</button>\n            <div class=\"separator\"></div>\n            <button soho-button=\"secondary\" icon=\"up-arrow\" data-action=\"goup\" title=\"Go Up\">Go Up</button>\n            <button soho-button=\"secondary\" icon=\"down-arrow\" data-action=\"godown\" title=\"Go Down\">Go Down</button>\n            <div class=\"separator\"></div>\n            <button soho-button=\"secondary\" icon=\"edit\" data-action=\"edit\" title=\"Edit\">Edit</button>\n            <button soho-button=\"secondary\" icon=\"delete\" data-action=\"delete\" title=\"Delete\">Delete</button>\n        </soho-toolbar-button-set>\n    </soho-toolbar>\n    <div class=\"listbuilder-content\">\n        <div class=\"listview\">\n            <ul>\n                <ng-content select=\"li[soho-listview-item]\"></ng-content>\n            </ul>\n            <ng-content select=\"[soho-emptymessage]\"></ng-content>\n        </div>\n    </div>\n</div>" }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }]; }, propDecorators: { isListbuilder: [{
                type: HostBinding,
                args: ['class.listbuilder']
            }], listBuilderRef: [{
                type: ViewChild,
                args: ['listbuilder', { static: true }]
            }], sohoListBuilderElementId: [{
                type: Input
            }], class: [{
                type: Input
            }], dataset: [{
                type: Input
            }], handle: [{
                type: Input
            }], btnAdd: [{
                type: Input
            }], btnEdit: [{
                type: Input
            }], btnDelete: [{
                type: Input
            }], btnGoUp: [{
                type: Input
            }], btnGoDown: [{
                type: Input
            }], attributes: [{
                type: Input
            }], template: [{
                type: Input
            }], templateNewItem: [{
                type: Input
            }], templateItemContent: [{
                type: Input
            }], afteradd: [{
                type: Output
            }], aftergoup: [{
                type: Output
            }], aftergodown: [{
                type: Output
            }], afterdelete: [{
                type: Output
            }], beforeedit: [{
                type: Output
            }], beforeadd: [{
                type: Output
            }], beforegoup: [{
                type: Output
            }], beforegodown: [{
                type: Output
            }], beforedelete: [{
                type: Output
            }], arrangeupdate: [{
                type: Output
            }], entereditmode: [{
                type: Output
            }], exiteditmode: [{
                type: Output
            }] } });
