// ids-enterprise-ng/esm2022/lib/listbuilder/soho-listbuilder.module.mjs
import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";
import { SohoButtonModule } from "../button";
import { SohoIconModule } from "../icon";
import { SohoListViewModule } from "../listview";
import { SohoToolbarModule } from "../toolbar";
import { SohoListBuilderComponent } from "./soho-listbuilder.component";
import * as i0 from "@angular/core";
export class SohoListBuilderModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderModule, declarations: [SohoListBuilderComponent], imports: [CommonModule,
            SohoButtonModule,
            SohoIconModule,
            SohoToolbarModule,
            SohoListViewModule], exports: [SohoListBuilderComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderModule, imports: [CommonModule,
            SohoButtonModule,
            SohoIconModule,
            SohoToolbarModule,
            SohoListViewModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoListBuilderModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoButtonModule,
                        SohoIconModule,
                        SohoToolbarModule,
                        SohoListViewModule
                    ],
                    declarations: [
                        SohoListBuilderComponent
                    ],
                    exports: [
                        SohoListBuilderComponent
                    ]
                }]
        }] });
