// ids-enterprise-ng/esm2022/lib/toolbar-flex/soho-toolbar-flex.component.mjs
import { Component, ChangeDetectionStrategy, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import { ArgumentHelper } from '../utils/argument.helper';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class SohoToolbarFlexSearchFieldComponent {
    /** Adds an X button for clearing the search value. */
    set clearable(value) {
        this.options.clearable = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.clearable = value;
            this.markForRefresh();
        }
    }
    /** Where it's collapsible or not */
    set collapsible(value) {
        this.options.collapsible = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.collapsible = value;
            this.markForRefresh();
        }
    }
    set collapsibleOnMobile(value) {
        this.options.collapsibleOnMobile = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.collapsibleOnMobile = value;
            this.markForRefresh();
        }
    }
    set filterMode(value) {
        this.options.filterMode = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.filterMode = value;
            this.markForRefresh();
        }
    }
    /** AutoComplete : Source Function/Data/Url/Array */
    set source(value) {
        this.options.source = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.source = value;
            this.markForRefresh();
        }
    }
    /** Template that can be passed */
    set template(value) {
        this.options.template = value;
        if (this.toolbarFlexSearchField) {
            this.toolbarFlexSearchField.settings.template = value;
            this.markForRefresh();
        }
    }
    get isSearchField() {
        return true;
    }
    constructor(changeDetector, element, ngZone) {
        this.changeDetector = changeDetector;
        this.element = element;
        this.ngZone = ngZone;
        /**
         * Options, overwriting any existing options already set on the component, and hence combining
         * this Input with the othera is not recommended.
         *
         * (Note: could change this to use {...this._options,...value})
         */
        this.options = {};
        // ------------------------------------------------------------
        this.selected = new EventEmitter();
        this.cleared = new EventEmitter();
        /**
         * @todo replace override of native element
         */
        // eslint-disable-next-line @angular-eslint/no-output-native
        this.change = new EventEmitter();
        this.searchFieldChanged = false;
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.searchfield(this.options);
            // Initialize title attribute as a soho tooltip
            if (this.jQueryElement.attr('title')) {
                this.jQueryElement.tooltip();
            }
            this.toolbarFlexSearchField = this.jQueryElement.data('searchfield');
            /**
             * Bind to jQueryElement's events
             */
            this.jQueryElement.on('selected', (...args) => this.ngZone.run(() => this.selected.emit(args)));
            this.jQueryElement.on('change', (...args) => this.ngZone.run(() => this.change.emit(args)));
            this.jQueryElement.on('cleared', (...args) => this.ngZone.run(() => this.cleared.emit(args)));
        });
    }
    ngAfterViewChecked() {
        if (this.searchFieldChanged) {
            this.ngZone.runOutsideAngular(() => this.toolbarFlexSearchField?.updated());
            this.searchFieldChanged = false;
        }
    }
    ngOnDestroy() {
        // Necessary clean up step (add additional here)
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                // clean up attached events.
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.toolbarFlexSearchField) {
                // destroy the soho component.
                this.toolbarFlexSearchField.destroy();
                this.toolbarFlexSearchField = undefined;
            }
        });
    }
    clear() {
        this.ngZone.runOutsideAngular(() => this.toolbarFlexSearchField?.clear());
    }
    // for testing
    get searchField() {
        return this.toolbarFlexSearchField;
    }
    markForRefresh() {
        this.searchFieldChanged = true;
        // ... make sure the change detector kicks in, otherwise if the inputs
        // were change programmatially the component may not be eligible for
        // updating.
        this.changeDetector.markForCheck();
    }
    updated(settings) {
        this.ngZone.runOutsideAngular(() => this.toolbarFlexSearchField?.updated(settings));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSearchFieldComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexSearchFieldComponent, selector: "input[soho-toolbar-flex-searchfield]", inputs: { options: "options", clearable: "clearable", collapsible: "collapsible", collapsibleOnMobile: "collapsibleOnMobile", filterMode: "filterMode", source: "source", template: "template" }, outputs: { selected: "selected", cleared: "cleared", change: "change" }, host: { properties: { "class.searchfield": "this.isSearchField" } }, ngImport: i0, template: '<div #toolbarFlexSearchField><ng-content></ng-content></div>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSearchFieldComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'input[soho-toolbar-flex-searchfield]',
                    template: '<div #toolbarFlexSearchField><ng-content></ng-content></div>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }, { type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { options: [{
                type: Input
            }], clearable: [{
                type: Input
            }], collapsible: [{
                type: Input
            }], collapsibleOnMobile: [{
                type: Input
            }], filterMode: [{
                type: Input
            }], source: [{
                type: Input
            }], template: [{
                type: Input
            }], selected: [{
                type: Output
            }], cleared: [{
                type: Output
            }], change: [{
                type: Output
            }], isSearchField: [{
                type: HostBinding,
                args: ['class.searchfield']
            }] } });
/**
 * soho toolbar flex more button
 */
export class SohoToolbarFlexMoreButtonComponent {
    constructor() {
        this.isMoreButton = true;
        this.isToolbarSection = true;
        this.isPageChanger = false;
        this.isDisabled = false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexMoreButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexMoreButtonComponent, selector: "soho-toolbar-flex-more-button", inputs: { isPageChanger: "isPageChanger", isDisabled: "isDisabled", moreButtonId: "moreButtonId", ajaxBeforeFunction: "ajaxBeforeFunction", menuId: "menuId" }, host: { properties: { "class.more": "this.isMoreButton", "class.toolbar-section": "this.isToolbarSection" } }, ngImport: i0, template: `<button
      class="btn-actions"
      type="button"
      [ngClass]="{'page-changer': isPageChanger}"
      [attr.id]="moreButtonId"
      [attr.disabled]="isDisabled ? 'disabled' : null"
    >
      <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
        <use href="#icon-more"></use>
      </svg>
      <span class="audible">More Actions</span>
    </button>
    <ng-content></ng-content>`, isInline: true, dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexMoreButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-toolbar-flex-more-button',
                    template: `<button
      class="btn-actions"
      type="button"
      [ngClass]="{'page-changer': isPageChanger}"
      [attr.id]="moreButtonId"
      [attr.disabled]="isDisabled ? 'disabled' : null"
    >
      <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
        <use href="#icon-more"></use>
      </svg>
      <span class="audible">More Actions</span>
    </button>
    <ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isMoreButton: [{
                type: HostBinding,
                args: ['class.more']
            }], isToolbarSection: [{
                type: HostBinding,
                args: ['class.toolbar-section']
            }], isPageChanger: [{
                type: Input
            }], isDisabled: [{
                type: Input
            }], moreButtonId: [{
                type: Input
            }], ajaxBeforeFunction: [{
                type: Input
            }], menuId: [{
                type: Input
            }] } });
/**
 * soho toolbar flex page title.
 */
export class SohoToolbarFlexPageTitleComponent {
    constructor() {
        this.isPageTitle = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexPageTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexPageTitleComponent, selector: "[soho-toolbar-flex-page-title]", host: { properties: { "class.page-title": "this.isPageTitle" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexPageTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-toolbar-flex-page-title]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isPageTitle: [{
                type: HostBinding,
                args: ['class.page-title']
            }] } });
/**
 * soho toolbar flex section title.
 */
export class SohoToolbarFlexSectionTitleComponent {
    constructor() {
        this.isSectionTitle = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSectionTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexSectionTitleComponent, selector: "[soho-toolbar-flex-section-title]", host: { properties: { "class.section-title": "this.isSectionTitle" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSectionTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-toolbar-flex-section-title]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isSectionTitle: [{
                type: HostBinding,
                args: ['class.section-title']
            }] } });
/**
 * Generic soho toolbar flex section
 */
export class SohoToolbarFlexSectionComponent {
    constructor() {
        this.isToolbarSection = true;
        this.isTitleFavor = false;
        this.isTitle = false;
        this.isButtonSet = false;
        this.isSearch = false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSectionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexSectionComponent, selector: "soho-toolbar-flex-section", inputs: { isTitleFavor: "isTitleFavor", isTitle: "isTitle", isButtonSet: "isButtonSet", isSearch: "isSearch" }, host: { properties: { "class.toolbar-section": "this.isToolbarSection", "class.favor": "this.isTitleFavor", "class.title": "this.isTitle", "class.buttonset": "this.isButtonSet", "class.search": "this.isSearch" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexSectionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-toolbar-flex-section',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isToolbarSection: [{
                type: HostBinding,
                args: ['class.toolbar-section']
            }], isTitleFavor: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.favor']
            }], isTitle: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.title']
            }], isButtonSet: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.buttonset']
            }], isSearch: [{
                type: Input
            }, {
                type: HostBinding,
                args: ['class.search']
            }] } });
/**
 * Soho toolbar Navigation button
 */
export class SohoToolbarFlexNavButtonComponent {
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        this.isIconButton = true;
        this.isAppMenuTrigger = true;
        this.typeAttr = 'button';
        this._buttonOptions = {};
    }
    set buttonOptions(buttonOptions) {
        ArgumentHelper.checkNotNull('buttonOptions', buttonOptions);
        this._buttonOptions = buttonOptions;
        if (this.button) {
            // todo: how to update the button when options change?
        }
    }
    get buttonOptions() {
        return this._buttonOptions;
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            const self = this;
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.button(this._buttonOptions);
            this.button = this.jQueryElement.data('button');
        });
    }
    updated(settings) {
        if (settings) {
            this._buttonOptions = Soho.utils.mergeSettings(this.element[0], settings, this._buttonOptions);
        }
        if (this.button) {
            this.ngZone.runOutsideAngular(() => {
                this.button?.updated(this._buttonOptions);
            });
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexNavButtonComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexNavButtonComponent, selector: "button[soho-toolbar-flex-nav-button]", inputs: { buttonOptions: "buttonOptions" }, host: { properties: { "class.btn-icon": "this.isIconButton", "class.application-menu-trigger": "this.isAppMenuTrigger", "attr.type": "this.typeAttr" } }, ngImport: i0, template: `
    <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
      <use href="#icon-menu"></use>
    </svg>
    <span class="audible">
        <ng-content></ng-content>
    </span>
            `, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexNavButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'button[soho-toolbar-flex-nav-button]',
                    template: `
    <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
      <use href="#icon-menu"></use>
    </svg>
    <span class="audible">
        <ng-content></ng-content>
    </span>
            `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isIconButton: [{
                type: HostBinding,
                args: ['class.btn-icon']
            }], isAppMenuTrigger: [{
                type: HostBinding,
                args: ['class.application-menu-trigger']
            }], typeAttr: [{
                type: HostBinding,
                args: ['attr.type']
            }], buttonOptions: [{
                type: Input
            }] } });
/**
 * The main soho toolbar flex component
 */
export class SohoToolbarFlexComponent {
    /**
     * Allows a moreMenuSettings object to be propagated down into the Toolbar Flex.
     * the jQuery Toolbar Flex component handles passing these settings into the More Actions
     * menu's jQuery Popupmenu.
     */
    set moreMenuBeforeOpenFunction(beforeOpen) {
        if (!this._options.moreMenuSettings) {
            this._options.moreMenuSettings = {};
        }
        this._options.moreMenuSettings.beforeOpen = beforeOpen;
        if (this.toolbarFlex) {
            this.toolbarFlex.settings.beforeMoreMenuOpen = beforeOpen;
            this.markForRefresh();
        }
    }
    /** Add extra attributes like id's to the component **/
    set attributes(attributes) {
        this._options.attributes = attributes;
        if (this.toolbarFlex) {
            this.toolbarFlex.settings.attributes = attributes;
            this.markForRefresh();
        }
    }
    constructor(changeDetector, element, ngZone) {
        this.changeDetector = changeDetector;
        this.element = element;
        this.ngZone = ngZone;
        this.isToolbar = true;
        /**
         * The selected event is fired when a toolbar button has been clicked.
         */
        this.selected = new EventEmitter();
        this._options = {};
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            // Assign element to local variable
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.toolbarflex(this._options);
            this.toolbarFlex = this.jQueryElement.data('toolbarFlex');
            this.jQueryElement.on('selected', (event, item) => this.ngZone.run(() => {
                this.selected.emit({ event, item });
            }));
        });
    }
    ngAfterViewChecked() {
        if (this.toolbarFlexChanged) {
            this.updated();
            this.toolbarFlexChanged = false;
        }
    }
    ngOnDestroy() {
        // call outside the angular zone so change detection isn't triggered by the soho component.
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.toolbarFlex) {
                this.toolbarFlex.destroy();
                this.toolbarFlex = null;
            }
        });
    }
    updated(settings) {
        if (this.toolbarFlex) {
            this.ngZone.runOutsideAngular(() => this.toolbarFlex?.updated(settings));
        }
    }
    /**
     * The buttonset API for the modal dialog.
     *
     * @returns the buttonset API for the modal dialog, if initialised.
     */
    get buttonsetAPIs() {
        return this.toolbarFlex ? this.toolbarFlex.buttonsetAPIs : undefined;
    }
    /**
     * The buttonset API for the modal dialog.
     *
     * @returns the buttonset API for the modal dialog, if initialised.
     */
    get buttonsets() {
        return this.toolbarFlex ? this.toolbarFlex.buttonsets : undefined;
    }
    // For testing
    get options() {
        return this._options;
    }
    markForRefresh() {
        this.toolbarFlexChanged = true;
        this.changeDetector.markForCheck();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoToolbarFlexComponent, selector: "soho-toolbar-flex, div[soho-toolbar-flex]", inputs: { moreMenuBeforeOpenFunction: "moreMenuBeforeOpenFunction", attributes: "attributes" }, outputs: { selected: "selected" }, host: { properties: { "class.flex-toolbar": "this.isToolbar" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-toolbar-flex, div[soho-toolbar-flex]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }, { type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isToolbar: [{
                type: HostBinding,
                args: ['class.flex-toolbar']
            }], moreMenuBeforeOpenFunction: [{
                type: Input
            }], attributes: [{
                type: Input
            }], selected: [{
                type: Output
            }] } });
