// ids-enterprise-ng/esm2022/lib/toolbar-flex/soho-toolbar-flex.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoToolbarFlexComponent, SohoToolbarFlexNavButtonComponent, SohoToolbarFlexSectionTitleComponent, SohoToolbarFlexPageTitleComponent, SohoToolbarFlexMoreButtonComponent, SohoToolbarFlexSectionComponent, SohoToolbarFlexSearchFieldComponent, } from './soho-toolbar-flex.component';
import * as i0 from "@angular/core";
export class SohoToolbarFlexModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexModule, declarations: [SohoToolbarFlexComponent,
            SohoToolbarFlexNavButtonComponent,
            SohoToolbarFlexSectionTitleComponent,
            SohoToolbarFlexPageTitleComponent,
            SohoToolbarFlexMoreButtonComponent,
            SohoToolbarFlexSectionComponent,
            SohoToolbarFlexSearchFieldComponent], imports: [CommonModule], exports: [SohoToolbarFlexComponent,
            SohoToolbarFlexNavButtonComponent,
            SohoToolbarFlexSectionTitleComponent,
            SohoToolbarFlexPageTitleComponent,
            SohoToolbarFlexMoreButtonComponent,
            SohoToolbarFlexSectionComponent,
            SohoToolbarFlexSearchFieldComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarFlexModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoToolbarFlexComponent,
                        SohoToolbarFlexNavButtonComponent,
                        SohoToolbarFlexSectionTitleComponent,
                        SohoToolbarFlexPageTitleComponent,
                        SohoToolbarFlexMoreButtonComponent,
                        SohoToolbarFlexSectionComponent,
                        SohoToolbarFlexSearchFieldComponent,
                    ],
                    exports: [
                        SohoToolbarFlexComponent,
                        SohoToolbarFlexNavButtonComponent,
                        SohoToolbarFlexSectionTitleComponent,
                        SohoToolbarFlexPageTitleComponent,
                        SohoToolbarFlexMoreButtonComponent,
                        SohoToolbarFlexSectionComponent,
                        SohoToolbarFlexSearchFieldComponent,
                    ]
                }]
        }] });
