// ids-enterprise-ng/esm2022/lib/drag/soho-drag.directive.mjs
import { Directive, Output, EventEmitter, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoDragDirective {
    set dragOptions(dragOptions) {
        this._dragOptions = dragOptions;
    }
    constructor(elementRef, ngZone) {
        this.elementRef = elementRef;
        this.ngZone = ngZone;
        this.dragStart = new EventEmitter();
        this.dragEnd = new EventEmitter();
        this.dragging = new EventEmitter();
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.elementRef.nativeElement);
            this.jQueryElement.drag(this._dragOptions);
            this.drag = this.jQueryElement.data('drag');
            // bind to jquery events and emit as angular events
            this.jQueryElement.on('dragstart', (event, ui) => this.ngZone.run(() => this.dragStart.emit({ event, ui })));
            this.jQueryElement.on('dragend', (event, ui) => this.ngZone.run(() => this.dragEnd.emit({ event, ui })));
            this.jQueryElement.on('drag', (event, ui) => this.ngZone.run(() => this.dragging.emit({ event, ui })));
        });
    }
    getElementsFromPoint(x, y) {
        return this.drag?.getElementsFromPoint(x, y);
    }
    ngOnDestroy() {
        this.drag?.destroy();
    }
    updated(settings) {
        if (settings) {
            this.drag?.updated(settings);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoDragDirective, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoDragDirective, selector: "[soho-drag]", inputs: { dragOptions: "dragOptions" }, outputs: { dragStart: "dragStart", dragEnd: "dragEnd", dragging: "dragging" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoDragDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-drag]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { dragOptions: [{
                type: Input
            }], dragStart: [{
                type: Output
            }], dragEnd: [{
                type: Output
            }], dragging: [{
                type: Output
            }] } });
