// ids-enterprise-ng/esm2022/lib/module-nav-switcher/soho-module-nav-switcher.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoModuleNavSwitcherComponent } from './soho-module-nav-switcher.component';
import * as i0 from "@angular/core";
export class SohoModuleNavSwitcherModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSwitcherModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSwitcherModule, declarations: [SohoModuleNavSwitcherComponent], imports: [CommonModule], exports: [SohoModuleNavSwitcherComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSwitcherModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoModuleNavSwitcherModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoModuleNavSwitcherComponent,
                    ],
                    exports: [
                        SohoModuleNavSwitcherComponent,
                    ]
                }]
        }] });
