// ids-enterprise-ng/esm2022/lib/form-compact/soho-form-compact.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoFormCompactComponent } from './soho-form-compact.component';
import * as i0 from "@angular/core";
export class SohoFormCompactModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactModule, declarations: [SohoFormCompactComponent], imports: [CommonModule], exports: [SohoFormCompactComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoFormCompactComponent
                    ],
                    exports: [
                        SohoFormCompactComponent
                    ],
                }]
        }] });
