// ids-enterprise-ng/esm2022/lib/form-compact/soho-form-compact.component.mjs
import { ChangeDetectionStrategy, Component, HostBinding, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoFormCompactComponent {
    get isFormCompact() {
        return true;
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        /** Options. */
        this.options = {};
        this.updateRequired = false;
    }
    /** Setup */
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.formcompact(this.options);
            this.formcompact = this.jQueryElement.data('formcompact');
        });
    }
    ngAfterViewChecked() {
        if (this.formcompact && this.updateRequired) {
            this.ngZone.runOutsideAngular(() => this.formcompact?.updated(this.formcompact?.settings));
            this.updateRequired = false;
        }
    }
    /** Tear Down */
    ngOnDestroy() {
        // call outside the angular zone so change detection isn't triggered by the soho component.
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.formcompact) {
                this.formcompact.destroy();
                this.formcompact = null;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoFormCompactComponent, selector: "[soho-form-compact]", host: { properties: { "class.form-compact": "this.isFormCompact" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFormCompactComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-form-compact]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isFormCompact: [{
                type: HostBinding,
                args: ['class.form-compact']
            }] } });
