// ids-enterprise-ng/esm2022/lib/toolbar/soho-toolbar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoToolbarComponent, SohoToolbarButtonSetComponent, SohoToolbarTitleComponent, SohoToolbarNavButtonComponent, SohoSectionTitleComponent, SohoSelectionCountComponent, SohoPageTitleComponent, SohoToolbarMoreButtonComponent, SohoToolbarSearchFieldComponent, SohoToolbarSearchFieldWrapperComponent, } from './soho-toolbar.component';
import * as i0 from "@angular/core";
export class SohoToolbarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarModule, declarations: [SohoToolbarComponent,
            SohoToolbarButtonSetComponent,
            SohoToolbarTitleComponent,
            SohoToolbarNavButtonComponent,
            SohoSectionTitleComponent,
            SohoSelectionCountComponent,
            SohoPageTitleComponent,
            SohoToolbarMoreButtonComponent,
            SohoToolbarSearchFieldComponent,
            SohoToolbarSearchFieldWrapperComponent], imports: [CommonModule], exports: [SohoToolbarComponent,
            SohoToolbarButtonSetComponent,
            SohoToolbarTitleComponent,
            SohoToolbarNavButtonComponent,
            SohoSectionTitleComponent,
            SohoSelectionCountComponent,
            SohoPageTitleComponent,
            SohoToolbarMoreButtonComponent,
            SohoToolbarSearchFieldComponent,
            SohoToolbarSearchFieldWrapperComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoToolbarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoToolbarComponent,
                        SohoToolbarButtonSetComponent,
                        SohoToolbarTitleComponent,
                        SohoToolbarNavButtonComponent,
                        SohoSectionTitleComponent,
                        SohoSelectionCountComponent,
                        SohoPageTitleComponent,
                        SohoToolbarMoreButtonComponent,
                        SohoToolbarSearchFieldComponent,
                        SohoToolbarSearchFieldWrapperComponent,
                    ],
                    exports: [
                        SohoToolbarComponent,
                        SohoToolbarButtonSetComponent,
                        SohoToolbarTitleComponent,
                        SohoToolbarNavButtonComponent,
                        SohoSectionTitleComponent,
                        SohoSelectionCountComponent,
                        SohoPageTitleComponent,
                        SohoToolbarMoreButtonComponent,
                        SohoToolbarSearchFieldComponent,
                        SohoToolbarSearchFieldWrapperComponent,
                    ]
                }]
        }] });
