// ids-enterprise-ng/esm2022/lib/checkbox/soho-checkbox.component.mjs
import { ChangeDetectionStrategy, Component, HostBinding, EventEmitter, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Wrapper for soho checkboxes.
 *
 * Note the control must have type="checkbox" for the values to be passed
 * to and from any ng model instances.
 */
export class SohoCheckBoxComponent {
    /**
     * Bind attributes to the host input element
     */
    get isCheckBoxType() {
        return 'checkbox';
    }
    get isCheckBox() {
        return !this.switch;
    }
    get isPartialCheckBox() {
        return this.partial ? true : false;
    }
    get isPartialAriaChecked() {
        return this.partial ? 'mixed' : null;
    }
    get isSwitch() {
        return this.switch;
    }
    /**
     * Constructor.
     *
     * @param elementRef - the element matching the component's selector.
     */
    constructor(elementRef, ngZone) {
        this.elementRef = elementRef;
        this.ngZone = ngZone;
        /**
         * Indicate that the checkbox is displayed as a switch.
         */
        this.switch = false;
        // -------------------------------------------
        // Component Output
        // -------------------------------------------
        /** Called when the checkbox value changes. */
        this.changeEvent = new EventEmitter();
        /** Called when the checkbox updates in some way. */
        this.updateEvent = new EventEmitter();
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.elementRef.nativeElement);
            /**
             * Bind to jQueryElement's events
             */
            this.jQueryElement.on('change', (event) => this.ngZone.run(() => this.changeEvent.emit(event)));
            this.jQueryElement.on('updated', (event) => this.ngZone.run(() => this.updateEvent.emit(event)));
            // no control initializer for checkbox
        });
    }
    /**
     * Destructor.
     */
    ngOnDestroy() {
        // No jQuery control.
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCheckBoxComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCheckBoxComponent, selector: "[soho-checkbox]", inputs: { partial: "partial", switch: "switch", checked: "checked" }, outputs: { changeEvent: "changeEvent", updateEvent: "updateEvent" }, host: { properties: { "attr.type": "this.isCheckBoxType", "class.checkbox": "this.isCheckBox", "class.partial": "this.isPartialCheckBox", "attr.aria-checked": "this.isPartialAriaChecked", "class.switch": "this.isSwitch", "attr.checked": "this.checked" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCheckBoxComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-checkbox]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { partial: [{
                type: Input
            }], switch: [{
                type: Input
            }], isCheckBoxType: [{
                type: HostBinding,
                args: ['attr.type']
            }], isCheckBox: [{
                type: HostBinding,
                args: ['class.checkbox']
            }], isPartialCheckBox: [{
                type: HostBinding,
                args: ['class.partial']
            }], isPartialAriaChecked: [{
                type: HostBinding,
                args: ['attr.aria-checked']
            }], isSwitch: [{
                type: HostBinding,
                args: ['class.switch']
            }], checked: [{
                type: HostBinding,
                args: ['attr.checked']
            }, {
                type: Input
            }], changeEvent: [{
                type: Output
            }], updateEvent: [{
                type: Output
            }] } });
