// ids-enterprise-ng/esm2022/lib/soho-components.module.mjs
import { NgModule } from '@angular/core';
import { SohoAboutModule } from './about/soho-about.module';
import { SohoAccordionModule } from './accordion/soho-accordion.module';
import { SohoActionsheetModule } from './actionsheet/soho-actionsheet.module';
import { SohoAlertModule } from './alert/soho-alert.module';
import { SohoApplicationMenuModule } from './application-menu/soho-application-menu.module';
import { SohoArrangeModule } from './arrange/soho-arrange.module';
import { SohoAutoCompleteModule } from './autocomplete/soho-autocomplete.module';
import { SohoBarModule } from './bar/soho-bar.module';
import { SohoBulletModule } from './bullet/soho-bullet.module';
import { SohoBlockGridModule } from './blockgrid/soho-blockgrid.module';
import { SohoBreadcrumbModule } from './breadcrumb/soho-breadcrumb.module';
import { SohoBusyIndicatorModule } from './busyindicator/soho-busyindicator.module';
import { SohoButtonModule } from './button/soho-button.module';
import { SohoButtonsetModule } from './buttonset/soho-buttonset.module';
import { SohoCalendarModule } from './calendar/soho-calendar.module';
import { SohoCardModule } from './card/soho-card.module';
import { SohoChartModule } from './chart/soho-chart.module';
import { SohoCheckBoxModule } from './checkbox/soho-checkbox.module';
import { SohoCirclepagerModule } from './circlepager/soho-circlepager.module';
import { SohoColorPickerModule } from './colorpicker/soho-colorpicker.module';
import { SohoColumnModule } from './column/soho-column.module';
import { SohoContextMenuModule } from './context-menu/soho-context-menu.module';
import { SohoContextualActionPanelModule } from './contextual-action-panel/soho-contextual-action-panel.module';
import { SohoDataGridModule } from './datagrid/soho-datagrid.module';
import { SohoDatePickerModule } from './datepicker/soho-datepicker.module';
import { SohoDragModule } from './drag/soho-drag.module';
import { SohoDropDownModule } from './dropdown/soho-dropdown.module';
import { SohoEditorModule } from './editor/soho-editor.module';
import { SohoEmptyMessageModule } from './emptymessage/soho-emptymessage.module';
import { SohoErrorModule } from './error/soho-error.module';
import { SohoExpandableAreaModule } from './expandablearea/soho-expandablearea.module';
import { SohoFieldFilterModule } from './field-filter/soho-field-filter.module';
import { SohoFieldOptionsModule } from './field-options/soho-field-options.module';
import { SohoFileUploadAdvancedModule } from './fileupload-advanced/soho-fileupload-advanced.module';
import { SohoFileUploadModule } from './fileupload/soho-fileupload.module';
import { SohoFormCompactModule } from './form-compact/soho-form-compact.module';
import { SohoHeaderModule } from './header/soho-header.module';
import { SohoHierarchyModule } from './hierarchy/soho-hierarchy.module';
import { SohoHomePageModule } from './homepage/soho-homepage.module';
import { SohoHyperlinkModule } from './hyperlink/soho-hyperlink.module';
import { SohoIconModule } from './icon/soho-icon.module';
import { SohoInputModule } from './input/soho-input.module';
import { SohoInputValidateModule } from './input-validate/soho-input-validate.module';
import { SohoLabelModule } from './label/soho-label.module';
import { SohoLineModule } from './line/soho-line.module';
import { SohoListViewModule } from './listview/soho-listview.module';
import { SohoLocaleModule } from './locale/soho-locale.module';
import { SohoLookupModule } from './lookup/soho-lookup.module';
import { SohoMaskModule } from './mask/soho-mask.module';
import { SohoMastheadModule } from './masthead/soho-masthead.module';
import { SohoMenuButtonModule } from './menu-button/soho-menu-button.module';
import { SohoMessageModule } from './message/soho-message.module';
import { SohoModalDialogModule } from './modal-dialog/soho-modal-dialog.module';
import { SohoModuleNavModule } from './module-nav/soho-module-nav.module';
import { SohoModuleNavContainerModule } from './module-nav-container/soho-module-nav-container.module';
import { SohoModuleNavSettingsModule } from './module-nav-settings/soho-module-nav-settings.module';
import { SohoModuleNavSwitcherModule } from './module-nav-switcher/soho-module-nav-switcher.module';
import { SohoMonthViewModule } from './monthview/soho-monthview.module';
import { SohoNotificationModule } from './notification/soho-notification.module';
import { SohoNotificationBadgeModule } from './notification-badge/soho-notification-badge.module';
import { SohoPagerModule } from './pager/soho-pager.module';
import { SohoPersonalizeModule } from './personalize/soho-personalize.module';
import { SohoPieModule } from './pie/soho-pie.module';
import { SohoProgressModule } from './progress/soho-progress.module';
import { SohoPopDownModule } from './popdown/soho-popdown.module';
import { SohoPopupMenuModule } from './popupmenu/soho-popupmenu.module';
import { SohoRadarModule } from './radar/soho-radar.module';
import { SohoRadioButtonModule } from './radiobutton/soho-radiobutton.module';
import { SohoRatingModule } from './rating/soho-rating.module';
import { SohoRenderLoopModule } from './renderLoop/soho-render-loop.module';
import { SohoSearchFieldModule } from './searchfield/soho-searchfield.module';
import { SohoSliderModule } from './slider/soho-slider.module';
import { SohoSparklineModule } from './sparkline/soho-sparkline.module';
import { SohoSplitterModule } from './splitter/soho-splitter.module';
import { SohoSpinboxModule } from './spinbox/soho-spinbox.module';
import { SohoStepChartModule } from './stepchart';
import { SohoStepProcessModule } from './stepprocess/soho-stepprocess.module';
import { SohoSwapListModule } from './swaplist/soho-swaplist.module';
import { SohoSwipeActionModule } from './swipe-action/soho-swipe-action.module';
import { SohoTabsModule } from './tabs/soho-tabs.module';
import { SohoTagModule } from './tag/soho-tag.module';
import { SohoTextAreaModule } from './textarea/soho-textarea.module';
import { SohoTimePickerModule } from './timepicker/soho-timepicker.module';
import { SohoToastModule } from './toast/soho-toast.module';
import { SohoToolbarFlexModule } from './toolbar-flex/soho-toolbar-flex.module';
import { SohoToolbarModule } from './toolbar/soho-toolbar.module';
import { SohoTooltipModule } from './tooltip/soho-tooltip.module';
import { SohoTrackDirtyModule } from './trackdirty/soho-trackdirty.module';
import { SohoTreeModule } from './tree/soho-tree.module';
import { SohoTreemapModule } from './treemap/soho-treemap.module';
import { SohoVersionInitializerModule } from './version/version-initializer.module';
import { SohoWeekViewModule } from './week-view/soho-week-view.module';
import { SohoWizardModule } from './wizard/soho-wizard.module';
import { SohoListBuilderModule } from './listbuilder';
import * as i0 from "@angular/core";
export class SohoComponentsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoComponentsModule, imports: [SohoAboutModule,
            SohoAccordionModule,
            SohoActionsheetModule,
            SohoAlertModule,
            SohoApplicationMenuModule,
            SohoArrangeModule,
            SohoAutoCompleteModule,
            SohoBarModule,
            SohoBlockGridModule,
            SohoBreadcrumbModule,
            SohoBulletModule,
            SohoBusyIndicatorModule,
            SohoButtonModule,
            SohoButtonsetModule,
            SohoCalendarModule,
            SohoChartModule,
            SohoCheckBoxModule,
            SohoCardModule,
            SohoCirclepagerModule,
            SohoColorPickerModule,
            SohoColumnModule,
            SohoContextualActionPanelModule,
            SohoContextMenuModule,
            SohoDataGridModule,
            SohoDatePickerModule,
            SohoDragModule,
            SohoDropDownModule,
            SohoEditorModule,
            SohoEmptyMessageModule,
            SohoErrorModule,
            SohoExpandableAreaModule,
            SohoFieldFilterModule,
            SohoFieldOptionsModule,
            SohoFileUploadAdvancedModule,
            SohoFileUploadModule,
            SohoFormCompactModule,
            SohoHeaderModule,
            SohoHierarchyModule,
            SohoHomePageModule,
            SohoHyperlinkModule,
            SohoIconModule,
            SohoInputModule,
            SohoInputValidateModule,
            SohoLabelModule,
            SohoLineModule,
            SohoListViewModule,
            SohoListBuilderModule,
            SohoLocaleModule,
            SohoLookupModule,
            SohoMaskModule,
            SohoMastheadModule,
            SohoMenuButtonModule,
            SohoMessageModule,
            SohoModalDialogModule,
            SohoModuleNavModule,
            SohoModuleNavContainerModule,
            SohoModuleNavSettingsModule,
            SohoModuleNavSwitcherModule,
            SohoMonthViewModule,
            SohoNotificationModule,
            SohoNotificationBadgeModule,
            SohoPagerModule,
            SohoPersonalizeModule,
            SohoPieModule,
            SohoPopDownModule,
            SohoPopupMenuModule,
            SohoProgressModule,
            SohoRadarModule,
            SohoRadioButtonModule,
            SohoRatingModule,
            SohoRenderLoopModule,
            SohoSearchFieldModule,
            SohoSparklineModule,
            SohoSplitterModule,
            SohoSliderModule,
            SohoSpinboxModule,
            SohoStepChartModule,
            SohoStepProcessModule,
            SohoSwapListModule,
            SohoSwipeActionModule,
            SohoTabsModule,
            SohoTagModule,
            SohoTextAreaModule,
            SohoTimePickerModule,
            SohoToastModule,
            SohoToolbarModule,
            SohoToolbarFlexModule,
            SohoTooltipModule,
            SohoTrackDirtyModule,
            SohoTreeModule,
            SohoTreemapModule,
            SohoWeekViewModule,
            SohoWizardModule], exports: [SohoAboutModule,
            SohoAccordionModule,
            SohoActionsheetModule,
            SohoAlertModule,
            SohoApplicationMenuModule,
            SohoArrangeModule,
            SohoAutoCompleteModule,
            SohoBarModule,
            SohoBlockGridModule,
            SohoBreadcrumbModule,
            SohoBulletModule,
            SohoBusyIndicatorModule,
            SohoButtonModule,
            SohoButtonsetModule,
            SohoCalendarModule,
            SohoCardModule,
            SohoChartModule,
            SohoCheckBoxModule,
            SohoCirclepagerModule,
            SohoColorPickerModule,
            SohoColumnModule,
            SohoContextMenuModule,
            SohoContextualActionPanelModule,
            SohoDataGridModule,
            SohoDatePickerModule,
            SohoDragModule,
            SohoDropDownModule,
            SohoEditorModule,
            SohoEmptyMessageModule,
            SohoErrorModule,
            SohoExpandableAreaModule,
            SohoFieldFilterModule,
            SohoFieldOptionsModule,
            SohoFileUploadAdvancedModule,
            SohoFileUploadModule,
            SohoFormCompactModule,
            SohoHeaderModule,
            SohoHierarchyModule,
            SohoHomePageModule,
            SohoHyperlinkModule,
            SohoIconModule,
            SohoInputModule,
            SohoInputValidateModule,
            SohoLabelModule,
            SohoLineModule,
            SohoListViewModule,
            SohoListBuilderModule,
            SohoLocaleModule,
            SohoLookupModule,
            SohoMaskModule,
            SohoMastheadModule,
            SohoMenuButtonModule,
            SohoModalDialogModule,
            SohoModuleNavModule,
            SohoModuleNavContainerModule,
            SohoModuleNavSettingsModule,
            SohoModuleNavSwitcherModule,
            SohoMonthViewModule,
            SohoNotificationModule,
            SohoNotificationBadgeModule,
            SohoPagerModule,
            SohoPersonalizeModule,
            SohoPieModule,
            SohoPopDownModule,
            SohoPopupMenuModule,
            SohoProgressModule,
            SohoRadarModule,
            SohoRadioButtonModule,
            SohoRatingModule,
            SohoRenderLoopModule,
            SohoSearchFieldModule,
            SohoSliderModule,
            SohoSparklineModule,
            SohoSplitterModule,
            SohoSpinboxModule,
            SohoStepChartModule,
            SohoStepProcessModule,
            SohoSwapListModule,
            SohoSwipeActionModule,
            SohoTabsModule,
            SohoTagModule,
            SohoTextAreaModule,
            SohoTimePickerModule,
            SohoToastModule,
            SohoToolbarModule,
            SohoToolbarFlexModule,
            SohoTooltipModule,
            SohoTrackDirtyModule,
            SohoTreeModule,
            SohoTreemapModule,
            SohoVersionInitializerModule,
            SohoWeekViewModule,
            SohoWizardModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoComponentsModule, imports: [SohoAboutModule,
            SohoAccordionModule,
            SohoActionsheetModule,
            SohoAlertModule,
            SohoApplicationMenuModule,
            SohoArrangeModule,
            SohoAutoCompleteModule,
            SohoBarModule,
            SohoBlockGridModule,
            SohoBreadcrumbModule,
            SohoBulletModule,
            SohoBusyIndicatorModule,
            SohoButtonModule,
            SohoButtonsetModule,
            SohoCalendarModule,
            SohoChartModule,
            SohoCheckBoxModule,
            SohoCardModule,
            SohoCirclepagerModule,
            SohoColorPickerModule,
            SohoColumnModule,
            SohoContextualActionPanelModule,
            SohoContextMenuModule,
            SohoDataGridModule,
            SohoDatePickerModule,
            SohoDragModule,
            SohoDropDownModule,
            SohoEditorModule,
            SohoEmptyMessageModule,
            SohoErrorModule,
            SohoExpandableAreaModule,
            SohoFieldFilterModule,
            SohoFieldOptionsModule,
            SohoFileUploadAdvancedModule,
            SohoFileUploadModule,
            SohoFormCompactModule,
            SohoHeaderModule,
            SohoHierarchyModule,
            SohoHomePageModule,
            SohoHyperlinkModule,
            SohoIconModule,
            SohoInputModule,
            SohoInputValidateModule,
            SohoLabelModule,
            SohoLineModule,
            SohoListViewModule,
            SohoListBuilderModule,
            SohoLocaleModule,
            SohoLookupModule,
            SohoMaskModule,
            SohoMastheadModule,
            SohoMenuButtonModule,
            SohoMessageModule,
            SohoModalDialogModule,
            SohoModuleNavModule,
            SohoModuleNavContainerModule,
            SohoModuleNavSettingsModule,
            SohoModuleNavSwitcherModule,
            SohoMonthViewModule,
            SohoNotificationModule,
            SohoNotificationBadgeModule,
            SohoPagerModule,
            SohoPersonalizeModule,
            SohoPieModule,
            SohoPopDownModule,
            SohoPopupMenuModule,
            SohoProgressModule,
            SohoRadarModule,
            SohoRadioButtonModule,
            SohoRatingModule,
            SohoRenderLoopModule,
            SohoSearchFieldModule,
            SohoSparklineModule,
            SohoSplitterModule,
            SohoSliderModule,
            SohoSpinboxModule,
            SohoStepChartModule,
            SohoStepProcessModule,
            SohoSwapListModule,
            SohoSwipeActionModule,
            SohoTabsModule,
            SohoTagModule,
            SohoTextAreaModule,
            SohoTimePickerModule,
            SohoToastModule,
            SohoToolbarModule,
            SohoToolbarFlexModule,
            SohoTooltipModule,
            SohoTrackDirtyModule,
            SohoTreeModule,
            SohoTreemapModule,
            SohoWeekViewModule,
            SohoWizardModule, SohoAboutModule,
            SohoAccordionModule,
            SohoActionsheetModule,
            SohoAlertModule,
            SohoApplicationMenuModule,
            SohoArrangeModule,
            SohoAutoCompleteModule,
            SohoBarModule,
            SohoBlockGridModule,
            SohoBreadcrumbModule,
            SohoBulletModule,
            SohoBusyIndicatorModule,
            SohoButtonModule,
            SohoButtonsetModule,
            SohoCalendarModule,
            SohoCardModule,
            SohoChartModule,
            SohoCheckBoxModule,
            SohoCirclepagerModule,
            SohoColorPickerModule,
            SohoColumnModule,
            SohoContextMenuModule,
            SohoContextualActionPanelModule,
            SohoDataGridModule,
            SohoDatePickerModule,
            SohoDragModule,
            SohoDropDownModule,
            SohoEditorModule,
            SohoEmptyMessageModule,
            SohoErrorModule,
            SohoExpandableAreaModule,
            SohoFieldFilterModule,
            SohoFieldOptionsModule,
            SohoFileUploadAdvancedModule,
            SohoFileUploadModule,
            SohoFormCompactModule,
            SohoHeaderModule,
            SohoHierarchyModule,
            SohoHomePageModule,
            SohoHyperlinkModule,
            SohoIconModule,
            SohoInputModule,
            SohoInputValidateModule,
            SohoLabelModule,
            SohoLineModule,
            SohoListViewModule,
            SohoListBuilderModule,
            SohoLocaleModule,
            SohoLookupModule,
            SohoMaskModule,
            SohoMastheadModule,
            SohoMenuButtonModule,
            SohoModalDialogModule,
            SohoModuleNavModule,
            SohoModuleNavContainerModule,
            SohoModuleNavSettingsModule,
            SohoModuleNavSwitcherModule,
            SohoMonthViewModule,
            SohoNotificationModule,
            SohoNotificationBadgeModule,
            SohoPagerModule,
            SohoPersonalizeModule,
            SohoPieModule,
            SohoPopDownModule,
            SohoPopupMenuModule,
            SohoProgressModule,
            SohoRadarModule,
            SohoRadioButtonModule,
            SohoRatingModule,
            SohoRenderLoopModule,
            SohoSearchFieldModule,
            SohoSliderModule,
            SohoSparklineModule,
            SohoSplitterModule,
            SohoSpinboxModule,
            SohoStepChartModule,
            SohoStepProcessModule,
            SohoSwapListModule,
            SohoSwipeActionModule,
            SohoTabsModule,
            SohoTagModule,
            SohoTextAreaModule,
            SohoTimePickerModule,
            SohoToastModule,
            SohoToolbarModule,
            SohoToolbarFlexModule,
            SohoTooltipModule,
            SohoTrackDirtyModule,
            SohoTreeModule,
            SohoTreemapModule,
            SohoVersionInitializerModule,
            SohoWeekViewModule,
            SohoWizardModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        SohoAboutModule,
                        SohoAccordionModule,
                        SohoActionsheetModule,
                        SohoAlertModule,
                        SohoApplicationMenuModule,
                        SohoArrangeModule,
                        SohoAutoCompleteModule,
                        SohoBarModule,
                        SohoBlockGridModule,
                        SohoBreadcrumbModule,
                        SohoBulletModule,
                        SohoBusyIndicatorModule,
                        SohoButtonModule,
                        SohoButtonsetModule,
                        SohoCalendarModule,
                        SohoChartModule,
                        SohoCheckBoxModule,
                        SohoCardModule,
                        SohoCirclepagerModule,
                        SohoColorPickerModule,
                        SohoColumnModule,
                        SohoContextualActionPanelModule,
                        SohoContextMenuModule,
                        SohoDataGridModule,
                        SohoDatePickerModule,
                        SohoDragModule,
                        SohoDropDownModule,
                        SohoEditorModule,
                        SohoEmptyMessageModule,
                        SohoErrorModule,
                        SohoExpandableAreaModule,
                        SohoFieldFilterModule,
                        SohoFieldOptionsModule,
                        SohoFileUploadAdvancedModule,
                        SohoFileUploadModule,
                        SohoFormCompactModule,
                        SohoHeaderModule,
                        SohoHierarchyModule,
                        SohoHomePageModule,
                        SohoHyperlinkModule,
                        SohoIconModule,
                        SohoInputModule,
                        SohoInputValidateModule,
                        SohoLabelModule,
                        SohoLineModule,
                        SohoListViewModule,
                        SohoListBuilderModule,
                        SohoLocaleModule,
                        SohoLookupModule,
                        SohoMaskModule,
                        SohoMastheadModule,
                        SohoMenuButtonModule,
                        SohoMessageModule,
                        SohoModalDialogModule,
                        SohoModuleNavModule,
                        SohoModuleNavContainerModule,
                        SohoModuleNavSettingsModule,
                        SohoModuleNavSwitcherModule,
                        SohoMonthViewModule,
                        SohoNotificationModule,
                        SohoNotificationBadgeModule,
                        SohoPagerModule,
                        SohoPersonalizeModule,
                        SohoPieModule,
                        SohoPopDownModule,
                        SohoPopupMenuModule,
                        SohoProgressModule,
                        SohoRadarModule,
                        SohoRadioButtonModule,
                        SohoRatingModule,
                        SohoRenderLoopModule,
                        SohoSearchFieldModule,
                        SohoSparklineModule,
                        SohoSplitterModule,
                        SohoSliderModule,
                        SohoSpinboxModule,
                        SohoStepChartModule,
                        SohoStepProcessModule,
                        SohoSwapListModule,
                        SohoSwipeActionModule,
                        SohoTabsModule,
                        SohoTagModule,
                        SohoTextAreaModule,
                        SohoTimePickerModule,
                        SohoToastModule,
                        SohoToolbarModule,
                        SohoToolbarFlexModule,
                        SohoTooltipModule,
                        SohoTrackDirtyModule,
                        SohoTreeModule,
                        SohoTreemapModule,
                        SohoWeekViewModule,
                        SohoWizardModule
                    ],
                    declarations: [],
                    exports: [
                        SohoAboutModule,
                        SohoAccordionModule,
                        SohoActionsheetModule,
                        SohoAlertModule,
                        SohoApplicationMenuModule,
                        SohoArrangeModule,
                        SohoAutoCompleteModule,
                        SohoBarModule,
                        SohoBlockGridModule,
                        SohoBreadcrumbModule,
                        SohoBulletModule,
                        SohoBusyIndicatorModule,
                        SohoButtonModule,
                        SohoButtonsetModule,
                        SohoCalendarModule,
                        SohoCardModule,
                        SohoChartModule,
                        SohoCheckBoxModule,
                        SohoCirclepagerModule,
                        SohoColorPickerModule,
                        SohoColumnModule,
                        SohoContextMenuModule,
                        SohoContextualActionPanelModule,
                        SohoDataGridModule,
                        SohoDatePickerModule,
                        SohoDragModule,
                        SohoDropDownModule,
                        SohoEditorModule,
                        SohoEmptyMessageModule,
                        SohoErrorModule,
                        SohoExpandableAreaModule,
                        SohoFieldFilterModule,
                        SohoFieldOptionsModule,
                        SohoFileUploadAdvancedModule,
                        SohoFileUploadModule,
                        SohoFormCompactModule,
                        SohoHeaderModule,
                        SohoHierarchyModule,
                        SohoHomePageModule,
                        SohoHyperlinkModule,
                        SohoIconModule,
                        SohoInputModule,
                        SohoInputValidateModule,
                        SohoLabelModule,
                        SohoLineModule,
                        SohoListViewModule,
                        SohoListBuilderModule,
                        SohoLocaleModule,
                        SohoLookupModule,
                        SohoMaskModule,
                        SohoMastheadModule,
                        SohoMenuButtonModule,
                        SohoModalDialogModule,
                        SohoModuleNavModule,
                        SohoModuleNavContainerModule,
                        SohoModuleNavSettingsModule,
                        SohoModuleNavSwitcherModule,
                        SohoMonthViewModule,
                        SohoNotificationModule,
                        SohoNotificationBadgeModule,
                        SohoPagerModule,
                        SohoPersonalizeModule,
                        SohoPieModule,
                        SohoPopDownModule,
                        SohoPopupMenuModule,
                        SohoProgressModule,
                        SohoRadarModule,
                        SohoRadioButtonModule,
                        SohoRatingModule,
                        SohoRenderLoopModule,
                        SohoSearchFieldModule,
                        SohoSliderModule,
                        SohoSparklineModule,
                        SohoSplitterModule,
                        SohoSpinboxModule,
                        SohoStepChartModule,
                        SohoStepProcessModule,
                        SohoSwapListModule,
                        SohoSwipeActionModule,
                        SohoTabsModule,
                        SohoTagModule,
                        SohoTextAreaModule,
                        SohoTimePickerModule,
                        SohoToastModule,
                        SohoToolbarModule,
                        SohoToolbarFlexModule,
                        SohoTooltipModule,
                        SohoTrackDirtyModule,
                        SohoTreeModule,
                        SohoTreemapModule,
                        SohoVersionInitializerModule,
                        SohoWeekViewModule,
                        SohoWizardModule
                    ]
                }]
        }] });
