// ids-enterprise-ng/esm2022/lib/emptymessage/soho-emptymessage.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoEmptyMessageDirective } from './soho-emptymessage.directive';
import * as i0 from "@angular/core";
export class SohoEmptyMessageModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoEmptyMessageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoEmptyMessageModule, declarations: [SohoEmptyMessageDirective], imports: [CommonModule], exports: [SohoEmptyMessageDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoEmptyMessageModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoEmptyMessageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoEmptyMessageDirective
                    ],
                    exports: [
                        SohoEmptyMessageDirective
                    ]
                }]
        }] });
