// ids-enterprise-ng/esm2022/lib/popupmenu/soho-popupmenu.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoPopupMenuComponent, SohoPopupMenuHeadingComponent, SohoPopupMenuShortCutTextComponent, SohoPopupMenuItemComponent, SohoPopupMenuItemLabelComponent, SohoPopupMenuSeparatorComponent } from './soho-popupmenu.component';
import * as i0 from "@angular/core";
export class SohoPopupMenuModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopupMenuModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoPopupMenuModule, declarations: [SohoPopupMenuComponent,
            SohoPopupMenuHeadingComponent,
            SohoPopupMenuShortCutTextComponent,
            SohoPopupMenuItemComponent,
            SohoPopupMenuItemLabelComponent,
            SohoPopupMenuSeparatorComponent], imports: [CommonModule], exports: [SohoPopupMenuComponent,
            SohoPopupMenuHeadingComponent,
            SohoPopupMenuShortCutTextComponent,
            SohoPopupMenuItemComponent,
            SohoPopupMenuItemLabelComponent,
            SohoPopupMenuSeparatorComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopupMenuModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoPopupMenuModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoPopupMenuComponent,
                        SohoPopupMenuHeadingComponent,
                        SohoPopupMenuShortCutTextComponent,
                        SohoPopupMenuItemComponent,
                        SohoPopupMenuItemLabelComponent,
                        SohoPopupMenuSeparatorComponent
                    ],
                    exports: [
                        SohoPopupMenuComponent,
                        SohoPopupMenuHeadingComponent,
                        SohoPopupMenuShortCutTextComponent,
                        SohoPopupMenuItemComponent,
                        SohoPopupMenuItemLabelComponent,
                        SohoPopupMenuSeparatorComponent
                    ]
                }]
        }] });
