// ids-enterprise-ng/esm2022/lib/hyperlink/soho-hyperlink.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoHyperlinkComponent } from './soho-hyperlink.component';
import { SohoIconModule } from '../icon/soho-icon.module';
import * as i0 from "@angular/core";
export class SohoHyperlinkModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkModule, declarations: [SohoHyperlinkComponent], imports: [CommonModule,
            SohoIconModule], exports: [SohoHyperlinkComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkModule, imports: [CommonModule,
            SohoIconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoIconModule
                    ],
                    declarations: [
                        SohoHyperlinkComponent
                    ],
                    exports: [
                        SohoHyperlinkComponent
                    ]
                }]
        }] });
