// ids-enterprise-ng/esm2022/lib/hyperlink/soho-hyperlink.component.mjs
import { Component, Output, HostBinding, Input, ChangeDetectionStrategy } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../icon/soho-icon.component";
export class SohoHyperlinkComponent {
    static { this.SHOWVISITED = 'show-visited'; }
    static { this.DIRECTIONAL = 'forward-caret'; }
    static { this.BACK = 'back-caret'; }
    /**
     * The type of the hyperlink, defaulting to 'hide-focus'.
     *
     * @todo replace override of native attribute
     */
    // eslint-disable-next-line @angular-eslint/no-input-rename
    set sohoHyperlink(type) {
        if (type) {
            this.hyperlinkType = type;
        }
    }
    get showVisitedClass() {
        if (this.hyperlinkType === SohoHyperlinkComponent.SHOWVISITED) {
            return true;
        }
    }
    get directionalClass() {
        if (this.hyperlinkType === SohoHyperlinkComponent.DIRECTIONAL) {
            return true;
        }
    }
    get backClass() {
        if (this.hyperlinkType === SohoHyperlinkComponent.BACK) {
            return true;
        }
    }
    constructor(element) {
        this.element = element;
        this.hyperLinkClass = true;
    }
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement);
        /*
         * hideFocus() is the plugin used for hyperlink
         * Initialize hideFocus()
         */
        this.jQueryElement.hideFocus();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoHyperlinkComponent, selector: "a[soho-hyperlink]", inputs: { sohoHyperlink: ["soho-hyperlink", "sohoHyperlink"], icon: "icon", href: "href", disabled: "disabled", isCaretRight: "isCaretRight", isCaretLeft: "isCaretLeft" }, outputs: { change: "change" }, host: { properties: { "class.hyperlink": "this.hyperLinkClass", "class.show-visited": "this.showVisitedClass", "class.directional": "this.directionalClass", "class.back": "this.backClass", "attr.href": "this.href", "attr.disabled": "this.disabled", "attr.isCaretRight": "this.isCaretRight", "attr.isCaretLeft": "this.isCaretLeft" } }, ngImport: i0, template: ` <svg soho-icon *ngIf="isCaretLeft" [icon]="'caret-left'"></svg>
              <ng-content></ng-content>
              <svg soho-icon *ngIf="isCaretRight" [icon]="'caret-right'"></svg>
              `, isInline: true, dependencies: [{ kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i2.SohoIconComponent, selector: "svg[soho-icon]", inputs: ["isEmptyState", "extraIconClass", "alert", "icon"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHyperlinkComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'a[soho-hyperlink]',
                    template: ` <svg soho-icon *ngIf="isCaretLeft" [icon]="'caret-left'"></svg>
              <ng-content></ng-content>
              <svg soho-icon *ngIf="isCaretRight" [icon]="'caret-right'"></svg>
              `,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { sohoHyperlink: [{
                type: Input,
                args: ['soho-hyperlink']
            }], icon: [{
                type: Input
            }], change: [{
                type: Output
            }], hyperLinkClass: [{
                type: HostBinding,
                args: ['class.hyperlink']
            }], showVisitedClass: [{
                type: HostBinding,
                args: ['class.show-visited']
            }], directionalClass: [{
                type: HostBinding,
                args: ['class.directional']
            }], backClass: [{
                type: HostBinding,
                args: ['class.back']
            }], href: [{
                type: HostBinding,
                args: ['attr.href']
            }, {
                type: Input
            }], disabled: [{
                type: HostBinding,
                args: ['attr.disabled']
            }, {
                type: Input
            }], isCaretRight: [{
                type: HostBinding,
                args: ['attr.isCaretRight']
            }, {
                type: Input
            }], isCaretLeft: [{
                type: HostBinding,
                args: ['attr.isCaretLeft']
            }, {
                type: Input
            }] } });
