// ids-enterprise-ng/esm2022/lib/stepprocess/soho-stepprocess.component.mjs
import { Component, EventEmitter, HostBinding, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../button/soho-button.component";
import * as i2 from "../toolbar/soho-toolbar.component";
/**************************************************************
 * STEP LIST TITLE
 **************************************************************/
export class SohoStepListTitleComponent {
    get title() {
        return true;
    }
    get titleWide() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepListTitleComponent, selector: "div[soho-step-list-title]", host: { properties: { "class.title": "this.title", "class.title-wide": "this.titleWide" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-step-list-title]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { title: [{
                type: HostBinding,
                args: ['class.title']
            }], titleWide: [{
                type: HostBinding,
                args: ['class.title-wide']
            }] } });
/**************************************************************
 * STEP LIST
 **************************************************************/
export class SohoStepListComponent {
    constructor() {
        this.isTree = true;
        this.isJSStepTreeScroll = true;
        // -----------------------------------------------------------------
        // required by stepcontrol. make sure this id and the one passed
        // into the stepcontrol settings match.
        // -----------------------------------------------------------------
        this.id = 'step-list';
        // -----------------------------------------------------------------
        // disallow jquery.initialize() from initializing this as a tree
        // control. Needed due to desire for tree styles but not an actual
        // tree itself.
        // -----------------------------------------------------------------
        this.dataInit = false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepListComponent, selector: "ul[soho-step-list]", host: { properties: { "class.tree": "this.isTree", "class.js-step-list-scroll": "this.isJSStepTreeScroll", "attr.id": "this.id", "attr.data-init": "this.dataInit" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'ul[soho-step-list]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isTree: [{
                type: HostBinding,
                args: ['class.tree']
            }], isJSStepTreeScroll: [{
                type: HostBinding,
                args: ['class.js-step-list-scroll']
            }], id: [{
                type: HostBinding,
                args: ['attr.id']
            }], dataInit: [{
                type: HostBinding,
                args: ['attr.data-init']
            }] } });
/**************************************************************
 * SUB STEP LIST
 **************************************************************/
export class SohoSubstepListComponent {
    constructor() {
        this.isFolder = true;
        this.isJSStepFolder = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSubstepListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoSubstepListComponent, selector: "ul[soho-substep-list]", host: { properties: { "class.folder": "this.isFolder", "class.js-step-folder": "this.isJSStepFolder" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoSubstepListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'ul[soho-substep-list]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isFolder: [{
                type: HostBinding,
                args: ['class.folder']
            }], isJSStepFolder: [{
                type: HostBinding,
                args: ['class.js-step-folder']
            }] } });
/**************************************************************
 * STEP LIST ITEM
 **************************************************************/
export class SohoStepListItemComponent {
    constructor() {
        this.isSelected = false;
    }
    get jsStep() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepListItemComponent, selector: "li[soho-step-list-item]", inputs: { isSelected: "isSelected" }, host: { properties: { "class.js-step": "this.jsStep", "class.is-selected": "this.isSelected" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'li[soho-step-list-item]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { jsStep: [{
                type: HostBinding,
                args: ['class.js-step']
            }], isSelected: [{
                type: HostBinding,
                args: ['class.is-selected']
            }, {
                type: Input
            }] } });
/**************************************************************
 * STEP LIST ITEM ANCHOR
 **************************************************************/
export class SohoStepListItemAnchorComponent {
    constructor() {
        this.isJsStepLink = true;
    }
    get hrefAttr() {
        return this.stepId ? '#' + this.stepId : null;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemAnchorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepListItemAnchorComponent, selector: "a[soho-step-list-item-anchor]", inputs: { stepId: "stepId" }, host: { properties: { "class.js-step-link": "this.isJsStepLink", "attr.href": "this.hrefAttr" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemAnchorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'a[soho-step-list-item-anchor]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isJsStepLink: [{
                type: HostBinding,
                args: ['class.js-step-link']
            }], hrefAttr: [{
                type: HostBinding,
                args: ['attr.href']
            }], stepId: [{
                type: Input
            }] } });
/**************************************************************
 * STEP LIST ITEM TITLE
 **************************************************************/
export class SohoStepListItemTitleComponent {
    constructor() {
        this.treeText = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepListItemTitleComponent, selector: "span[soho-step-list-item-title]", host: { properties: { "class.tree-text": "this.treeText" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepListItemTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'span[soho-step-list-item-title]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { treeText: [{
                type: HostBinding,
                args: ['class.tree-text']
            }] } });
/**************************************************************
 * STEP CONTENT TITLE
 **************************************************************/
export class SohoStepContentTitleComponent {
    constructor() {
        this.isHeading = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepContentTitleComponent, selector: "div[soho-step-content-title]", host: { properties: { "class.heading": "this.isHeading" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-step-content-title]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isHeading: [{
                type: HostBinding,
                args: ['class.heading']
            }] } });
/**************************************************************
 * STEP CONTENT
 **************************************************************/
export class SohoStepContentComponent {
    get isScrollable() {
        return true;
    }
    get iStepContainer() {
        return true;
    }
    get isJsStepPanelsScroll() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepContentComponent, selector: "div[soho-step-content]", host: { properties: { "class.scrollable": "this.isScrollable", "class.step-container": "this.iStepContainer", "class.js-step-container-scroll": "this.isJsStepPanelsScroll" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-step-content]',
                    template: `<ng-content></ng-content>`,
                }]
        }], propDecorators: { isScrollable: [{
                type: HostBinding,
                args: ['class.scrollable']
            }], iStepContainer: [{
                type: HostBinding,
                args: ['class.step-container']
            }], isJsStepPanelsScroll: [{
                type: HostBinding,
                args: ['class.js-step-container-scroll']
            }] } });
/**************************************************************
 * STEP CONTENT PANEL
 **************************************************************/
export class SohoStepContentPanelComponent {
    get isJsStepProcessPanel() {
        return true;
    }
    get idAttr() {
        return this.stepId;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentPanelComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepContentPanelComponent, selector: "div[soho-step-content-panel]", inputs: { stepId: "stepId" }, host: { properties: { "class.js-step-panel": "this.isJsStepProcessPanel", "attr.id": "this.idAttr" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepContentPanelComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-step-content-panel]',
                    template: `<ng-content></ng-content>`,
                }]
        }], propDecorators: { isJsStepProcessPanel: [{
                type: HostBinding,
                args: ['class.js-step-panel']
            }], idAttr: [{
                type: HostBinding,
                args: ['attr.id']
            }], stepId: [{
                type: Input
            }] } });
/**************************************************************
 * MAIN STEP PROCESS COMPONENT
 **************************************************************/
export class SohoStepProcessComponent {
    // ------------------------------------------------------------------------
    // @Inputs
    // ------------------------------------------------------------------------
    /**
     * Whether or not to restrict navigation to next and previous buttons only.
     */
    set linearProgression(linearProgression) {
        this.stepProcessOptions.linearProgression = linearProgression;
        if (this.jQueryElement) {
            this.stepprocess.settings.linearProgression = linearProgression;
        }
    }
    set nextButtonEnable(enabled) {
        if (this.jQueryElement) {
            this.jQueryElement.find('.js-step-link-next').prop('disabled', !enabled);
        }
    }
    set previousButtonEnable(enabled) {
        if (this.jQueryElement) {
            this.jQueryElement.find('.js-step-link-prev').prop('disabled', !enabled);
        }
    }
    set saveCloseButtonEnable(enabled) {
        if (this.jQueryElement) {
            this.jQueryElement.find('.js-btn-save-changes').prop('disabled', !enabled);
        }
    }
    constructor(element) {
        this.element = element;
        this.isStepProcessContainer = true;
        this.isTwoColumn = true;
        this.isFixed = true;
        this.isPageContainer = true;
        this.isNoScroll = true;
        this.nextButtonLabel = 'Next';
        this.previousButtonLabel = 'Previous';
        this.continueButtonLabel = 'Continue';
        this.saveAndCloseButtonLabel = 'Save & Close';
        // ------------------------------------------------------------------------
        // @Outputs
        // ------------------------------------------------------------------------
        this.beforeSelectStep = new EventEmitter();
        this.saveClose = new EventEmitter();
        // The internal stepsOptions object used to construct the stepproces control
        this.stepProcessOptions = {};
        // beforeStepSelect callback that is adapted to an Angular EventEmittier
        this.beforeSelectStepDeferred = $.Deferred();
        this.beforeSelectStepPromise = (args) => {
            this.beforeSelectStepDeferred = $.Deferred();
            if (this.beforeSelectStep.observers.length > 0) {
                // not sure what it would mean to have a multiple observers potentially
                // calling the response function.
                if (this.beforeSelectStep.observers.length !== 1) {
                    console.warn('stepprocess: only 1 observer is allowed for the beforeSelectStep event emitter');
                }
                const beforeSelectStepEvent = {};
                // ------------------------------------------------------------------------------
                // The first beforeSelectStep is called before the this.stepprocess can be set.
                // In that case there is no currentSelectedStep so getting teh currentStepId
                // can be skipped.
                // ------------------------------------------------------------------------------
                if (this.stepprocess) {
                    const $selectedStep = this.stepprocess.getSelectedStep();
                    beforeSelectStepEvent.currentStepId = $selectedStep?.children('a')?.attr('href')?.substring(1);
                }
                if (args.stepLink) {
                    beforeSelectStepEvent.targetStepId = $(args.stepLink)?.attr('href')?.substring(1);
                }
                else {
                    beforeSelectStepEvent.targetStepId = beforeSelectStepEvent.currentStepId;
                }
                beforeSelectStepEvent.isStepping = args.isStepping;
                beforeSelectStepEvent.response = this.beforeSelectStepResponse;
                this.beforeSelectStep.emit(beforeSelectStepEvent);
            }
            else {
                this.beforeSelectStepDeferred.resolve(true);
            }
            return this.beforeSelectStepDeferred.promise();
        };
        this.beforeSelectStepResponse = (response) => {
            if (response.overrideTargetStepId) {
                const stepLinkToSelect = $('.js-step-link[href="#' + response.overrideTargetStepId + '"]');
                this.beforeSelectStepDeferred.resolve(response.continue, stepLinkToSelect);
            }
            else {
                this.beforeSelectStepDeferred.resolve(response.continue);
            }
        };
    }
    ngAfterViewInit() {
        this.stepProcessOptions.stepList = '#step-list';
        this.stepProcessOptions.beforeSelectStep = this.beforeSelectStepPromise;
        this.jQueryElement = jQuery(this.element.nativeElement);
        this.jQueryElement.stepprocess(this.stepProcessOptions);
        this.stepprocess = this.jQueryElement.data('stepprocess');
        this.jQueryElement.find('.js-btn-save-changes').
            on('click', (_e) => {
            this.fireOnSaveClose();
        });
    }
    fireOnSaveClose() {
        const $selectedStep = this.stepprocess?.getSelectedStep();
        const currentStepId = $selectedStep?.children('a')?.attr('href')?.substring(1);
        const event = {
            currentStepId
        };
        this.saveClose.emit(event);
    }
    ngOnDestroy() {
        if (this.stepprocess) {
            this.stepprocess.destroy();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoStepProcessComponent, selector: "div[soho-stepprocess]", inputs: { linearProgression: "linearProgression", nextButtonLabel: "nextButtonLabel", previousButtonLabel: "previousButtonLabel", continueButtonLabel: "continueButtonLabel", saveAndCloseButtonLabel: "saveAndCloseButtonLabel", nextButtonEnable: "nextButtonEnable", previousButtonEnable: "previousButtonEnable", saveCloseButtonEnable: "saveCloseButtonEnable" }, outputs: { beforeSelectStep: "beforeSelectStep", saveClose: "saveClose" }, host: { properties: { "class.step-process-container": "this.isStepProcessContainer", "class.two-column": "this.isTwoColumn", "class.fixed": "this.isFixed", "class.page-container": "this.isPageContainer", "class.no-scroll": "this.isNoScroll" } }, ngImport: i0, template: "\n<!-- STEP CONTROL STEPS -->\n<div class=\"sidebar\">\n  <div soho-toolbar class=\"toolbar-custom toolbar-collapsible phone-visible\">\n    <soho-toolbar-title>\n      <ng-content select=\"div[soho-step-list-title]\"></ng-content>\n    </soho-toolbar-title>\n  </div>\n  <ng-content select=\"ul[soho-step-list]\"></ng-content>\n  <div class=\"phone-action-bar phone-visible\">\n    <button soho-button=\"secondary\" class=\"btn btn-secondary js-btn-save-changes\">{{saveAndCloseButtonLabel}}</button>\n    <button soho-button=\"primary\" class=\"btn btn-primary js-toggle-sidebar\">{{continueButtonLabel}}</button>\n  </div>\n</div>\n\n<!-- STEP CONTROL CONTENT -->\n<div class=\"main no-scroll\">\n\n  <div class=\"toolbar-custom toolbar-collapsible\">\n    <button soho-button=\"icon\" icon=\"bullet-steps\" class=\"btn-icon btn-toggle-steps js-toggle-sidebar\"></button>\n\n    <ng-content select=\"div[soho-step-content-title]\"></ng-content>\n\n    <div class=\"actions phone-hidden\">\n      <button soho-button=\"tertiary\" class=\"btn-tertiary js-step-link-prev\">{{previousButtonLabel}}</button>\n      <button soho-button=\"tertiary\" class=\"btn-tertiary js-step-link-next\">{{nextButtonLabel}}</button>\n    </div>\n  </div>\n\n  <!-- STEP CONTROL PHONE CONTROLS -->\n  <ng-content select=\"div[soho-step-content]\"></ng-content>\n    <div class=\"phone-action-bar phone-visible\">\n      <button soho-button=\"secondary\" class=\"btn btn-secondary js-btn-save-changes\">{{saveAndCloseButtonLabel}}</button>\n      <button soho-button=\"primary\" class=\"btn btn-primary js-step-link-next\">{{nextButtonLabel}}</button>\n    </div>\n</div>\n", dependencies: [{ kind: "component", type: i1.SohoButtonComponent, selector: "button[soho-button]", inputs: ["soho-button", "buttonOptions", "toggleOnIcon", "toggleOffIcon", "replaceText", "hideMenuArrow", "hitbox", "notificationBadge", "notificationBadgeOptions", "extraIconClass", "isToggle", "isTogglePressed", "icon", "iconAlign", "isSubmit", "ripple", "state", "toggle", "expandableExpander", "moveToLeft", "moveToRight", "moveToSelected"] }, { kind: "component", type: i2.SohoToolbarComponent, selector: "soho-toolbar, div[soho-toolbar]", inputs: ["noActionsButton", "hasMoreButton", "maxVisibleButtons", "rightAligned", "rightAlign", "resizeContainers", "favorButtonset", "moreMenuSettings", "attributes"], outputs: ["beforeActivated", "activated", "afterActivated", "selected", "menuItemMouseOver"] }, { kind: "component", type: i2.SohoToolbarTitleComponent, selector: "soho-toolbar-title" }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessComponent, decorators: [{
            type: Component,
            args: [{ selector: 'div[soho-stepprocess]', template: "\n<!-- STEP CONTROL STEPS -->\n<div class=\"sidebar\">\n  <div soho-toolbar class=\"toolbar-custom toolbar-collapsible phone-visible\">\n    <soho-toolbar-title>\n      <ng-content select=\"div[soho-step-list-title]\"></ng-content>\n    </soho-toolbar-title>\n  </div>\n  <ng-content select=\"ul[soho-step-list]\"></ng-content>\n  <div class=\"phone-action-bar phone-visible\">\n    <button soho-button=\"secondary\" class=\"btn btn-secondary js-btn-save-changes\">{{saveAndCloseButtonLabel}}</button>\n    <button soho-button=\"primary\" class=\"btn btn-primary js-toggle-sidebar\">{{continueButtonLabel}}</button>\n  </div>\n</div>\n\n<!-- STEP CONTROL CONTENT -->\n<div class=\"main no-scroll\">\n\n  <div class=\"toolbar-custom toolbar-collapsible\">\n    <button soho-button=\"icon\" icon=\"bullet-steps\" class=\"btn-icon btn-toggle-steps js-toggle-sidebar\"></button>\n\n    <ng-content select=\"div[soho-step-content-title]\"></ng-content>\n\n    <div class=\"actions phone-hidden\">\n      <button soho-button=\"tertiary\" class=\"btn-tertiary js-step-link-prev\">{{previousButtonLabel}}</button>\n      <button soho-button=\"tertiary\" class=\"btn-tertiary js-step-link-next\">{{nextButtonLabel}}</button>\n    </div>\n  </div>\n\n  <!-- STEP CONTROL PHONE CONTROLS -->\n  <ng-content select=\"div[soho-step-content]\"></ng-content>\n    <div class=\"phone-action-bar phone-visible\">\n      <button soho-button=\"secondary\" class=\"btn btn-secondary js-btn-save-changes\">{{saveAndCloseButtonLabel}}</button>\n      <button soho-button=\"primary\" class=\"btn btn-primary js-step-link-next\">{{nextButtonLabel}}</button>\n    </div>\n</div>\n" }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { isStepProcessContainer: [{
                type: HostBinding,
                args: ['class.step-process-container']
            }], isTwoColumn: [{
                type: HostBinding,
                args: ['class.two-column']
            }], isFixed: [{
                type: HostBinding,
                args: ['class.fixed']
            }], isPageContainer: [{
                type: HostBinding,
                args: ['class.page-container']
            }], isNoScroll: [{
                type: HostBinding,
                args: ['class.no-scroll']
            }], linearProgression: [{
                type: Input
            }], nextButtonLabel: [{
                type: Input
            }], previousButtonLabel: [{
                type: Input
            }], continueButtonLabel: [{
                type: Input
            }], saveAndCloseButtonLabel: [{
                type: Input
            }], nextButtonEnable: [{
                type: Input
            }], previousButtonEnable: [{
                type: Input
            }], saveCloseButtonEnable: [{
                type: Input
            }], beforeSelectStep: [{
                type: Output
            }], saveClose: [{
                type: Output
            }] } });
