// ids-enterprise-ng/esm2022/lib/stepprocess/soho-stepprocess.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoStepListComponent, SohoStepListTitleComponent, SohoStepListItemComponent, SohoStepListItemAnchorComponent, SohoStepListItemTitleComponent, SohoSubstepListComponent, SohoStepContentTitleComponent, SohoStepContentComponent, SohoStepContentPanelComponent, SohoStepProcessComponent } from '../stepprocess/soho-stepprocess.component';
import { SohoIconModule } from '../icon/soho-icon.module';
import { SohoButtonModule } from '../button/soho-button.module';
import { SohoToolbarModule } from '../toolbar/soho-toolbar.module';
import * as i0 from "@angular/core";
export class SohoStepProcessModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessModule, declarations: [SohoStepListComponent,
            SohoStepListTitleComponent,
            SohoStepListItemComponent,
            SohoStepListItemAnchorComponent,
            SohoStepListItemTitleComponent,
            SohoSubstepListComponent,
            SohoStepContentTitleComponent,
            SohoStepContentComponent,
            SohoStepContentPanelComponent,
            SohoStepProcessComponent], imports: [CommonModule,
            SohoIconModule,
            SohoButtonModule,
            SohoToolbarModule], exports: [SohoStepListComponent,
            SohoStepListTitleComponent,
            SohoStepListItemComponent,
            SohoStepListItemAnchorComponent,
            SohoStepListItemTitleComponent,
            SohoSubstepListComponent,
            SohoStepContentTitleComponent,
            SohoStepContentComponent,
            SohoStepContentPanelComponent,
            SohoStepProcessComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessModule, imports: [CommonModule,
            SohoIconModule,
            SohoButtonModule,
            SohoToolbarModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoStepProcessModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SohoIconModule,
                        SohoButtonModule,
                        SohoToolbarModule
                    ],
                    declarations: [
                        SohoStepListComponent,
                        SohoStepListTitleComponent,
                        SohoStepListItemComponent,
                        SohoStepListItemAnchorComponent,
                        SohoStepListItemTitleComponent,
                        SohoSubstepListComponent,
                        SohoStepContentTitleComponent,
                        SohoStepContentComponent,
                        SohoStepContentPanelComponent,
                        SohoStepProcessComponent
                    ],
                    exports: [
                        SohoStepListComponent,
                        SohoStepListTitleComponent,
                        SohoStepListItemComponent,
                        SohoStepListItemAnchorComponent,
                        SohoStepListItemTitleComponent,
                        SohoSubstepListComponent,
                        SohoStepContentTitleComponent,
                        SohoStepContentComponent,
                        SohoStepContentPanelComponent,
                        SohoStepProcessComponent
                    ]
                }]
        }] });
