// ids-enterprise-ng/esm2022/lib/actionsheet/soho-actionsheet.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoActionsheetComponent } from './soho-actionsheet.component';
import * as i0 from "@angular/core";
export class SohoActionsheetModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoActionsheetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoActionsheetModule, declarations: [SohoActionsheetComponent], imports: [CommonModule], exports: [SohoActionsheetComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoActionsheetModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoActionsheetModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoActionsheetComponent
                    ],
                    exports: [
                        SohoActionsheetComponent
                    ],
                }]
        }] });
