// ids-enterprise-ng/esm2022/lib/card/soho-card.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoCardsComponent, SohoCardActionableComponent, SohoCardHeaderComponent, SohoCardPaneComponent, SohoCardComponent, } from './soho-card.component';
import * as i0 from "@angular/core";
export class SohoCardModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoCardModule, declarations: [SohoCardsComponent,
            SohoCardActionableComponent,
            SohoCardHeaderComponent,
            SohoCardPaneComponent,
            SohoCardComponent], imports: [CommonModule], exports: [SohoCardsComponent,
            SohoCardActionableComponent,
            SohoCardHeaderComponent,
            SohoCardPaneComponent,
            SohoCardComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoCardsComponent,
                        SohoCardActionableComponent,
                        SohoCardHeaderComponent,
                        SohoCardPaneComponent,
                        SohoCardComponent,
                    ],
                    exports: [
                        SohoCardsComponent,
                        SohoCardActionableComponent,
                        SohoCardHeaderComponent,
                        SohoCardPaneComponent,
                        SohoCardComponent,
                    ]
                }]
        }] });
