// ids-enterprise-ng/esm2022/lib/card/soho-card.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoCardsComponent {
    get isCards() {
        return true;
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        this.options = {};
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.cards({
                selectable: this.selectable,
            });
            this.cards = this.jQueryElement.data('cards');
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.cards) {
                this.cards?.destroy();
                this.cards = undefined;
            }
        });
    }
    /** Defines the data to use, must be specified for this component */
    set dataset(dataset) {
        this.options.dataset = dataset;
        if (this.card) {
            this.card.settings.dataset = dataset;
            this.updated(this.card.settings);
        }
    }
    get dataset() {
        if (!this.card) {
            return this.options.dataset;
        }
        return this.card.settings.dataset;
    }
    /** Defines the selection type of cards */
    set selectable(selectable) {
        this.options.selectable = selectable;
        if (this.card) {
            this.card.settings.selectable = selectable;
            this.updated(this.card.settings);
        }
    }
    get selectable() {
        if (!this.card) {
            return this.options.selectable;
        }
        return this.card.settings.selectable;
    }
    updated(settings) {
        this.ngZone.runOutsideAngular(() => this.card?.updated(settings));
        return this;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardsComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCardsComponent, selector: "[soho-cards]", inputs: { dataset: "dataset", selectable: "selectable" }, host: { properties: { "class.cards": "this.isCards" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-cards]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { isCards: [{
                type: HostBinding,
                args: ['class.cards']
            }], dataset: [{
                type: Input
            }], selectable: [{
                type: Input
            }] } });
export class SohoCardActionableComponent {
    get isCardActionable() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardActionableComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCardActionableComponent, selector: "soho-card-actionable", host: { properties: { "class.card-actionable": "this.isCardActionable" } }, ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardActionableComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-card-actionable',
                    template: `
    <ng-content></ng-content>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isCardActionable: [{
                type: HostBinding,
                args: ['class.card-actionable']
            }] } });
export class SohoCardHeaderComponent {
    constructor() {
        this.flex = 'flex';
    }
    get isCardHeader() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCardHeaderComponent, selector: "soho-card-header", host: { properties: { "class.card-header": "this.isCardHeader", "style.display": "this.flex" } }, ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-card-header',
                    template: `
    <ng-content></ng-content>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isCardHeader: [{
                type: HostBinding,
                args: ['class.card-header']
            }], flex: [{
                type: HostBinding,
                args: ['style.display']
            }] } });
export class SohoCardPaneComponent {
    constructor() {
        this.none = 'none';
    }
    get isCardPane() {
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardPaneComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCardPaneComponent, selector: "soho-card-pane", host: { properties: { "class.card-pane": "this.isCardPane", "style.display": "this.none" } }, ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardPaneComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-card-pane',
                    template: `
    <ng-content></ng-content>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { isCardPane: [{
                type: HostBinding,
                args: ['class.card-pane']
            }], none: [{
                type: HostBinding,
                args: ['style.display']
            }] } });
export class SohoCardComponent {
    get isCard() {
        return true;
    }
    get isExpandableHeader() {
        return this.expandableHeader;
    }
    get isAutoHeight() {
        return this.autoHeight;
    }
    set bordered(value) {
        this.options.bordered = value;
        if (this.cards) {
            this.options.bordered = value;
        }
    }
    set noHeader(value) {
        this.options.noHeader = value;
        if (this.cards) {
            this.options.noHeader = value;
        }
    }
    set contentPaddingX(value) {
        this.options.contentPaddingX = value;
        if (this.cards) {
            this.options.contentPaddingX = value;
        }
    }
    set contentPaddingY(value) {
        this.options.contentPaddingY = value;
        if (this.cards) {
            this.options.contentPaddingY = value;
        }
    }
    set noShadow(value) {
        this.options.noShadow = value;
        if (this.cards) {
            this.options.noShadow = value;
        }
    }
    set detailRefId(value) {
        this.options.detailRefId = value;
        if (this.cards) {
            this.options.detailRefId = value;
        }
    }
    set closed(value) {
        this._closed = value;
        if (value && this.jQueryElement) {
            this.close();
        }
    }
    /**
     * Closed the state of the component
     */
    get closed() {
        return this._closed;
    }
    set attributes(value) {
        this.options.attributes = value;
        if (this.cards) {
            this.options.attributes = value;
            this.updated(this.cards.settings);
        }
    }
    get attributes() {
        return this.options.attributes;
    }
    /** Html template string */
    set template(template) {
        this.options.template = template;
        if (this.jQueryElement && this.cards) {
            this.cards.settings.template = template;
            this.updated(this.cards.settings);
        }
    }
    updated(settings) {
        this.ngZone.runOutsideAngular(() => this.cards?.updated(settings));
        return this;
    }
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        this.block = 'block';
        // Add events for Angular elements to listen to
        this.beforeexpand = new EventEmitter();
        this.beforecollapse = new EventEmitter();
        this.expand = new EventEmitter();
        this.collapse = new EventEmitter();
        this.afterexpand = new EventEmitter();
        this.aftercollapse = new EventEmitter();
        this.options = {};
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            if (typeof (this.options.bordered) === 'string' && this.options.bordered !== null) {
                // Convert the string value to a boolean
                this.bordered = this.options.bordered.toLowerCase() === 'true';
            }
        });
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            this.jQueryElement.cards(this.options);
            // Add listeners to emit events
            // Initiate the element via jQuery
            this.jQueryElement.cards({
                id: this.id,
                expandableHeader: this.expandableHeader,
                expanded: !this.closed,
                verticalButtonAction: this.verticalButtonAction,
                attributes: this.options.attributes,
                bordered: this.bordered,
                noHeader: this.noHeader,
                contentPaddingX: this.contentPaddingX,
                contentPaddingY: this.contentPaddingY,
                noShadow: this.noShadow,
                detailRefId: this.detailRefId,
            });
            if (this.toggle) {
                this.toggle.subscribe(value => this.toggleOpen(value));
            }
            // Add listeners to emit events
            this.jQueryElement.on('beforeexpand', (event) => this.onBeforeExpand(event));
            this.jQueryElement.on('beforecollapse', (event) => this.onBeforeCollapse(event));
            this.jQueryElement.on('expand', (event) => this.onExpand(event));
            this.jQueryElement.on('collapse', (event) => this.onCollapse(event));
            this.jQueryElement.on('afterexpand', (event) => this.onAfterExpand(event));
            this.jQueryElement.on('aftercollapse', (event) => this.onAfterCollapse(event));
            this.cards = this.jQueryElement.data('cards');
            if (this.closed) {
                this.toggleOpen(false);
            }
            else if (typeof this.closed !== 'undefined') {
                this.toggleOpen(true);
            }
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            // this.toggle = undefined;
            this.element.nativeElement = undefined;
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.cards) {
                this.cards?.destroy();
                this.cards = undefined;
            }
        });
    }
    /**
     * Toggles the state of the expandable card based on the passed param.
     */
    toggleOpen(open) {
        this.ngZone.runOutsideAngular(() => {
            this._closed = !open;
            if (open) {
                this.cards?.open();
            }
            else {
                this.cards?.close();
            }
        });
    }
    close() {
        this.ngZone.runOutsideAngular(() => this.cards?.close());
    }
    open() {
        this.ngZone.runOutsideAngular(() => this.cards?.open());
    }
    onBeforeExpand(event) {
        // ensure we are back in the angular zone
        this.ngZone.run(() => this.beforeexpand.emit(event));
    }
    onBeforeCollapse(event) {
        this.ngZone.run(() => this.beforecollapse.emit(event));
    }
    onExpand(event) {
        this.ngZone.run(() => this.expand.emit(event));
    }
    onCollapse(event) {
        this.ngZone.run(() => this.collapse.emit(event));
    }
    onAfterExpand(event) {
        this.ngZone.run(() => this.afterexpand.emit(event));
    }
    onAfterCollapse(event) {
        this.ngZone.run(() => this.aftercollapse.emit(event));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCardComponent, selector: "soho-card", inputs: { id: ["soho-card", "id"], expandableHeader: "expandableHeader", verticalButtonAction: "verticalButtonAction", autoHeight: "autoHeight", bordered: "bordered", noHeader: "noHeader", contentPaddingX: "contentPaddingX", contentPaddingY: "contentPaddingY", noShadow: "noShadow", detailRefId: "detailRefId", closed: "closed", attributes: "attributes", template: "template", toggle: "toggle" }, outputs: { beforeexpand: "beforeexpand", beforecollapse: "beforecollapse", expand: "expand", collapse: "collapse", afterexpand: "afterexpand", aftercollapse: "aftercollapse" }, host: { properties: { "style.display": "this.block", "class.card": "this.isCard", "class.expandable-card": "this.isExpandableHeader", "class.auto-height": "this.isAutoHeight" } }, ngImport: i0, template: `
    <ng-content></ng-content>
  `, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'soho-card',
                    template: `
    <ng-content></ng-content>
  `,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { id: [{
                type: Input,
                args: ['soho-card']
            }], expandableHeader: [{
                type: Input
            }], verticalButtonAction: [{
                type: Input
            }], autoHeight: [{
                type: Input
            }], block: [{
                type: HostBinding,
                args: ['style.display']
            }], isCard: [{
                type: HostBinding,
                args: ['class.card']
            }], isExpandableHeader: [{
                type: HostBinding,
                args: ['class.expandable-card']
            }], isAutoHeight: [{
                type: HostBinding,
                args: ['class.auto-height']
            }], bordered: [{
                type: Input
            }], noHeader: [{
                type: Input
            }], contentPaddingX: [{
                type: Input
            }], contentPaddingY: [{
                type: Input
            }], noShadow: [{
                type: Input
            }], detailRefId: [{
                type: Input
            }], closed: [{
                type: Input
            }], attributes: [{
                type: Input
            }], template: [{
                type: Input
            }], toggle: [{
                type: Input
            }], beforeexpand: [{
                type: Output
            }], beforecollapse: [{
                type: Output
            }], expand: [{
                type: Output
            }], collapse: [{
                type: Output
            }], afterexpand: [{
                type: Output
            }], aftercollapse: [{
                type: Output
            }] } });
