// ids-enterprise-ng/esm2022/lib/version/version-initializer.service.mjs
import { Injectable } from '@angular/core';
import packageInfo from '../../version.json';
import * as i0 from "@angular/core";
/**
 * Initialise the locale service BEFORE the root application component is
 * loaded.  Simply add the module to the root module, and the initialiser
 * will be automatically added.
 */
export class SohoVersionInitializerService {
    static { this.VERSION_ATTR_NAME = 'data-ids-enterprise-ng-version'; }
    /**
     * Initializes the version attribute.
     */
    initialize() {
        $('html').attr(SohoVersionInitializerService.VERSION_ATTR_NAME, packageInfo.version);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export function SohoVersionInitializerFactory(service) {
    return () => service.initialize();
}
