// ids-enterprise-ng/esm2022/lib/version/version-initializer.module.mjs
import { NgModule, APP_INITIALIZER } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoVersionInitializerService, SohoVersionInitializerFactory } from './version-initializer.service';
import * as i0 from "@angular/core";
export class SohoVersionInitializerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerModule, imports: [CommonModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerModule, providers: [
            {
                provide: APP_INITIALIZER,
                useFactory: SohoVersionInitializerFactory,
                deps: [SohoVersionInitializerService],
                multi: true
            }
        ], imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoVersionInitializerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [],
                    providers: [
                        {
                            provide: APP_INITIALIZER,
                            useFactory: SohoVersionInitializerFactory,
                            deps: [SohoVersionInitializerService],
                            multi: true
                        }
                    ]
                }]
        }] });
