// ids-enterprise-ng/esm2022/lib/notification-badge/soho-notification-badge.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoNotificationBadgeComponent } from './soho-notification-badge.component';
import * as i0 from "@angular/core";
export class SohoNotificationBadgeModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeModule, declarations: [SohoNotificationBadgeComponent], imports: [CommonModule], exports: [SohoNotificationBadgeComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        SohoNotificationBadgeComponent,
                    ],
                    exports: [
                        SohoNotificationBadgeComponent,
                    ]
                }]
        }] });
