// ids-enterprise-ng/esm2022/lib/notification-badge/soho-notification-badge.component.mjs
import { ChangeDetectionStrategy, Component, Input, } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoNotificationBadgeComponent {
    constructor(element, ngZone) {
        this.element = element;
        this.ngZone = ngZone;
        this.options = {};
    }
    set position(position) {
        this.options.position = position;
        if (this.notificationBadge) {
            this.notificationBadge.settings.position = position;
        }
    }
    get position() {
        if (this.notificationBadge) {
            return this.notificationBadge.settings.position;
        }
        return this.options.position;
    }
    set color(color) {
        this.options.color = color;
        if (this.notificationBadge) {
            this.notificationBadge.settings.color = color;
        }
    }
    get color() {
        if (this.notificationBadge) {
            return this.notificationBadge.settings.color;
        }
        return this.options.color;
    }
    set attributes(attributes) {
        this.options.attributes = attributes;
        if (this.notificationBadge) {
            this.notificationBadge.settings.attributes = attributes;
        }
    }
    get attributes() {
        if (this.notificationBadge) {
            return this.notificationBadge.settings.attributes;
        }
        return this.options.attributes;
    }
    show() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement.find('.notification-dot').removeClass('is-disabled');
        });
    }
    hide() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement.find('.notification-dot').addClass('is-disabled');
        });
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement = jQuery(this.element.nativeElement);
            const options = {
                position: this.position,
                color: this.color,
                attributes: this.attributes
            };
            // Initialize the notification badge component.
            this.jQueryElement.notificationbadge(options);
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoNotificationBadgeComponent, selector: "[soho-notification-badge]", inputs: { position: "position", color: "color", attributes: "attributes" }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoNotificationBadgeComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-notification-badge]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { position: [{
                type: Input
            }], color: [{
                type: Input
            }], attributes: [{
                type: Input
            }] } });
