// ids-enterprise-ng/esm2022/lib/fileupload-advanced/soho-fileupload-advanced.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoFileUploadAdvancedComponent } from './soho-fileupload-advanced.component';
import * as i0 from "@angular/core";
export class SohoFileUploadAdvancedModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFileUploadAdvancedModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoFileUploadAdvancedModule, declarations: [SohoFileUploadAdvancedComponent], imports: [CommonModule], exports: [SohoFileUploadAdvancedComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFileUploadAdvancedModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoFileUploadAdvancedModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoFileUploadAdvancedComponent
                    ],
                    exports: [
                        SohoFileUploadAdvancedComponent
                    ]
                }]
        }] });
