// ids-enterprise-ng/esm2022/lib/calendar/soho-calendar.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, HostBinding, Input, Output, } from '@angular/core';
import * as i0 from "@angular/core";
/********************************************************************
 * Calendar Week View Element
 *******************************************************************/
// @ts-ignore
export class SohoCalendarWeekViewComponent {
    constructor() {
        this.isCalendarWeekView = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarWeekViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCalendarWeekViewComponent, selector: "div[soho-calendar-week-view]", host: { properties: { "class.calendar-weekview": "this.isCalendarWeekView" } }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarWeekViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-calendar-week-view]',
                    template: '<ng-content></ng-content>',
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], propDecorators: { isCalendarWeekView: [{
                type: HostBinding,
                args: ['class.calendar-weekview']
            }] } });
/********************************************************************
 * Calendar Month View Element
 *******************************************************************/
// @ts-ignore
export class SohoCalendarMonthViewComponent {
    constructor() {
        this.isCalendarMonthView = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarMonthViewComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCalendarMonthViewComponent, selector: "div[soho-calendar-monthview]", host: { properties: { "class.calendar-monthview": "this.isCalendarMonthView" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarMonthViewComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-calendar-monthview]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], propDecorators: { isCalendarMonthView: [{
                type: HostBinding,
                args: ['class.calendar-monthview']
            }] } });
/********************************************************************
 * Main Calendar component
 *******************************************************************/
export class SohoCalendarComponent {
    set calendarOptions(calendarOptions) {
        this._calendarOptions = calendarOptions;
        if (this.jQueryElement) {
            // No need to set the 'settings' as the Rebuild will create
            // a new control with the _gridOptions.
            this.markForRefresh();
        }
    }
    get calendarOptions() {
        if (this.calendar) {
            return this.calendar.settings;
        }
        return this._calendarOptions;
    }
    set calendarWeekOptions(calendarWeekOptions) {
        this._calendarWeekOptions = calendarWeekOptions;
        if (this.jQueryElement) {
            // No need to set the 'settings' as the Rebuild will create
            // a new control with the _gridOptions.
            this.markForRefresh();
        }
    }
    get calendarWeekOptions() {
        if (this.calendar) {
            return this.calendar.settings.weekOptions;
        }
        return this._calendarWeekOptions;
    }
    set disable(disable) {
        this._calendarOptions.disable = disable;
        if (this.calendar) {
            this.calendar.settings.disable = disable;
            this.markForRefresh();
        }
    }
    get disable() {
        if (this.calendar) {
            return this.calendar.settings.disable;
        }
        return this.disable;
    }
    set dayLegend(dayLegend) {
        this._calendarOptions.dayLegend = dayLegend;
        if (this.calendar) {
            this.calendar.settings.dayLegend = dayLegend;
            this.markForRefresh();
        }
    }
    get dayLegend() {
        if (this.calendar) {
            return this.calendar.settings.dayLegend;
        }
        return this.dayLegend;
    }
    set attributes(attributes) {
        this._calendarOptions.attributes = attributes;
        if (this.calendar) {
            this.calendar.settings.attributes = attributes;
            this.markForRefresh();
        }
    }
    get attributes() {
        if (this.calendar) {
            return this.calendar.settings.attributes;
        }
        return this.attributes;
    }
    /**
     * An array of objects with data for the event types.
     */
    set eventTypes(eventTypes) {
        this._calendarOptions.eventTypes = eventTypes;
        if (this.calendar) {
            this.calendar.settings.eventTypes = eventTypes;
            this.markForRefresh();
        }
    }
    get eventTypes() {
        if (this.calendar) {
            return this.calendar.settings.eventTypes;
        }
        return this._calendarOptions.eventTypes;
    }
    /**
     * An array of objects with data for the events.
     */
    set events(events) {
        this._calendarOptions.events = events;
        if (this.calendar) {
            this.calendar.settings.events = events;
            this.markForRefresh();
        }
    }
    get events() {
        if (this.calendar) {
            return this.calendar.settings.events;
        }
        return this._calendarOptions.events;
    }
    /**
     * The name of the locale to use for this instance. If not set the current locale will be used.
     */
    set locale(locale) {
        this._calendarOptions.locale = locale;
        if (this.calendar) {
            this.calendar.settings.locale = locale;
            this.markForRefresh();
        }
    }
    get locale() {
        if (this.calendar) {
            return this.calendar.settings.locale;
        }
        return this._calendarOptions.locale;
    }
    /**
     * Initial month to show.
     */
    set month(month) {
        this._calendarOptions.month = month;
        if (this.calendar) {
            this.calendar.settings.month = month;
            this.markForRefresh();
        }
    }
    get month() {
        if (this.calendar) {
            return this.calendar.settings.month;
        }
        return this._calendarOptions.month;
    }
    /**
     * Initial year to show.
     */
    set year(year) {
        this._calendarOptions.year = year;
        if (this.calendar) {
            this.calendar.settings.year = year;
            this.markForRefresh();
        }
    }
    get year() {
        if (this.calendar) {
            return this.calendar.settings.year;
        }
        return this._calendarOptions.year;
    }
    /**
     * If false the dropdown to change views will not be shown.
     */
    set showViewChanger(showViewChanger) {
        this._calendarOptions.showViewChanger = showViewChanger;
        if (this.calendar) {
            this.calendar.settings.showViewChanger = showViewChanger;
            this.markForRefresh();
        }
    }
    get showViewChanger() {
        if (this.calendar) {
            return this.calendar.settings.showViewChanger;
        }
        return this._calendarOptions.showViewChanger;
    }
    /**
     * If false the legend will not show below.
     */
    set showEventLegend(showEventLegend) {
        this._calendarOptions.showEventLegend = showEventLegend;
        if (this.calendar) {
            this.calendar.settings.showEventLegend = showEventLegend;
            this.markForRefresh();
        }
    }
    get showEventLegend() {
        if (this.calendar) {
            return this.calendar.settings.showEventLegend;
        }
        return this._calendarOptions.showEventLegend;
    }
    /**
     * If false the mouseover text or day event will not be shown.
     */
    set eventTooltip(eventTooltip) {
        this._calendarOptions.eventTooltip = eventTooltip;
        if (this.calendar) {
            this.calendar.settings.eventTooltip = eventTooltip;
            this.markForRefresh();
        }
    }
    get eventTooltip() {
        if (this.calendar) {
            return this.calendar.settings.eventTooltip;
        }
        return this._calendarOptions.eventTooltip;
    }
    /**
     * If false the mouseover text for event icon will not be shown.
     */
    set iconTooltip(iconTooltip) {
        this._calendarOptions.iconTooltip = iconTooltip;
        if (this.calendar) {
            this.calendar.settings.iconTooltip = iconTooltip;
            this.markForRefresh();
        }
    }
    get iconTooltip() {
        if (this.calendar) {
            return this.calendar.settings.iconTooltip;
        }
        return this._calendarOptions.iconTooltip;
    }
    /**
     * Fires when a month is rendered, allowing you to pass back events or event types to show.
     */
    set renderMonthCallback(renderMonthCallback) {
        this._calendarOptions.onRenderMonth = renderMonthCallback;
        if (this.calendar) {
            this.calendar.settings.onRenderMonth = renderMonthCallback;
            this.markForRefresh();
        }
    }
    get renderMonthCallback() {
        if (this.calendar) {
            return this.calendar.settings.onRenderMonth;
        }
        return this._calendarOptions.onRenderMonth;
    }
    /**
     * Fires when a month day is clicked. Allowing you to do something.
     */
    set selectedCallback(selectedCallback) {
        this._calendarOptions.onSelected = selectedCallback;
        if (this.calendar) {
            this.calendar.settings.onSelected = selectedCallback;
            this.markForRefresh();
        }
    }
    get selectedCallback() {
        if (this.calendar) {
            return this.calendar.settings.onSelected;
        }
        return this._calendarOptions.onSelected;
    }
    /**
     * The ID of the template used for the events.
     */
    set template(template) {
        this._calendarOptions.template = template;
        if (this.calendar) {
            this.calendar.settings.template = template;
            this.markForRefresh();
        }
    }
    get template() {
        if (this.calendar) {
            return this.calendar.settings.template;
        }
        return this._calendarOptions.template;
    }
    /**
     * How many days in advance should we show in the upcoming events pane.
     */
    set upcomingEventDays(upcomingEventDays) {
        this._calendarOptions.upcomingEventDays = upcomingEventDays;
        if (this.calendar) {
            this.calendar.settings.upcomingEventDays = upcomingEventDays;
            this.markForRefresh();
        }
    }
    get upcomingEventDays() {
        if (this.calendar) {
            return this.calendar.settings.upcomingEventDays;
        }
        return this._calendarOptions.upcomingEventDays;
    }
    /**
     * The ID of the template used for the modal dialog on events.
     */
    set modalTemplate(modalTemplate) {
        this._calendarOptions.modalTemplate = modalTemplate;
        if (this.calendar) {
            this.calendar.settings.modalTemplate = modalTemplate;
            this.markForRefresh();
        }
    }
    get modalTemplate() {
        if (this.calendar) {
            return this.calendar.settings.modalTemplate;
        }
        return this._calendarOptions.modalTemplate;
    }
    /**
     * ID of the menu to use for an event right click context menu
     */
    set menuId(menuId) {
        this._calendarOptions.menuId = menuId;
        if (this.calendar) {
            this.calendar.settings.menuId = menuId;
            this.markForRefresh();
        }
    }
    get menuId() {
        if (this.calendar) {
            return this.calendar.settings.menuId;
        }
        return this._calendarOptions.menuId;
    }
    /**
     * Callback for the  right click context menu
     */
    set menuSelected(menuSelected) {
        this._calendarOptions.menuSelected = menuSelected;
        if (this.calendar) {
            this.calendar.settings.menuSelected = menuSelected;
            this.markForRefresh();
        }
    }
    get menuSelected() {
        if (this.calendar) {
            return this.calendar.settings.menuSelected;
        }
        return this._calendarOptions.menuSelected;
    }
    /**
     * Initial event properties for the new events dialog.
     */
    set newEventDefaults(newEventDefaults) {
        this._calendarOptions.newEventDefaults = newEventDefaults;
        if (this.calendar) {
            this.calendar.settings.newEventDefaults = newEventDefaults;
            this.markForRefresh();
        }
    }
    get newEventDefaults() {
        if (this.calendar) {
            return this.calendar.settings.newEventDefaults;
        }
        return this._calendarOptions.newEventDefaults;
    }
    /**
     * Call back for when the view changer is changed
     */
    set changeViewCallback(changeViewCallback) {
        this._calendarOptions.onChangeView = changeViewCallback;
        if (this.calendar) {
            this.calendar.settings.onChangeView = changeViewCallback;
            this.markForRefresh();
        }
    }
    get changeViewCallback() {
        if (this.calendar) {
            return this.calendar.settings.onChangeView;
        }
        return this._calendarOptions.onChangeView;
    }
    /**
     * Deterimines if the today button should be shown.
     */
    set showToday(showToday) {
        this._calendarOptions.showToday = showToday;
        if (this.calendar) {
            this.calendar.settings.showToday = showToday;
            this.markForRefresh();
        }
    }
    get showToday() {
        if (this.calendar) {
            return this.calendar.settings.showToday;
        }
        return this._calendarOptions.showToday;
    }
    /**
     * Configure date range for calendar
     */
    set displayRange(displayRange) {
        this._calendarOptions.displayRange = displayRange;
        if (this.calendar) {
            this.calendar.settings.displayRange = displayRange;
            this.markForRefresh();
        }
    }
    get displayRange() {
        if (this.calendar) {
            return this.calendar.settings.displayRange;
        }
        return this._calendarOptions.displayRange;
    }
    /**
     * Configure day of week setting
     */
    set firstDayOfWeek(day) {
        this._calendarOptions.firstDayOfWeek = day;
        if (this.calendar) {
            this.calendar.settings.firstDayOfWeek = day;
            this.markForRefresh();
        }
    }
    get firstDayOfWeek() {
        if (this.calendar) {
            return this.calendar.settings.firstDayOfWeek;
        }
        return this.firstDayOfWeek;
    }
    constructor(element, ngZone, ref) {
        this.element = element;
        this.ngZone = ngZone;
        this.ref = ref;
        this.isCalendar = true;
        // -------------------------------------------
        // Component Output
        // -------------------------------------------
        this.selected = new EventEmitter();
        this.monthRendered = new EventEmitter();
        this.eventClick = new EventEmitter();
        this.eventDblClick = new EventEmitter();
        this.eventContextMenu = new EventEmitter();
        this._calendarOptions = {};
        this._calendarWeekOptions = {};
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            // Wrap the element in a jQuery selector.
            this.jQueryElement = jQuery(this.element.nativeElement);
            // Add listeners to emit events
            this.jQueryElement
                .on('selected', (_e, event) => this.onSelectedEvent(event))
                .on('monthrendered', (_e, args) => this.onMonthRenderedEvent(args))
                .on('eventclick', (_e, args) => this.onEventClick(args))
                .on('eventdblclick', (_e, args) => this.onEventDblClick(args))
                .on('contextmenu', (_e, args) => this.onEventContextMenu(args));
            // Initialise the Soho control.
            this.jQueryElement.calendar(this._calendarOptions);
            // Once the control is initialised, extract the control
            // plug-in from the element.  The element name is defined
            // by the plug-in, but in this case is 'calendar'.
            this.calendar = this.jQueryElement.data('calendar');
        });
    }
    ngAfterViewChecked() {
        if (!this.calendar || !this.jQueryElement) {
            return;
        }
        if (this.updateRequired) {
            // call outside the angular zone so change detection isn't triggered by the soho component.
            this.updated();
            this.updateRequired = false;
        }
    }
    onSelectedEvent(event) {
        this.ngZone.run(() => this.selected.emit(event));
    }
    onMonthRenderedEvent(event) {
        this.ngZone.run(() => this.monthRendered.emit(event));
    }
    onEventClick(event) {
        this.ngZone.run(() => this.eventClick.emit(event));
    }
    onEventDblClick(event) {
        this.ngZone.run(() => this.eventDblClick.emit(event));
    }
    onEventContextMenu(event) {
        this.ngZone.run(() => this.eventContextMenu.emit(event));
    }
    /**
     * Get the current selected date on the calendar.
     *
     * @returns the currently selected date on the control.
     */
    currentDate() {
        return this.ngZone.runOutsideAngular(() => this.calendar.currentDate());
    }
    /**
     * Get the events and date for the currently selected calendar day.
     *
     * @param date The date to find the events for.
     * @returns dayEvents An object with all the events and the event date.
     */
    getDayEvents(date) {
        return this.ngZone.runOutsideAngular(() => this.calendar.getDayEvents(date));
    }
    /**
     * Add a new event via the event object and show it if it should be visible in the calendar.
     *
     * @param event The event object with common event properties.
     */
    addEvent(event) {
        this.ngZone.runOutsideAngular(() => this.calendar.addEvent(event));
    }
    /**
     * Update an event via the event object and show it if it should be visible in the calendar.
     * It uses the event id to do this.
     *
     * @param event The event object with common event properties.
     */
    updateEvent(event) {
        this.ngZone.runOutsideAngular(() => this.calendar.updateEvent(event));
    }
    /**
     * Remove an event from the dataset and page. It uses the id property.
     *
     * @param event The event object with common event properties.
     */
    deleteEvent(event) {
        this.ngZone.runOutsideAngular(() => this.calendar.deleteEvent(event));
    }
    /**
     * Show a modal used to add/edit events. This uses the modalTemplate setting for the modal contents.
     *
     * @param event The event object with common event properties for defaulting fields in the template.
     * @param done The callback for when the modal closes.
     */
    showEventModal(event, done) {
        this.ngZone.runOutsideAngular(() => this.calendar.showEventModal(event, done));
    }
    /**
     * @returns whether or not this Modal is currently being displayed
     */
    modalVisible() {
        return this.ngZone.runOutsideAngular(() => this.calendar.modalVisible());
    }
    /**
     * Remove all events from the calendar
     */
    clearEvents() {
        this.ngZone.runOutsideAngular(() => this.calendar.clearEvents());
    }
    /**
     * Handle updated settings and values.
     */
    updated(settings) {
        this.ngZone.runOutsideAngular(() => this.calendar.updated(settings));
    }
    /**
     * Marks the components as requiring a rebuild after the next update.
     */
    markForRefresh() {
        // Run updated on the next updated check.
        this.updateRequired = true;
        // ... make sure the change detector kicks in, otherwise if the inputs
        // were change programmatically the component may not be eligible for
        // updating.
        this.ref.markForCheck();
    }
    /**
     * Destructor.
     */
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.jQueryElement) {
                this.jQueryElement.off();
                this.jQueryElement = undefined;
            }
            if (this.calendar) {
                this.calendar.destroy();
                this.calendar = null;
            }
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarComponent, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoCalendarComponent, selector: "[soho-calendar]", inputs: { calendarOptions: "calendarOptions", calendarWeekOptions: "calendarWeekOptions", disable: "disable", dayLegend: "dayLegend", attributes: "attributes", eventTypes: "eventTypes", events: "events", locale: "locale", month: "month", year: "year", showViewChanger: "showViewChanger", showEventLegend: "showEventLegend", eventTooltip: "eventTooltip", iconTooltip: "iconTooltip", renderMonthCallback: "renderMonthCallback", selectedCallback: "selectedCallback", template: "template", upcomingEventDays: "upcomingEventDays", modalTemplate: "modalTemplate", menuId: "menuId", menuSelected: "menuSelected", newEventDefaults: "newEventDefaults", changeViewCallback: "changeViewCallback", showToday: "showToday", displayRange: "displayRange", firstDayOfWeek: "firstDayOfWeek" }, outputs: { selected: "selected", monthRendered: "monthRendered", eventClick: "eventClick", eventDblClick: "eventDblClick", eventContextMenu: "eventContextMenu" }, host: { properties: { "class.calendar": "this.isCalendar" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-calendar]',
                    template: `<ng-content></ng-content>`,
                    changeDetection: ChangeDetectionStrategy.OnPush
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { isCalendar: [{
                type: HostBinding,
                args: ['class.calendar']
            }], calendarOptions: [{
                type: Input
            }], calendarWeekOptions: [{
                type: Input
            }], disable: [{
                type: Input
            }], dayLegend: [{
                type: Input
            }], attributes: [{
                type: Input
            }], eventTypes: [{
                type: Input
            }], events: [{
                type: Input
            }], locale: [{
                type: Input
            }], month: [{
                type: Input
            }], year: [{
                type: Input
            }], showViewChanger: [{
                type: Input
            }], showEventLegend: [{
                type: Input
            }], eventTooltip: [{
                type: Input
            }], iconTooltip: [{
                type: Input
            }], renderMonthCallback: [{
                type: Input
            }], selectedCallback: [{
                type: Input
            }], template: [{
                type: Input
            }], upcomingEventDays: [{
                type: Input
            }], modalTemplate: [{
                type: Input
            }], menuId: [{
                type: Input
            }], menuSelected: [{
                type: Input
            }], newEventDefaults: [{
                type: Input
            }], changeViewCallback: [{
                type: Input
            }], showToday: [{
                type: Input
            }], displayRange: [{
                type: Input
            }], firstDayOfWeek: [{
                type: Input
            }], selected: [{
                type: Output
            }], monthRendered: [{
                type: Output
            }], eventClick: [{
                type: Output
            }], eventDblClick: [{
                type: Output
            }], eventContextMenu: [{
                type: Output
            }] } });
