// ids-enterprise-ng/esm2022/lib/calendar/soho-calendar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoCalendarComponent, SohoCalendarMonthViewComponent, SohoCalendarWeekViewComponent } from './soho-calendar.component';
import * as i0 from "@angular/core";
export class SohoCalendarModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarModule, declarations: [SohoCalendarComponent,
            SohoCalendarMonthViewComponent,
            SohoCalendarWeekViewComponent], imports: [CommonModule], exports: [SohoCalendarComponent,
            SohoCalendarMonthViewComponent,
            SohoCalendarWeekViewComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoCalendarModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                    ],
                    declarations: [
                        SohoCalendarComponent,
                        SohoCalendarMonthViewComponent,
                        SohoCalendarWeekViewComponent
                    ],
                    exports: [
                        SohoCalendarComponent,
                        SohoCalendarMonthViewComponent,
                        SohoCalendarWeekViewComponent
                    ]
                }]
        }] });
