// ids-enterprise-ng/esm2022/lib/label/soho-label.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoLabelDirective {
    /**
     * Bind attributes to the host label element
     */
    get isLabel() {
        return !this.forCheckBox && !this.forRadioButton;
    }
    get isCheckBoxLabel() {
        return this.forCheckBox && !this.forRadioButton;
    }
    get isRadioButtonLabel() {
        return this.forRadioButton && !this.forCheckBox;
    }
    get isAudible() {
        return this.audible ? true : false;
    }
    get isRequired() {
        return this.required ? true : false;
    }
    constructor(element) {
        this.element = element;
    }
    ngAfterViewInit() {
        this.jQueryElement = jQuery(this.element.nativeElement);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoLabelDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoLabelDirective, selector: "label[soho-label]", inputs: { audible: "audible", required: "required", forCheckBox: "forCheckBox", forRadioButton: "forRadioButton" }, host: { properties: { "class.label": "this.isLabel", "class.checkbox-label": "this.isCheckBoxLabel", "class.radio-label": "this.isRadioButtonLabel", "class.audible": "this.isAudible", "class.required": "this.isRequired" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoLabelDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'label[soho-label]' // eslint-disable-line
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { audible: [{
                type: Input
            }], required: [{
                type: Input
            }], forCheckBox: [{
                type: Input
            }], forRadioButton: [{
                type: Input
            }], isLabel: [{
                type: HostBinding,
                args: ['class.label']
            }], isCheckBoxLabel: [{
                type: HostBinding,
                args: ['class.checkbox-label']
            }], isRadioButtonLabel: [{
                type: HostBinding,
                args: ['class.radio-label']
            }], isAudible: [{
                type: HostBinding,
                args: ['class.audible']
            }], isRequired: [{
                type: HostBinding,
                args: ['class.required']
            }] } });
