// ids-enterprise-ng/esm2022/lib/homepage/soho-widget-header.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoWidgetHeaderComponent {
    constructor() {
        this.isWidgetHeader = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWidgetHeaderComponent, selector: "div[soho-widget-header]", host: { properties: { "class.widget-header": "this.isWidgetHeader" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-widget-header]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isWidgetHeader: [{
                type: HostBinding,
                args: ['class.widget-header']
            }] } });
