// ids-enterprise-ng/esm2022/lib/homepage/soho-widget-title.component.mjs
import { Component, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoWidgetTitleComponent {
    constructor() {
        this.isWidgetTitle = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetTitleComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWidgetTitleComponent, selector: "[soho-widget-title]", inputs: { tabIndex: "tabIndex" }, host: { properties: { "attr.tabindex": "this.tabIndex", "class.widget-title": "this.isWidgetTitle" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetTitleComponent, decorators: [{
            type: Component,
            args: [{
                    selector: '[soho-widget-title]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { tabIndex: [{
                type: HostBinding,
                args: ['attr.tabindex']
            }, {
                type: Input
            }], isWidgetTitle: [{
                type: HostBinding,
                args: ['class.widget-title']
            }] } });
