// ids-enterprise-ng/esm2022/lib/homepage/soho-homepage.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SohoHomePageComponent } from './soho-homepage.component';
import { SohoWidgetComponent } from './soho-widget.component';
import { SohoWidgetContentComponent } from './soho-widget-content.component';
import { SohoWidgetHeaderComponent } from './soho-widget-header.component';
import { SohoWidgetTitleComponent } from './soho-widget-title.component';
import { SohoHomepageSizerDirective } from './soho-homepage-sizer.directive';
import * as i0 from "@angular/core";
export class SohoHomePageModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHomePageModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.4", ngImport: i0, type: SohoHomePageModule, declarations: [SohoHomePageComponent,
            SohoHomepageSizerDirective,
            SohoWidgetComponent,
            SohoWidgetContentComponent,
            SohoWidgetHeaderComponent,
            SohoWidgetTitleComponent], imports: [CommonModule], exports: [SohoHomePageComponent,
            SohoHomepageSizerDirective,
            SohoWidgetComponent,
            SohoWidgetContentComponent,
            SohoWidgetHeaderComponent,
            SohoWidgetTitleComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHomePageModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHomePageModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [
                        SohoHomePageComponent,
                        SohoHomepageSizerDirective,
                        SohoWidgetComponent,
                        SohoWidgetContentComponent,
                        SohoWidgetHeaderComponent,
                        SohoWidgetTitleComponent
                    ],
                    exports: [
                        SohoHomePageComponent,
                        SohoHomepageSizerDirective,
                        SohoWidgetComponent,
                        SohoWidgetContentComponent,
                        SohoWidgetHeaderComponent,
                        SohoWidgetTitleComponent
                    ]
                }]
        }] });
