// ids-enterprise-ng/esm2022/lib/homepage/soho-homepage-sizer.directive.mjs
import { Directive, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Angular Directive for keeping a homepage elements height correctly up to date.
 */
export class SohoHomepageSizerDirective {
    get heightStyle() {
        return this.containerHeight;
    }
    constructor(elementRef, ngZone) {
        this.elementRef = elementRef;
        this.ngZone = ngZone;
    }
    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            // Wrap for later.
            this.jQueryElement = jQuery(this.elementRef.nativeElement);
            this.jQueryElement.on('resize', (_e, _columns, stats) => this.ngZone.run(() => setTimeout(() => this.containerHeight = stats.containerHeight)));
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            this.jQueryElement?.off();
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHomepageSizerDirective, deps: [{ token: i0.ElementRef }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.4", type: SohoHomepageSizerDirective, selector: "[soho-homepage-sizer]", host: { properties: { "style.height.px": "this.heightStyle" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoHomepageSizerDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[soho-homepage-sizer]', // eslint-disable-line
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.NgZone }]; }, propDecorators: { heightStyle: [{
                type: HostBinding,
                args: ['style.height.px']
            }] } });
