// ids-enterprise-ng/esm2022/lib/homepage/soho-widget-content.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class SohoWidgetContentComponent {
    constructor() {
        this.isWidgetContent = true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetContentComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.4", type: SohoWidgetContentComponent, selector: "div[soho-widget-content]", host: { properties: { "class.widget-content": "this.isWidgetContent" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.4", ngImport: i0, type: SohoWidgetContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'div[soho-widget-content]',
                    template: `<ng-content></ng-content>`
                }]
        }], propDecorators: { isWidgetContent: [{
                type: HostBinding,
                args: ['class.widget-content']
            }] } });
